# WRC18_RhizoTeaDecomp
"Experimental nutrient enrichment increases bacterial diversity and decomposition rates of different litter types in a coastal plain wetland" 

This study was conducted at the long-term fertilization and mowing experiment at West Research Campus, East Carolina University.

This repository contains open-source code, data, & text files.

For information regarding the project, please visit: Koceja, M.E., R.B. Bledsoe§, C. Goodwillie, and A.L. Peralta (In revision). Distinct microbial communities alter litter decomposition rates in a fertilized coastal plain wetland. *[bioRxiv pre-print](https://doi.org/10.1101/732883)*

Raw sequence files can be found at NCBI SRA [BioProject ID PRJNA635951](https://www.ncbi.nlm.nih.gov/bioproject/635951)

## We address the following questions

* **Aim 1.)**: How does long-term fertilization and disturbance (by mowing) and litter chemical quality influence litter-associated microbiomes and decomposition rates?

* **Aim 2.)**: What are the specific bacterial taxa associated with decomposition of high vs. low C:N ratio litter types? 

### Repo Contents

* **analyses:** An R Markdown file (2018_WRC_SoilTeaDecomp.Rmd) that includes R script written by Ariane Peralta and Regina Bledsoe containing functions used in analysis of soil and microbial sequence data.

* **bin:** 
	* *MothurTools.R*: An R script written by Mario Muscarella (currently at Université du Quebéc à Montréal, soon to be Assistant Professor, University of Alaska Fairbanks) containing functions used in the analysis of community sequence data.

* **data:** Files associated with soil and microbial data sets. 

* **figures:** Figures (of plant community composition) generated according to R script located in R Markdown file.

* **mothur:** Files containing script for bioinformatic analysis of 16S rRNA gene sequences (Illumina MiSeq platform) using mothur pipeline.

## Funding Sources  
This work was supported by the East Carolina University Undergraduate Research and Creative Activity Award to MEK and the National Science Foundation (GRFP to RBB, DEB 1845845 to ALP).

## Contributors
Megan E. Koceja: Undergraduate Research Assistant, Department of Biology, East Carolina University.

[Dr. Regina Bledsoe](mailto:ginabbledsoe@gmail.com) [(website)](https://ginabbledsoe.wixsite.com/microbes): Co-Principal Investigator, PhD Student in the [Peralta Lab](http://www.peraltalab.com) at East Carolina University

[Dr. Carol Goodwillie](mailto:goodwilliec@ecu.edu) [(website)](https://www.goodwillielab.com/): Co-Principal Investigator, Assistant Professor, Department of Biology, East Carolina University.

[Dr. Ariane Peralta](mailto:peraltaa@ecu.edu) [(website)](http://www.peraltalab.com): Principal Investigator, Assistant Professor, Department of Biology, East Carolina University. 
