---
title: "WRC2018_Rhizo"
author: "Regina A. B. Bledsoe, Ariane L. Peralta"
date: "May 8, 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning=FALSE)
#use to set working directory 
knitr::opts_knit$set(root.dir="../GitHub/WRC18_RhizoTeaDecomp/analyses")

```

```{r load packages, include=FALSE}
getwd()

#Set source R tools
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")
source("../bin/R_rainclouds.R")

#load req'd packages 
require("vegan")
require("dplyr")
require("tidyverse")
require("phyloseq")
require("agricolae")
require("ape")
require("picante")
require("reshape2")
require("reshape")
require("ade4")
require("labdsv")
require("ggpubr")

#Set Std Err and Conf Int
se <- function(x, ...) {
  sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))
}
ci <- function(x, ...) {
  1.96 * sd(x, na.rm = TRUE)
}

set.seed(42)
```

```{r load otu tax and design data, include=FALSE}
getwd()
#Load data files 
#Assign file names to variables
sharedfile = "../data/WRC18_Rhizo/WRC15_18Rhizo.opti_mcc.shared"
taxfile = "../data/WRC18_Rhizo/WRC15_18Rhizo.opti_mcc.0.03.cons.taxonomy"
metafile = "../data/WRC18_Rhizo/2015_2018_WestResearchCampus_Sampling_Summary.csv"
fungalfile = "../data/WRC18_Rhizo/fungal_otu_table_rhizo.csv"

#Read in design file
meta <- read.csv(metafile)
meta <- meta %>%
  filter(year=='2018')%>%
  filter(treatment=='M'|treatment=='MF')%>%
  filter(block=='2'|block=='4'|block=='6'|block=='8')

```

```{r read in fungal OTUs}
#Read in fungal file
fungal_otu <- read.csv(fungalfile)

fungal_otu <- fungal_otu %>%
  column_to_rownames("OTU_ID")

fungal_otu <- t(fungal_otu)

#Removing mock community
fungal_otu <- fungal_otu[-c(1:2),]

#fungal taxa file - read in a clean up underscores (_) and "unidentified"
fungal_taxa <- read.delim("../data/WRC18_Rhizo/otu_taxa_table_v1.txt", header=T, sep="\t")

fungal_taxa_trim <- fungal_taxa %>%
  mutate(k = substr(kingdom, 4, str_length(kingdom)), p = substr(phylum, 4, str_length(phylum)),c = substr(class, 4, str_length(class)), o = substr(order, 4, str_length(order)),  f = substr(family, 4, str_length(family)), g = substr(genus, 4, str_length(genus)), s = substr(species, 4, str_length(species))) %>%
  dplyr::select(OTU_ID,domain=k,phylum=p,class=c,order=o,family=f,genus=g,species=s)%>%
  mutate(phylum = str_replace_all(phylum,"_", " "), class = str_replace_all(class,"_", " "), order = str_replace_all(order,"_", " "), family = str_replace_all(family,"_", " "), genus = str_replace_all(genus,"_", " "), species = str_replace_all(species,"_", " "))%>%
  mutate(phylum = str_replace(phylum,"unidentified", " "), class = str_replace(class,"unclassified", " "), order = str_replace(order,"unidentified", " "), family = str_replace(family,"unidentified", " "), genus = str_replace(genus,"unidentified", " "), species = str_replace(species,"unidentified", " "))

```

```{r Fungal Relative abundance and PERMANOVA}
b <- ncol(fungal_otu)
reads <- sum(colSums(fungal_otu))
#total reads before filtering = 229296


fungal_otu <- fungal_otu[, which(colSums(fungal_otu) > 10)]

fun_otu<- colSums(t(fungal_otu))
fun_otu
fun_otu[which.min(colSums(t(fungal_otu)))]
fun_otu[which.max(colSums(t(fungal_otu)))]

#Rarefy
#Set min sample number
min.N <- min(rowSums(fungal_otu))
#rrarefy returns a randomly rarefied community data frame or vector of selected size
#This is the otu set that should be used for beta diversity metrics
fungal_otu <- rrarefy(fungal_otu, min.N)



#transforms read counts to relative abundance
#Using rarefied OTU dataset
fungal_otu.rel <- fungal_otu
for(i in 1:dim(fungal_otu)[1]){
  fungal_otu.rel[i,] <- fungal_otu[i,]/sum(fungal_otu[i,])
}

meta <- meta %>%
  filter(year==2018) %>%
  filter(treatment=="M" | treatment =="MF") %>%
  filter(block==2 | block==4 | block==6 | block==8)

#Combine experimental design meta data with OTU relative abundance data
fun_metaotu <- cbind(meta, fungal_otu.rel)
rownames(fun_metaotu) <- meta$sample


#PERMANOVA with source and treatment as factors
#Among all treatments source (soil source) and treatment
fun_metaotu.ad <- adonis(fun_metaotu[,-c(1:26)] ~source*treatment, method = "bray", data = fun_metaotu, perm=1000, set.seed=42)

fun_metaotu.ad 

```

```{r Fungal Calculate distance matrix and PCoA components - Taxa/Otus}

fungal_otu.rel.dist<-vegdist(fungal_otu.rel, method="bray")

# Principal Coordinates Analysis
fungal_otu.rel.dist.pcoa <- cmdscale(fungal_otu.rel.dist, k=3, eig=TRUE, add=FALSE)
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1a <- round(fungal_otu.rel.dist.pcoa$eig[1] / sum(fungal_otu.rel.dist.pcoa$eig), 3) * 100
explainvar2a <- round(fungal_otu.rel.dist.pcoa$eig[2] / sum(fungal_otu.rel.dist.pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)

explainvar1a #25.3
explainvar2a #11.0

```

```{r PCoA Plot based on Taxa/Otus}
rownames(fungal_otu.rel) <-meta$sample
all.equal(rownames(meta), rownames(fungal_otu.rel))
# Set treatments
treat1 <- as.factor(meta$source)
levels(treat1) <- ordered(c("bulk","grass", "forb"))
treat2 <- as.factor(meta$treatment)
#levels(treat2) <- c("M", "MF")

pcoa.groups <- paste(meta$source, meta$treatment, sep = "_")
pcoa.points <- data.frame(fungal_otu.rel.dist.pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # treatment
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # source

levels(pcoa.col) <- c("M","MF")

#Run envfit to see which soil varibales are predicators 
#These are just the environmental parameters have values for bulk and rhizospehre soils
fit<-envfit(fungal_otu.rel.dist.pcoa$points, meta[,-c(1:14, 17,20,22:25,27:33)], perm=999, na.rm=T, set.seed=42)
fit

A <-as.list(fit$vectors)
vec<-as.data.frame(fit$vectors$arrows*sqrt(fit$vectors$r)*0.35)
p<- as.data.frame(A$pvals)
vec <- cbind(vec, p) 
vec <-  subset(vec, A$pvals<=0.05)

vec$parm<-rownames(vec)
colnames(vec)
vec$parm<-c("C%","N%","NH4", "PO4_HCl", "OM%")

#plot PCoA
df1a <- as.data.frame(pcoa.cent.dataframe)
plot1 <- ggplot(df1a, aes(x=V1, y=V2), group = interaction(pcoa.col, pcoa.shape)) + 
  theme_bw() +
#group = interaction(pcoa.col, pcoa.shape), colour=pcoa.col2))
  geom_point(aes(shape = pcoa.shape, colour = pcoa.col), size=10, stroke = 2) + 
  #geom_point(aes(colour=pcoa.col2), size =4.0)+
  #removes gridlines from plot
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black"))+ 
  #Plot soil property vectors from envfit
  geom_segment(data=vec, aes(x=0,xend=Dim1,y=0,yend=Dim2), size=1, 
     arrow = arrow(length = unit(0.2, "cm")),colour="black",inherit.aes=F)+ 
  #Label soil vectors. Using annotate because I can't figure out how to move segment text
  #Note when re-run should double check envfit stats to make sure that soil parameters are the same
  #geom_text(data=vec,aes(x=Dim1,y=Dim2,label=vec$parm),size=5, inherit.aes=F)+
  annotate("text",x=-0.15,y=0.14, label="C%") +
  annotate("text",x=-0.17,y=0.17, label="N%") +
  annotate("text",x=-0.07,y=0.2, label="OM%") +
  annotate("text",x=-0.27,y=0.005, label="PO4-HCl") +
  annotate("text",x=-0.02,y=-.22, label="NH4") +
  #Set error bacrs for geom_point+ 
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0), colour="black") +
  #Set colors for treatments
  #scale_colour_manual(values = c("#999999", "#000000")) +
  scale_colour_manual(values = c("gray", "darkgreen"), labels=c("Unfertilized", "Fertilized")) +
  #Would have liked to set shapes but couldn't get this to work
  scale_shape_manual(values = c(16,22,15)) +
  #Sets map coordinates
  #coord_cartesian(xlim = c(-0.2, 0.2), ylim = c(-0.2, 0.2)) +
  #Sets axis text and put border around plot
  theme(axis.title = element_text(size=14),# face="bold"), 
        axis.text.x = element_text(size=14, color="black"),  axis.text.y = element_text(size=14, color="black"),
        panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=14)) +
  #Set legend text size
  theme(legend.text=element_text(size=14, face="bold"), legend.title = element_text(size=14, face="bold"))+
  #Sets size of tick marks on axis
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  #Sets labels for plot title, axis titles, and legend headings
  xlab("PCoA 1 (25.3%)") + ylab("PCoA 2 (11.0%)") + 
  labs(shape = "Soil Source") +
  labs(colour="Treatment") +
  ggtitle("Fungal ITS2 Taxonomic Composition")+
  theme(legend.position="right")+
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))
  
plot1 

#Save a copy of the plot
ggsave("../figures/WRC18_Rhizo/pub/ITS2_WRC2018Rhizo.jpg", plot=plot1, device=NULL, path=NULL, scale=1, width=7.5, height=6, dpi=900, limitsize=TRUE, bg="transparent")

ggsave("../figures/WRC18_Rhizo/pub/ITS2_WRC2018Rhizo.pdf", plot=plot1, device=NULL, path=NULL, scale=1, width=7.5, height=6, dpi=900, limitsize=TRUE, bg="transparent")

```

```{r fungal Presence_Absence and Species richness, include =FALSE }
#Transform data to reflect only presence or absence ie 0 or 1
fun.otu.2.PA<-(fungal_otu.rel>0*1)
#sanity check 
#otu.2.PA[,1:5]
#Add meta data
fun.metaotu.2 <- cbind(meta, fun.otu.2.PA)
#metaotu.2[1:15]
#Transform True/False to 0/1 
cols <- sapply(fun.metaotu.2, is.logical)
fun.metaotu.2[,cols]<-lapply(fun.metaotu.2[,cols],as.numeric)
#head(metaotu.2)
```

```{r fungal diversity metrics: richness}
#chao1 richness index
fun.chao1 <- estimateR(fungal_otu)
c <- as.data.frame(fun.chao1)
ct <- as.data.frame(t(c))

meta$fun.s.obs <- ct$S.obs
meta$fun.s.chao1 <- ct$S.chao1
meta$fun.se.chao1 <- ct$se.chao1

#interaction variable so can use Tukey
i <- with(meta, interaction(treatment, source))

fun.richness.lm <- aov(fun.s.chao1 ~ treatment * source + i, data = meta)
plot(fun.richness.lm)
anova(fun.richness.lm)

#Get Tukey post hoc groups p<=0.01*
t <- HSD.test(fun.richness.lm, "i", group=TRUE, alpha=.05)
t
plot(t)

#Plot
f_chao<- ggplot(meta, aes(x=source, y=fun.s.chao1, fill=treatment))  +
  geom_boxplot(binaxis='y', stackdir='center') +
  scale_fill_manual(values=c("gray", "darkgreen"), labels=c("Unfertilized","Fertilized")) +
  geom_point(aes(fill=treatment), size = 2, shape = 21, position = position_jitterdodge()) +
#Set labels for Tukey groups
#annotate("text",x=0.82,y=5200, label="b") +
#annotate("text",x=1.81,y=5400, label="b") +
#annotate("text",x=2.81,y=5725, label="b") +
#annotate("text",x=1.1815,y=7050, label="a") +
#annotate("text",x=2.19,y=6600, label="a") + 
#annotate("text",x=3.19,y=6250, label="a") + 
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=14),
          axis.text=element_text(size=14, color="black"), axis.text.x = element_blank(),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          #Set plot title textsize
          theme(plot.title=element_text(size=14)) +
         labs(x = "", y = "Chao1 richness", title = "Fungal", fill="Treatment") +
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))
          
f_chao
#Save a copy of the plot
ggsave("../figures/WRC18_Rhizo/pub/fungal_chao1.jpg", plot=f_chao, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE,bg="transparent")

ggsave("../figures/WRC18_Rhizo/pub/fungal_chao1.pdf", plot=f_chao, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE,bg="transparent")


```

```{r diversity metrics - calculate diversity and evenness}
# Calculate diversity metrics
fun.shannon <- vegan::diversity(fungal_otu, "shannon")
fun.simp <- vegan::diversity(fungal_otu, "simpson")
fun.invsimp <-vegan::diversity(fungal_otu, "invsimpson")
# Pielous evenness (uses presence/absence data)
fun.J <- fun.shannon/log(specnumber(fun.otu.2.PA[,-c(1:1)]))

#Adding diversity fields to dataset
meta$fun.shannon <- fun.shannon
meta$fun.J <- fun.simp
meta$fun.simp <- fun.invsimp
meta$fun.invsimp <- fun.J

```

```{r diversity metrics - display shannon }
#Run richness first. interaction, levels, and treatments for plotting set there
fun.shannon.lm <- aov(fun.shannon ~ treatment + source + i, data = meta)
plot(fun.shannon.lm)
anova(fun.shannon.lm)

#Tukey groups at p<=0.01*
t <- HSD.test(fun.shannon.lm, "i", group=TRUE, alpha=0.05)
t
plot(t)

#p is set in richness chuck
f_shan<- ggplot(meta, aes(x=source, y=fun.shannon, fill=treatment))  +
  geom_boxplot() + scale_fill_manual(values=c("gray", "darkgreen"), labels=c("Unfertilized","Fertilized"))  + 
  geom_point(aes(fill=treatment), size = 2, shape = 21, position = position_jitterdodge()) +
  #annotate("text",x=0.82,y=6.1, label="bc") +
#annotate("text",x=1.81,y=5.9, label="c") +
#annotate("text",x=2.81,y=6.05, label="c") +
#annotate("text",x=1.1815,y=6.5, label="a") +
#annotate("text",x=2.19,y=6.3, label="ab") + 
#annotate("text",x=3.19,y=6.35, label="ab") + 
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=14),
          axis.text=element_text(size=14, color="black"), #axis.text.x = element_blank(),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          #Set plot title textsize
          theme(plot.title=element_text(size=14)) +
          labs(x = "", y = "Shannon H' diversity", title="", fill="Treatment") +
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))

f_shan

ggsave("../figures/WRC18_Rhizo/pub/fungal_shannon.png", plot=f_shan, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE, bg="transparent")

ggsave("../figures/WRC18_Rhizo/pub/fungal_shannon.pdf", plot=f_shan, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE, bg="transparent")
```

```{r diversity metrics - display invsimp}
#Run richness first. interaction, levels, and treatments for plotting set there
fun.invsimp.lm <- aov(fun.invsimp ~ treatment + source + i, data = meta)
plot(fun.invsimp.lm)
anova(fun.invsimp.lm)

#Tukey groups at p<=0.01*
t <- HSD.test(fun.invsimp.lm, "i", group=TRUE, alpha=0.05)
t
plot(t)

#p is set in richness chuck
f_invsimp<- ggplot(meta, aes(x=source, y=fun.invsimp, fill=treatment))  +
  geom_boxplot() + scale_fill_manual(values=c("gray", "darkgreen"), labels="Unfertilized","Fertilized")  + #annotate("text",x=0.82,y=6.1, label="bc") +
#annotate("text",x=1.81,y=5.9, label="c") +
#annotate("text",x=2.81,y=6.05, label="c") +
#annotate("text",x=1.1815,y=6.5, label="a") +
#annotate("text",x=2.19,y=6.3, label="ab") + 
#annotate("text",x=3.19,y=6.35, label="ab") + 
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=24),
          axis.text=element_text(size=24, color="black"), #axis.text.x = element_blank(),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          #Set plot title textsize
          theme(plot.title=element_text(size=24)) +
          labs(x = "", y = "Invserve Simpson diversity", title="Fungal InvSimp diversity") +
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))

f_invsimp

ggsave("../figures/WRC18_Rhizo/pub/fungal_invsimp.jpg", plot=f_invsimp, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE, bg="transparent")

ggsave("../figures/WRC18_Rhizo/pub/fungal_invsimp.pdf", plot=f_invsimp, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE, bg="transparent")
```

```{r indicator species}
new.data <-cbind(meta,fungal_otu.rel)
#i<- with(new.data, interaction(treatment, source))
head(new.data)

new.data<- new.data %>% unite(st, c("source", "treatment"))

design.type <- new.data$st

#new.data$treatment <- droplevels(design.type)

dataREL.2013 <- new.data[,-c(1:33)]
head(dataREL.2013)
set.seed(42)
dataREL <- dataREL.2013[, colSums(dataREL.2013) > 0.01]
bac.ind <- indval(dataREL, clustering =  design.type, numitr=1000, set.seed(42))

design.type
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.02)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$OTU <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

#rownames(fungal_taxa_trim) <- fungal_taxa_trim$OTU_ID
otu.tax<-as.data.frame(fungal_taxa_trim)
#suffix<- 1:nrow(otu.tax)
#ps <- sprintf("Otu%05d",suffix)
#otu.tax$OTU_ID<-ps

#ind.tax <- otu.tax[which(as.character(otu.tax$OTU_ID) %in% bac.indicators$OTU), ]
#ind.tax <- ind.tax[match(ind.tax$OTU, bac.indicators$OTU), ]

bac.indicators <- bac.indicators %>% dplyr::rename(OTU_ID = OTU)
indicator.bac <- inner_join(bac.indicators, otu.tax)

#indicator.bac <- cbind(bac.indicators, ind.tax[, -c(1)])

indicator.bac <- indicator.bac[order(as.numeric(indicator.bac$Cluster)), ]

table(indicator.bac$Cluster)
table(indicator.bac$Phylum)
table(indicator.bac$Cluster)
levels(design.type)

# Export Bacteria Indicator Table
write.table(indicator.bac, "../figures/WRC18_Rhizo/pub/FungalIndicators_fun_st_0.02.txt",
            sep="\t", row.names = F, quote = F)
  
```

```{r fungal heat maps}

fungal <- cbind(meta[,c(4,7)],fungal_otu.rel)

fungal_heat <- as.data.frame(fungal) %>%
  gather(OTU_ID, abund, -treatment, -source)%>%
  filter(abund >= 0.01)

fungal_taxa_trim$OTU_ID <- as.character(fungal_taxa_trim$OTU_ID)

fungal_heat <- fungal_heat %>%  
inner_join(fungal_taxa_trim, by="OTU_ID") %>%
  group_by(treatment, source, class) %>%
  #filter(Family=="Planctomycetaceae")%>%
  dplyr::summarise(avgAbund=mean(abund))

ind.tax.i.heat <- ggplot(fungal_heat, aes(class, treatment)) +
  geom_tile(aes(fill = avgAbund)) + 
  scale_fill_gradientn(colors=c("khaki", "green", "blue", "black")) +
  #scale_y_continuous(sec.axis=sec_axis(~., breaks = 1:length(genus), labels=genus))+
  labs(fill = "Relative \n Abundance %", x="", y="")+
  theme(legend.position = "") +
  coord_flip() +
  facet_grid(~ source)
  #theme(panel.background=element_rect(fill="blue", colour="blue"))
ind.tax.i.heat


```
```{r 16s rRNA Taxa}
#Read in taxonomy file
tax <- import_mothur(mothur_constaxonomy_file =  taxfile)
tax.df <- as.data.frame(tax) %>%
  rownames_to_column("otu")
colnames(tax.df)=c("otu","domain","phylum","class", "order","family","genus")

#Clean up taxonomy file. Remvoe underscores and unclassified
tax.df <- tax.df %>%
  mutate(phylum = str_replace_all(phylum,"_", " "), order = str_replace_all(order,"_", " "),class = str_replace_all(class,"_", " "), family = str_replace_all(family,"_", " "), genus = str_replace_all(genus,"_", " ")) %>%
  mutate(phylum = str_replace_all(phylum,"unclassified", " "), order = str_replace_all(order,"unclassified", " "),class = str_replace_all(class,"unclassified", " "), family = str_replace_all(family,"unclassified", " "), genus = str_replace_all(genus,"unclassified", " "))
```

```{r read in 16S rRNA, include=FALSE}
#Read in OTU file
otu <- read.otu(sharedfile)
#Remove OTUs with less than 2 occurences across all sites

b <- ncol(otu)
reads <- sum(colSums(otu))
#reads before filtering 

otu.2 <- otu[, which(colSums(otu) > 10)]

#Remove all RP samples
#In the original WRC2015 sampling I collected bulk soils, and rhizospehre soils. Soil loosley adhering to roots was shaken into a bag (RP) for DNA and soil NH4/NO3 extraction. Roots and adhering soils collected for DNA processing (RS). Removed RP from DNA analysis.
#WRC15_65-WRC15_80
rownames(otu.2)
otu.2 <-otu.2[-c(33:48),]
rownames(otu.2)

#FROM OTU - remove blocks 1, 3, 5, 7 and treatments C & F
#First join meta and otu. Filter and then remove extra columns
meta_all <- read.csv(metafile)
#meta_all <- meta_all[-c(33:48),]

otu.2.filter <- cbind(meta_all, otu.2)

otu.2.filter <- otu.2.filter %>%
  filter(year==2018) %>%
  filter(treatment=="M" | treatment =="MF") %>%
  filter(block==2 | block==4 | block==6 | block==8)

rownames(otu.2.filter)<- otu.2.filter$sample

otu.2 <- otu.2.filter[,-c(1:26)]
rownames(otu.2.filter)


#What sample has lowest read count? WRC18_201 16668, next lowest is WRC18_200 43192 highest WRC15_167 68819
otu2<- colSums(t(otu.2))
otu2
otu2[which.min(colSums(t(otu.2)))]
otu2[which.max(colSums(t(otu.2)))]
#Did not remove lowest sample, when ran adonis did not change results
#rownames(otu.2)
#otu.2 <- otu.2[-c(17),]
```

```{r sequence read counts by sample}
#graph of read counts, need to work on y axis
c<-as.data.frame(otu.2)
c$colsum <- colSums(t(c))
c$group <- rownames(c)
p<-ggplot(c, aes(x=group,y=colsum))+geom_bar(stat="identity")+coord_flip()
p

#Rarefy
#Set min sample number
min.N <- min(rowSums(otu.2))
#rrarefy returns a randomly rarefied community data frame or vector of selected size
#This is the otu set that should be used for beta diversity metrics
otu.rare <- rrarefy(otu.2, min.N)

```

```{r rarefaction curve}
#rarefy returns a species richness estimate in a random subsample of selected size
otu.rare.rich <- rarefy(x = otu.2, sample=min.N, se=T)
#Plot rarefaction curve
###I am not sure how to interpret this. I think it suggests that I did not capture all of the diversity? 
rarecurve(x = otu.2, step = 20, col="blue", cex=0.6, las=1)
abline(0,1,col='red')
text(1000,1000,"1:1",pos=2, col='red')
```

```{r Relative abundance and PERMANOVA}
#transforms read counts to relative abundance
#Using rarefied OTU dataset
otu.2.rel <- otu.rare
for(i in 1:dim(otu.rare)[1]){
  otu.2.rel[i,] <- otu.rare[i,]/sum(otu.rare[i,])
}

#Combine experimental deisng meta data with OTU relative abundance data
metaotu <- cbind(meta, otu.2.rel)
#Where does otu data begin, remove for PERMANOVA
#metaotu[1,20:21]

#PERMANOVA with source and treatment as factors
#Among all treatments source (soil source) and treatment
metaotu.ad <- adonis(metaotu[,-c(1:33)] ~source*treatment, method = "bray", data = metaotu, perm=1000, set.seed=42)

metaotu.ad 

```

```{r Calculate distance matrix and PCoA components - Taxa/Otus}

otu.2.rel.dist<-vegdist(otu.2.rel, method="bray")

# Principal Coordinates Analysis
otu.2.rel.dist.pcoa <- cmdscale(otu.2.rel.dist, k=3, eig=TRUE, add=FALSE)
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1a <- round(otu.2.rel.dist.pcoa$eig[1] / sum(otu.2.rel.dist.pcoa$eig), 3) * 100
explainvar2a <- round(otu.2.rel.dist.pcoa$eig[2] / sum(otu.2.rel.dist.pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)

explainvar1a #29.6
explainvar2a #12.5


```



```{r PCoA Plot based on Taxa/Otus}
rownames(meta)<-meta$sample
all.equal(rownames(meta), rownames(otu.2.rel))
# Set treatments
treat1 <- as.factor(meta$source)
levels(treat1) <- ordered(c("bulk","grass", "forb"))
treat2 <- as.factor(meta$treatment)
#levels(treat2) <- c("M", "MF")

pcoa.groups <- paste(meta$source, meta$treatment, sep = "_")
pcoa.points <- data.frame(otu.2.rel.dist.pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # treatment
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # source

levels(pcoa.col) <- c("Unfertilized","Fertilized")

#Run envfit to see which soil varibales are predicators 
#These are just the environmental parameters have values for bulk and rhizospehre soils
fit<-envfit(otu.2.rel.dist.pcoa$points, meta[,-c(1:14, 17,20,22,24:25,27:35)], perm=1000, na.rm=T, set.seed=42)
fit

A <-as.list(fit$vectors)
vec<-as.data.frame(fit$vectors$arrows*sqrt(fit$vectors$r)*0.35)
p<- as.data.frame(A$pvals)
vec <- cbind(vec, p) 
vec <-  subset(vec, A$pvals<=0.05)

vec$parm<-rownames(vec)
colnames(vec)
vec$parm<-c("N%", "C:N", "NH4", "PO4_HCl", "PO4_W", "OM%")

#plot PCoA
df1a <- as.data.frame(pcoa.cent.dataframe)
plot1 <- ggplot(df1a, aes(x=V1, y=V2), group = interaction(pcoa.col, pcoa.shape)) + 
  theme_bw() +
#group = interaction(pcoa.col, pcoa.shape), colour=pcoa.col2))
  geom_point(aes(shape = pcoa.shape, colour = pcoa.col), size=10, stroke = 2) + 
  #geom_point(aes(colour=pcoa.col2), size =4.0)+
  #removes gridlines from plot
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black"))+ 
  #Plot soil property vectors from envfit
  geom_segment(data=vec, aes(x=0,xend=Dim1,y=0,yend=Dim2), size=1, 
     arrow = arrow(length = unit(0.2, "cm")),colour="black",inherit.aes=F)+ 
  #Label soil vectors. Using annotate because I can't figure out how to move segment text
  #Note when re-run should double check envfit stats to make sure that soil parameters are the same
  #geom_text(data=vec,aes(x=Dim1,y=Dim2,label=vec$parm),size=5, inherit.aes=F)+
  annotate("text",x=0.12,y=0.26, label="OM%", size=4) +
  annotate("text",x=0.15,y=0.17, label="N%", size=4) +
  annotate("text",x=0.21,y=0.18, label="PO4-W", size=4) +
  annotate("text",x=0.28,y=0.16, label="PO4-HCl", size=4) +
  annotate("text",x=-0.13,y=-0.23, label="C:N", size=4) +
  annotate("text",x=0.036,y=-0.22, label="NH4", size=4) +
  #Set error bacrs for geom_point+ 
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0), colour="black") +
  #Set colors for treatments
  #scale_colour_manual(values = c("#999999", "#000000")) +
  scale_colour_manual(values = c("gray", "darkgreen","purple","red","yellow","blue","gray","pink","magenta")) +
  #Would have liked to set shapes but couldn't get this to work
  scale_shape_manual(values = c(16,22,15)) +
  #Sets map coordinates
  coord_cartesian(xlim = c(-0.2, 0.3), ylim = c(-0.3, 0.3)) +
  #Sets axis text and put border around plot
  theme(axis.title = element_text(size=14),# face="bold"), 
        axis.text.x = element_text(size=14, color="black"),  axis.text.y = element_text(size=14, color="black"),
        panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=14)) +
  #Set legend text size
  theme(legend.text=element_text(size=14, face="bold"), legend.title = element_text(size=14, face="bold"))+
  #Sets size of tick marks on axis
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  #Sets labels for plot title, axis titles, and legend headings
  xlab("PCoA 1 (29.7%)") + ylab("PCoA 2 (12.5%)") + 
  labs(shape = "Soil Source") +
  labs(colour="Treatment") +
  ggtitle("16S rRNA V4 Taxonomic Composition")+
  theme(legend.position="right")+
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))
  
plot1 

#Save a copy of the plot
ggsave("../figures/WRC18_Rhizo/pub/16SrRNA_WRC2018Rhizo.jpg", plot=plot1, device=NULL, path=NULL, scale=1, width=7.5, height=6, dpi=900, limitsize=TRUE, bg="transparent")

ggsave("../figures/WRC18_Rhizo/pub/16SrRNA_WRC2018Rhizo.pdf", plot=plot1, device=NULL, path=NULL, scale=1, width=7.5, height=6, dpi=900, limitsize=TRUE, bg="transparent")

```

```{r procrustes analysis fungal nad bacterial}
#Do fungal and bacterial communities exhibit the same pattern?
pro.1 <- procrustes(otu.2.rel.dist.pcoa, fungal_otu.rel.dist.pcoa, symmetric=TRUE, scores = "sites", choices = c(1, 2))
pro.1

pro.1 <- protest(X = fungal_otu.rel.dist.pcoa, Y =otu.2.rel.dist.pcoa , symmetric=TRUE, scores = "sites", permutations = 999) 
pro.1

#Sum of squates = 0.44
#Correlation rotation = 0.75
#Signifignce = 0.001
#There is a significant correlation between bacterial and fungal communities

plot(pro.1, kind=1, type='text')
plot(pro.1, kind=2, type='text')
```
```{r mantel}
bac.fun.mantel <- vegan::mantel(as.matrix(fungal_otu.rel.dist), as.matrix(otu.2.rel.dist), method="pearson", permutations=999)
bac.fun.mantel

```
```{r bac heat map}
bac <- cbind(meta[,c(4,7)],otu.2.rel)

bac_heat <- as.data.frame(bac) %>%
  gather(otu, abund, -treatment, -source)%>%
  filter(abund >= 0.01)

#tax.df$otu <- as.character(tax.df$otu)

bac_heat <- bac_heat %>%  
inner_join(tax.df, by="otu") %>%
  group_by(treatment, source, family) %>%
  #filter(Family=="Planctomycetaceae")%>%
  dplyr::summarise(avgAbund=mean(abund))

ind.tax.i.heat <- ggplot(bac_heat, aes(family, treatment)) +
  geom_tile(aes(fill = avgAbund)) + 
  scale_fill_gradientn(colors=c("khaki", "green", "blue", "black")) +
  #scale_y_continuous(sec.axis=sec_axis(~., breaks = 1:length(genus), labels=genus))+
  labs(fill = "Relative \n Abundance %", x="", y="")+
  theme(legend.position = "") +
  coord_flip() +
  facet_grid(~ source)
  #theme(panel.background=element_rect(fill="blue", colour="blue"))
ind.tax.i.heat

```
```{r network analysis}
require(igraph)  
require(Hmisc)  
require(Matrix)   
require(tidyr)
fungal_otu <- as.data.frame(fungal_otu)
#Select top 50 OTUs
fungal_otu_50 <- fungal_otu %>% dplyr::summarise_all(dplyr::funs(sum)) 
fungal_otu_50 <- as.data.frame(t(fungal_otu_50))
fungal_otu_50 <- fungal_otu_50 %>%
  tibble::rownames_to_column() %>%
  dplyr::arrange(desc(V1)) %>%
  dplyr::top_n(100)
rownames(fungal_otu_50) <- fungal_otu_50$rowname

fungal_otu_top_50 <- fungal_otu[,rownames(fungal_otu_50),drop=FALSE]
#Use count data
#Fungal first
#############
#require("SpiecEasi")
#otu.cor <- sparcc(fungal_otu.rel, iter = 20, inner_iter = 10, th = 0.1)
otu.cor <-rcorr(as.matrix(fungal_otu_top_50), type="pearson")
otu.pval <- forceSymmetric(otu.cor$P)
rownames(fungal_taxa_trim) = fungal_taxa_trim$OTU_ID
sel.tax <- fungal_taxa_trim[rownames(otu.pval),,drop=FALSE]
all.equal(rownames(sel.tax), rownames(otu.pval))
p.yes <- otu.pval<0.05
r.val = otu.cor$r # select all the correlation values 
p.yes.r <- r.val*p.yes # only select correlation values based on p-value criterion 
p.yes.r <- abs(p.yes.r)>0.75 # output is logical vector
p.yes.rr <- p.yes.r*r.val # use logical vector for subscripting.
adjm <- as.matrix(p.yes.rr)
colnames(adjm) <- as.vector(sel.tax$Family)
rownames(adjm) <- as.vector(sel.tax$Family)



net.grph=graph.adjacency(adjm,mode="undirected",weighted=TRUE,diag=FALSE)
#edgew<-E(net.grph)$weight
#bad.vs<-V(net.grph)[degree(net.grph) == 0] 
#net.grph <-delete.vertices(net.grph, bad.vs)


V(net.grph)$vertex_degree <- degree(net.grph)

plot(net.grph,
    layout=layout_with_dh(net.grph, weight.edge.lengths = edge_density(net.grph)/10000,    weight.node.dist = 10000, weight.node.edge.dist = 10000 ),
    vertex.size=V(net.grph)$vertex_degree * 2,
    vertex.frame.color="black",
    edge.curved=F,
    edge.width=2,
    vertex.label.color="black",
    vertex.label.family="Times New Roman",
    vertex.label.font=5,
    vertex.label = NA)



#Number of significant correlations in the top 100 fungal taxa
flattenCorrMatrix <- function(cormat, pmat) {
  ut <- upper.tri(cormat)
  data.frame(
    row = rownames(cormat)[row(cormat)[ut]],
    column = rownames(cormat)[col(cormat)[ut]],
    cor  =(cormat)[ut],
    p = pmat[ut]
    )
}

f <- flattenCorrMatrix(otu.cor$r, otu.cor$P)
f_p_row <- f %>%
  dplyr::filter(p<0.05) %>%
  dplyr::select(row, column) %>%
  dplyr::count(row)

f_p_col <- f %>%
  dplyr::filter(p<0.05) %>%
  dplyr::select(row, column) %>%
  dplyr::count(column)

f_p <- f %>%
  dplyr::filter(p<0.05) 
  
f_p_row <- f_p %>%
  dplyr::select(row)

f_p_col <- f_p %>%
  dplyr::select(column) %>%
  dplyr::rename(row=column)

f_p_rowcol <- dplyr::bind_rows(f_p_row, f_p_col) %>%
  dplyr::rename(OTU_ID=row)
  
f_count <- f_p_rowcol %>%
  dplyr::count(OTU_ID) %>%
  dplyr::arrange(desc(n)) %>%
  dplyr::left_join(sel.tax)

```


```{r indicator species}
new.data <-cbind(meta,otu.2.rel)
#i<- with(new.data, interaction(treatment, source))
head(new.data)

new.data<- new.data %>% unite(st, c("source", "treatment"))

design.type <- new.data$st

levels(design.type) 
dataREL.2013 <- new.data[,-c(1:33)]
head(dataREL.2013)
set.seed(42)
dataREL <- dataREL.2013[, colSums(dataREL.2013) > 0.01]
bac.ind <- indval(dataREL, clustering =  design.type, numitr=1000, set.seed(42))

design.type
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.02)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$OTU <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

otu.tax<-as.data.frame(tax.df)
suffix<- 1:nrow(otu.tax)
ps <- sprintf("Otu%05d",suffix)
otu.tax$OTU<-ps

#ind.tax <- otu.tax[which(as.character(otu.tax$OTU) %in% bac.indicators$OTU), ]
#ind.tax <- ind.tax[match(ind.tax$OTU, bac.indicators$OTU), ]

#indicator.bac <- cbind(bac.indicators, ind.tax[, -c(1)])

indicator.bac <- inner_join(bac.indicators, otu.tax)

indicator.bac <- indicator.bac[order(as.numeric(indicator.bac$Cluster)), ]

table(indicator.bac$Cluster)
table(indicator.bac$Phylum)
table(indicator.bac$Cluster)
levels(design.type)

# Export Bacteria Indicator Table
write.table(indicator.bac, "../figures/WRC18_Rhizo/pub/BacterialIndicators_bac_st_0.02.txt",
            sep="\t", row.names = F, quote = F)

#From what I can tell by comparing both methods of indicator analysis the second "cluster" 1, 2, 3, ... is the design.type in alphabetical order. So 1 = avM, 2 = avMF, 3 = bulkM, 4 = bulkMF, 5 = ecM, 6=ecMF
  
```

```{r Bac Presence_Absence and Species richness, include =FALSE }
#Transform data to reflect only presence or absence ie 0 or 1
otu.2.PA<-(otu.rare>0*1)
#sanity check 
#otu.2.PA[,1:5]
#Add meta data
metaotu.2 <- cbind(meta, otu.2.PA)
#metaotu.2[1:15]
#Transform True/False to 0/1 
cols <- sapply(metaotu.2, is.logical)
metaotu.2[,cols]<-lapply(metaotu.2[,cols],as.numeric)
#head(metaotu.2)
```

```{r bac diversity metrics: richness}
#chao1 richness index
chao1 <- estimateR(otu.2)
c <- as.data.frame(chao1)
ct <- as.data.frame(t(c))

meta$bac.s.obs <- ct$S.obs
meta$bac.s.chao1 <- ct$S.chao1
meta$bac.se.chao1 <- ct$se.chao1

#interaction variable so can use Tukey
i <- with(meta, interaction(treatment, source))

richness.lm <- aov(bac.s.chao1 ~ treatment + source + i, data = meta)
plot(richness.lm)
anova(richness.lm)

#Get Tukey post hoc groups p<=0.01*
t <- HSD.test(richness.lm, "treatment", group=TRUE, alpha=.05)
t
plot(t)

#Plot
b_chao<- ggplot(meta, aes(x=source, y=bac.s.chao1, fill=treatment))  +
  geom_boxplot(binaxis='y', stackdir='center') + 
  scale_fill_manual(values=c("gray", "darkgreen"), labels=c("Unfertilized","Fertilized")) +
    geom_point(aes(fill=treatment), size = 2, shape = 21, position = position_jitterdodge()) +
#Set labels for Tukey groups
annotate("text",x=1.30,y=6300, label="*", size=7) +
annotate("text",x=2.20,y=5310, label="*", size=7) +
annotate("text",x=3.20,y=5510, label="*", size=7) +
#annotate("text",x=1.1815,y=7050, label="a") +
#annotate("text",x=2.19,y=6600, label="a") + 
#annotate("text",x=3.19,y=6250, label="a") + 
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=14),
          axis.text=element_text(size=14, color="black"), axis.text.x = element_blank(),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          #Set plot title textsize
          theme(plot.title=element_text(size=14)) +
         labs(x = "", y = "", title="Bacterial", fill="Treatment") +
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))
          
b_chao
#Save a copy of the plot
ggsave("../figures/WRC18_Rhizo/pub/chao1.jpg", plot=b_chao, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE,bg="transparent")

ggsave("../figures/WRC18_Rhizo/pub/chao1.pdf", plot=b_chao, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE,bg="transparent")


```

```{r diversity metrics - calculate diversity and evenness}
# Calculate diversity metrics
bac.shannon <- vegan::diversity(otu.rare, "shannon")
bac.simp <- vegan::diversity(otu.rare, "simpson")
bac.invsimp <-vegan::diversity(otu.rare, "invsimpson")
# Pielous evenness (uses presence/absence data)
bac.J <- bac.shannon/log(specnumber(otu.2.PA[,-c(1:1)]))

#Adding diversity fields to dataset
meta$bac.shannon <- bac.shannon
meta$bac.J <- bac.J
meta$bac.simp <- bac.simp
meta$bac.invsimp <- bac.invsimp

```

```{r diversity metrics - display shannon }
#Run richness first. interaction, levels, and treatments for plotting set there
shannon.lm <- aov(bac.shannon ~ treatment + source + i, data = meta)
plot(shannon.lm)
anova(shannon.lm)

#Tukey groups at p<=0.01*
t <- HSD.test(shannon.lm, "treatment", group=TRUE, alpha=0.05)
t
plot(t)

#p is set in richness chuck
b_shan<- ggplot(meta, aes(x=source, y=bac.shannon, fill=treatment))  +
  geom_boxplot() + 
  scale_fill_manual(values=c("gray", "darkgreen"), labels=c("Unfertilized","Fertilized"))  +
    geom_point(aes(fill=treatment), size = 2, shape = 21, position = position_jitterdodge()) +
  #annotate("text",x=0.82,y=6.1, label="bc") +
annotate("text",x=1.20,y=6.60, label="**", size=7) +
annotate("text",x=2.20,y=6.60, label="**", size=7) +
annotate("text",x=3.2,y=6.60, label="**", size=7) +
#annotate("text",x=2.19,y=6.3, label="ab") + 
#annotate("text",x=3.19,y=6.35, label="ab") + 
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=14),
          axis.text=element_text(size=14, color="black"), #axis.text.x = element_blank(),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          #Set plot title textsize
          theme(plot.title=element_text(size=14)) +
          labs(x = "", y = "'", title="", fill="Treatment") +
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))

b_shan

ggsave("../figures/WRC18_Rhizo/pub/shannon.png", plot=b_shan, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE, bg="transparent")

ggsave("../figures/WRC18_Rhizo/pub/shannon.pdf", plot=b_shan, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE, bg="transparent")
```

```{r diversity metrics - Pielous evennes J}
J.lm <- aov(bac.J ~ treatment + source + i , data = meta)
plot(J.lm)
anova(J.lm)
#TukeyHSD(J.lm)
t <- HSD.test(J.lm, "i", group=TRUE, alpha=0.05)
t
plot(t)

even<- ggplot(metav, aes(x=i, y=bac.J, fill=treatment))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000")) +
#Set Tukey groups
#annotate("text",x=0.82,y=0.77, label="ab") +
#annotate("text",x=1.81,y=0.75, label="b") +
#annotate("text",x=2.81,y=0.76, label="b") +
#annotate("text",x=1.1815,y=0.80, label="a") +
#annotate("text",x=2.19,y=0.78, label="a")  + 
#annotate("text",x=3.19,y=0.80, label="a")  + 
#Set plot properties
theme_bw()  +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)), 
          axis.title.x=element_text(margin=margin(t=10))) +
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(size=14), panel.border = 
          element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Soil Source", y = "Pileou's eveness") + 
    theme(legend.text=element_text(size=14), legend.title = element_text(size=14), 
          legend.position = "bottom")

even 

ggsave("../figures/WRC18_Rhizo/even.jpg", plot=even, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

```

```{r combine diversity plots}

g <- ggarrange(f_chao, b_chao,f_shan, b_shan, ncol = 2, nrow = 2, heights=c(5,5,5,5), common.legend = TRUE, legend = "bottom", labels = c("A", "B", "C", "D"), label.x=0.1)

g

ggsave("../figures/WRC18_Rhizo/pub/all_div.pdf", plot=g, device=NULL, path=NULL, scale=1, width=6, height=6, dpi=300, limitsize=TRUE)

ggsave("../figures/WRC18_Rhizo/pub/all_div.jpg", plot=g, device=NULL, path=NULL, scale=1, width=6, height=6, dpi=300, limitsize=TRUE)
```

```{r environmental parameters}

#SUMMARY TABLES
meta.soilonly <- meta %>%
  dplyr::select(source, treatment, temp3_C, moist_percent, pH, c_percent, n_percent, C_N, n_nh4_ug_gsoil, p_po4_hcl_ug_gsoil, p.po4_w_ug_gsoil, percent_om)%>%
  mutate(N_P = n_nh4_ug_gsoil/p.po4_w_ug_gsoil)

by_plant_treatment <- meta.soilonly %>%
  group_by(source, treatment) %>%
  summarise_all(funs(mean,sd), na.rm=T)

write.csv(by_plant_treatment, file = "by_plant_treatment.csv",row.names=TRUE)


by_treatment <- meta.soilonly %>%
  dplyr::select(-source)%>%
  group_by(treatment) %>%
  summarise_all(funs(mean,sd), na.rm=T)

by_source <- meta.soilonly %>%
  dplyr::select(-treatment)%>%
  group_by(source) %>%
  summarise_all(funs(mean,sd), na.rm=T)


#KCl extractable Ammonium and Nitrate
nh4 <- aov(n_nh4_ug_gsoil ~ treatment * source, meta.soilonly)
plot(nh4)
anova(nh4)

#nh4<- ggplot(meta.env, aes(x=source, y=n_nh4_ug_gsoil, fill=treatment))  +
#  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333", "444444"))+
#  facet_grid(meta.env$ditch)
#nh4

#Bulk only - pH, temp, moist_percent
pH <- aov(pH ~ treatment, meta.soilonly)
plot(pH)
anova(pH)

avg_temp_C <- aov(temp3_C ~ treatment, data=meta.soilonly) 
plot(avg_temp_C)
anova(avg_temp_C)

moist_percent <- aov(log(moist_percent) ~ treatment , data=meta.soilonly) 
plot(moist_percent)
anova(moist_percent)

#treatment and source
c_percent <- aov(log(c_percent) ~ treatment * source, data=meta.soilonly) 
plot(c_percent)
anova(c_percent)

n_percent <- aov(log(n_percent) ~ treatment * source, data=meta.soilonly) 
plot(n_percent)
anova(n_percent)

C_N <- aov(log(C_N) ~ treatment * source, data=meta.soilonly) 
plot(C_N)
anova(C_N)

nh4 <- aov(n_nh4_ug_gsoil ~ treatment * source, data=meta.soilonly) 
plot(nh4)
anova(nh4)

po4_hcl <- aov(p_po4_hcl_ug_gsoil ~ treatment * source, data=meta.soilonly) 
plot(po4_hcl)
anova(po4_hcl)

po4_w <- aov(p.po4_w_ug_gsoil ~ treatment * source, data=meta.soilonly) 
plot(po4_w)
anova(po4_w)

om <- aov(log(percent_om) ~ treatment * source, data=meta.soilonly) 
plot(om)
anova(om)

# NP <- aov(log(N_P) ~ treatment * source, data=meta.soilonly) 
plot(NP)
anova(NP)
```

```{r tea decomp soil properties}
#SUMMARY TABLES

meta <- read.csv("../data/WRC18_Rhizo/2015_2018_WestResearchCampus_Sampling_Summary_ditch.csv", header=TRUE)
meta <- meta %>%
  filter(treatment=='M'|treatment=='MF')


meta.soilonly <- meta %>%
  dplyr::select(block, ditch, treatment, temp3_C, moist_percent, pH, c_percent, n_percent, C_N, n_nh4_ug_gsoil, p_po4_hcl_ug_gsoil, p.po4_w_ug_gsoil, percent_om, n_no3_ug_gsoil )%>%
  mutate(N_P = n_nh4_ug_gsoil/p.po4_w_ug_gsoil)

by_ditch_treatment <- meta.soilonly %>%
  group_by(treatment, ditch) %>%
  summarise_all(funs(mean,sd), na.rm=T)

#write.csv(by_plant_treatment, file = "by_ditch_treatment.csv",row.names=TRUE)


by_treatment <- meta.soilonly %>%
  dplyr::select(-ditch)%>%
  group_by(treatment) %>%
  summarise_all(funs(mean,sd), na.rm=T)

#write.csv(by_treatment, file = "by_treatment.csv",row.names=TRUE)

by_ditch <- meta.soilonly %>%
  dplyr::select(-treatment)%>%
  group_by(ditch) %>%
  summarise_all(funs(mean,sd), na.rm=T)

#write.csv(by_ditch, file = "by_ditch.csv",row.names=TRUE)

avg_temp_C <- aov(temp3_C ~ treatment * ditch + (1|block), data=meta.soilonly) 
anova(avg_temp_C)

moist_percent <- aov(log(moist_percent) ~ treatment * ditch + (1|block), data=meta.soilonly) 
anova(moist_percent)

pH <- aov(pH ~ treatment * ditch + (1|block), meta.soilonly) 
anova(pH)

c_percent <- aov(log(c_percent) ~ treatment  * ditch + (1|block), data=meta.soilonly) 
anova(c_percent)

n_percent <- aov(log(n_percent) ~ treatment  * ditch + (1|block), data=meta.soilonly) 
anova(n_percent)

C_N <- aov(log(C_N) ~ treatment  * ditch + (1|block), data=meta.soilonly) 
anova(C_N)

nh4 <- aov(n_nh4_ug_gsoil ~ treatment * ditch + (1|block), data=meta.soilonly) 
anova(nh4)

no3 <- aov(n_no3_ug_gsoil ~ treatment * ditch + (1|block), data=meta.soilonly) 
anova(no3)

po4_hcl <- aov(p_po4_hcl_ug_gsoil ~ treatment  * ditch + (1|block), data=meta.soilonly) 
anova(po4_hcl)

po4_w <- aov(p.po4_w_ug_gsoil ~ treatment * ditch + (1|block), data=meta.soilonly) 
anova(po4_w)




```
```{r soil c percent}
c_percent.lm <- aov(log(c_percent) ~ treatment * source, data=meta) 
plot(c_percent.lm)
anova(c_percent.lm)

t <- HSD.test(c_percent.lm, "source", group=TRUE, alpha=0.05)
t
plot(t)

cpercent<- ggplot(meta, aes(x=source, y=c_percent, fill=treatment))  +
  geom_boxplot() + 
  scale_fill_manual(values=c("gray", "darkgreen"))  +
   geom_point(aes(fill=treatment), size = 2, shape = 21, position = position_jitterdodge()) +
  #annotate("text",x=0.82,y=6.1, label="bc") +
#annotate("text",x=1.20,y=6.50, label="*", size=5) +
#annotate("text",x=2.20,y=6.65, label="*", size=5) +
#annotate("text",x=3.2,y=6.50, label="*", size=5) +
#annotate("text",x=2.19,y=6.3, label="ab") + 
#annotate("text",x=3.19,y=6.35, label="ab") + 
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=14),
          axis.text=element_text(size=14, color="black"), #axis.text.x = element_blank(),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          #Set plot title textsize
          theme(plot.title=element_text(size=14)) +
          labs(x = "", y = "Soil C%", title="") +
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))

cpercent

ggsave("../figures/WRC18_Rhizo/pub/cpercent.jpg", plot=cpercent, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE, bg="transparent")

ggsave("../figures/WRC18_Rhizo/pub/cpercent.pdf", plot=cpercent, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE, bg="transparent")

```

```{r mantel comparisons}

soil.dist <- vegdist(x = scale(meta.soilonly[, -c(1:5,13)]), na.rm=TRUE, "euclid")

soil.fun.mantel <- vegan::mantel(as.matrix(soil.dist), as.matrix(fungal_otu.rel.dist), method="pearson", permutations=999)
soil.fun.mantel

soil.bac.mantel <- vegan::mantel(as.matrix(soil.dist), as.matrix(otu.2.rel.dist), method="pearson", permutations=999)
soil.bac.mantel

```

```{r soil pcoa}
metaotu.ad <- adonis(meta.soilonly[,-c(1:5,13)] ~source*treatment, method = "euclid", data = metaotu, perm=1000, set.seed=42, na.rm = TRUE)

metaotu.ad 

# Principal Coordinates Analysis
soil.dist.pcoa <- cmdscale(soil.dist, k=3, eig=TRUE, add=FALSE)
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1a <- round(otu.2.rel.dist.pcoa$eig[1] / sum(soil.dist.pcoa$eig), 3) * 100
explainvar2a <- round(otu.2.rel.dist.pcoa$eig[2] / sum(soil.dist.pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)

explainvar1a #29.6
explainvar2a #12.5

rownames(meta)<-meta$sample
#all.equal(rownames(meta), rownames(otu.2.rel))
# Set treatments
treat1 <- as.factor(meta$source)
levels(treat1) <- ordered(c("bulk","grass", "forb"))
treat2 <- as.factor(meta$treatment)
#levels(treat2) <- c("M", "MF")

pcoa.groups <- paste(meta$source, meta$treatment, sep = "_")
pcoa.points <- data.frame(soil.dist.pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # treatment
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # source

levels(pcoa.col) <- c("Unfertilized","Fertilized")

#Run envfit to see which soil varibales are predicators 
#These are just the environmental parameters have values for bulk and rhizospehre soils
fit<-envfit(otu.2.rel.dist.pcoa$points, meta.soilonly[,-c(1:5,13)], perm=1000, na.rm=T, set.seed=42)
fit

A <-as.list(fit$vectors)
vec<-as.data.frame(fit$vectors$arrows*sqrt(fit$vectors$r)*2)
p<- as.data.frame(A$pvals)
vec <- cbind(vec, p) 
vec <-  subset(vec, A$pvals<=0.05)

vec$parm<-rownames(vec)
colnames(vec)
#vec$parm<-c("N%", "C:N", "NH4", "PO4_HCl", "PO4_W", "OM%")


#plot PCoA
df1a <- as.data.frame(pcoa.cent.dataframe)
plot1 <- ggplot(df1a, aes(x=V1, y=V2), group = interaction(pcoa.col, pcoa.shape)) + 
  theme_bw() +
#group = interaction(pcoa.col, pcoa.shape), colour=pcoa.col2))
  geom_point(aes(shape = pcoa.shape, colour = pcoa.col), size=10, stroke = 2) + 
  #geom_point(aes(colour=pcoa.col2), size =4.0)+
  #removes gridlines from plot
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black"))+ 
  #Plot soil property vectors from envfit
 # #geom_segment(data=vec, aes(x=0,xend=Dim1,y=0,yend=Dim2), size=1, 
 # #   arrow = arrow(length = unit(0.2, "cm")),colour="black",inherit.aes=F)+ 
  #Label soil vectors. Using annotate because I can't figure out how to move segment text
  #Note when re-run should double check envfit stats to make sure that soil parameters are the same
  ##geom_text(data=vec,aes(x=Dim1,y=Dim2,label=vec$parm),size=5, inherit.aes=F)+
  #annotate("text",x=0.12,y=0.26, label="OM%", size=4) +
  #annotate("text",x=0.15,y=0.17, label="N%", size=4) +
  #annotate("text",x=0.21,y=0.18, label="PO4-W", size=4) +
  #annotate("text",x=0.28,y=0.16, label="PO4-HCl", size=4) +
  #annotate("text",x=-0.13,y=-0.23, label="C:N", size=4) +
  #annotate("text",x=0.036,y=-0.22, label="NH4", size=4) +
  #Set error bacrs for geom_point+ 
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0), colour="black") +
  #Set colors for treatments
  #scale_colour_manual(values = c("#999999", "#000000")) +
  scale_colour_manual(values = c("gray", "darkgreen","purple","red","yellow","blue","gray","pink","magenta")) +
  #Would have liked to set shapes but couldn't get this to work
  scale_shape_manual(values = c(16,22,15)) +
  #Sets map coordinates
  #coord_cartesian(xlim = c(-0.2, 0.3), ylim = c(-0.3, 0.3)) +
  #Sets axis text and put border around plot
  theme(axis.title = element_text(size=14),# face="bold"), 
        axis.text.x = element_text(size=14, color="black"),  axis.text.y = element_text(size=14, color="black"),
        panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=14)) +
  #Set legend text size
  theme(legend.text=element_text(size=14, face="bold"), legend.title = element_text(size=14, face="bold"))+
  #Sets size of tick marks on axis
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  #Sets labels for plot title, axis titles, and legend headings
  xlab("PCoA 1 (40.0%)") + ylab("PCoA 2 (10.0%)") + 
  labs(shape = "Soil Source") +
  labs(colour="Treatment") +
  ggtitle("Soil Properties")+
  theme(legend.position="right")+
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))
  
plot1 

#Save a copy of the plot
ggsave("../figures/WRC18_Rhizo/pub/soil_pcoa.jpg", plot=plot1, device=NULL, path=NULL, scale=1, width=7.5, height=6, dpi=900, limitsize=TRUE, bg="transparent")

ggsave("../figures/WRC18_Rhizo/pub/soil_pcoa.pdf", plot=plot1, device=NULL, path=NULL, scale=1, width=7.5, height=6, dpi=900, limitsize=TRUE, bg="transparent")

```


```{r Phylogenetic Diversity - load data, include=FALSE}
# Phylogenetic Diversity - analysis by Mario Muscarella modified by Gina Bledsoe
# Load Tree
tree <- read.tree("../data/WRC18_Rhizo/WRC15-18Rhizo.bac.rename.tree")
tree$tip.label <- gsub("\\|", "", tree$tip.label)
sum(tree$tip.label %in% colnames(otu.2) == FALSE)
# Small Branches
sum(tree$edge.length < 0.0000001)
# Import Unifrac Matrix
unifrac.raw <- read.delim("../data/WRC18_Rhizo/WRC15-18Rhizo.bac.tree1.weighted.phylip.dist", skip = 1, header = F, row.names = 1)
colnames(unifrac.raw) <- rownames(unifrac.raw)
rownames(unifrac.raw)
#Remove spaces from rownames
require("stringr")
rownames(unifrac.raw) <- str_replace_all(rownames(unifrac.raw), fixed(" "), "")
rownames(unifrac.raw)
#narrow unifrac to 2018 Rhizopshere samples
unifrac <- unifrac.raw[which(row.names(unifrac.raw) %in%
                                   row.names(otu.2)),
                            which(row.names(unifrac.raw) %in%
                                   row.names(otu.2))]
dim(unifrac)
# Make into Distance Matrix
unifrac.dist <- as.dist(unifrac, upper = T, diag = T)

# Calculate Phylo Diversity
phylo_final <- pd(otu.2, tree, include.root = F)

rownames(meta)=meta$sample

all.equal(rownames(meta), rownames(phylo_final))

dim(phylo_final)
dim(meta)

wrc.phylo <- cbind(meta,phylo_final)
###At some point diversity measures get added twice
wrc.phylo <- wrc.phylo[,-c(33:36)]
```

```{r phylogenetic diversity - stats and graphs}
# run full parametric statistical model
PD.lm <- lm(PD ~ treatment * source, data = wrc.phylo)
plot(PD.lm)
anova(PD.lm)

# run linear regression on diversity
PD.reg <- lm(PD~shannon, data = wrc.phylo)
summary(PD.reg)

ggplot(wrc.phylo,aes(x=shannon,y=PD))+stat_summary(fun.data=mean_cl_normal) + 
  geom_smooth(method='lm',formula=y~x) +
  geom_point(aes(color=treatment))

ggplot(wrc.phylo,aes(x=shannon,y=PD, color=treatment))+stat_summary(fun.data=mean_cl_normal) + 
  geom_smooth(method='lm',formula=y~x) +
  geom_point(aes(color=treatment))

ggplot(wrc.phylo,aes(x=shannon,y=PD, color=source))+stat_summary(fun.data=mean_cl_normal) + 
  geom_smooth(method='lm',formula=y~x) +
  geom_point(aes(color=source))

p1 <- ggplot(wrc.phylo, aes(x = source, y = PD, fill = treatment), size=16) +
  geom_flat_violin(aes(fill = treatment),position = position_nudge(x = .1, y = 0), adjust = 1.5, trim = FALSE, alpha = .5, colour = NA)+
  geom_point(aes(x = source, y = PD, colour = treatment),position = position_jitter(width = .05), size = 1.5, shape = 20)+
  geom_boxplot(aes(x = source, y = PD, fill = treatment),outlier.shape = NA, alpha = .5, width = .1, colour = "black") + labs(x = "Sampling Date", y = "Phylogenetic Diversity") 
p1

p2 <- ggplot(wrc.phylo, aes(x = treatment, y = PD, fill = source), size=16) +
  geom_flat_violin(aes(fill = source),position = position_nudge(x = .1, y = 0), adjust = 1.5, trim = FALSE, alpha = .5, colour = NA)+
  geom_point(aes(x = treatment, y = PD, colour = source),position = position_jitter(width = .05), size = 1.5, shape = 20)+
  geom_boxplot(aes(x = treatment, y = PD, fill = source),outlier.shape = NA, alpha = .5, width = .1, colour = "black") + labs(x = "Sampling Date", y = "Phylogenetic Diversity") 
p2
```

```{r Calculate distance matrix and PCoA components, inlcued = FALSE }
# Principal Coordinates Analysis
otu.2.rel.dist.pcoa <- cmdscale(unifrac.dist, k=3, eig=TRUE, add=FALSE)
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1a <- round(otu.2.rel.dist.pcoa$eig[1] / sum(otu.2.rel.dist.pcoa$eig), 3) * 100
explainvar2a <- round(otu.2.rel.dist.pcoa$eig[2] / sum(otu.2.rel.dist.pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)

explainvar1a #42.2
explainvar2a #23.6

```

```{r Unifrac - PERMANOVA}
#PERMANOVA with source and treatment as factors
#Among all treatments source (soil source) and treatment
####I need to combine meta and unifrac.dist. Then run adonis 
m <- as.matrix(unifrac.dist)
rownames(meta) <- meta$sample
all.equal(rownames(m), rownames(meta))
unifrac.dist.meta<- cbind(meta, m)
head(unifrac.dist.meta[,-c(1:36)])

metaotu.ad <- adonis(unifrac.dist.meta[,-c(1:32)] ~source*treatment, method = "bray", data = unifrac.dist.meta, perm=1000, set.seed=42)

metaotu.ad 
```

```{r unifrac - PCoA}
all.equal(rownames(meta), rownames(phylo_final))
# Set treatments
treat1 <- as.factor(meta$source)
levels(treat1) <- ordered(c("bulk","grass", "forb"))
treat2 <- as.factor(meta$treatment)
#treat3 <-as.factor(meta$year)
#levels(treat2) <- c("M", "MF")

#pcoa.groups <- paste(meta$source, meta$treatment, meta$year, sep = "_")
pcoa.groups <- paste(meta$source, meta$treatment, sep = "_")


pcoa.points <- data.frame(otu.2.rel.dist.pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # treatment
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # source
#pcoa.col2 <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 3))

levels(pcoa.col) <- c("Unfertilized","Fertilized")

#Removed RS samples becuase only have soil porperty values for RP samples
#Run envfit to see which soil varibales are predicators 
#colnames(meta)

###Are these just the environmental parameters have values for bulk and rhizospehre soils?
#fit<-envfit(otu.2.rel.dist.pcoa$points, meta[,-c(1:12)], perm=1000, na.rm=T, set.seed=42)
#fit

#A <-as.list(fit$vectors)
#vec<-as.data.frame(fit$vectors$arrows*sqrt(fit$vectors$r)*0.35)
#p<- as.data.frame(A$pvals)
#vec <- cbind(vec, p) 
#vec <-  subset(vec, A$pvals<=0.05)

#vec$parm<-rownames(vec)
#colnames(vec)
#vec$parm<-c("Moisture","pH","C%","N%")

#plot PCoA
df1a <- as.data.frame(pcoa.cent.dataframe)
plot1 <- ggplot(df1a, aes(x=V1, y=V2), 
                 group = interaction(pcoa.col, pcoa.shape)) + theme_bw() +
#group = interaction(pcoa.col, pcoa.shape), colour=pcoa.col2))
  geom_point(aes(shape = pcoa.shape, colour = pcoa.col), size=10, stroke=2) + 
  #geom_point(aes(colour=pcoa.col2), size =4.0)+
  #removes gridlines from plot
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  #Plot soil property vectors from envfit
  #geom_segment(data=vec, aes(x=0,xend=Dim1,y=0,yend=Dim2), size=1, 
  #   arrow = arrow(length = unit(0.2, "cm")),colour="black",inherit.aes=F)+ 
  #Label soil vectors. Using annotate because I can't figure out how to move segment text
  #Note when re-run should double check envfit stats to make sure that soil parameters are the same
  #geom_text(data=vec,aes(x=Dim1,y=Dim2,label=vec$parm),size=5, inherit.aes=F)+
  #annotate("text",x=-0.02,y=0.2, label="C%") +
  #annotate("text",x=0.02,y=0.2, label="N%") +
  #annotate("text",x=0,y=-.2, label="moisture") +
  #annotate("text",x=0.1,y=-.225, label="pH") +
  #Set error bacrs for geom_point
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0), colour="black") + 
  #Set colors for treatments
  #scale_colour_manual(values = c("#999999", "#000000")) +
  scale_colour_manual(values = c("gray", "darkgreen","purple","red","yellow","blue","gray","pink","magenta")) +
  #Would have liked to set shapes but couldn't get this to work
  scale_shape_manual(values = c(16,22,15)) +
  #Sets map coordinates
  coord_cartesian(xlim = c(-.1, 0.1), ylim = c(-0.1, 0.1)) +
  #Sets axis text and put border around plot
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24, face="bold"),
        panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.text=element_text(size=24, face="bold"), legend.title = element_text(size=24, face="bold"))+
  #Sets size of tick marks on axis
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  #Sets labels for plot title, axis titles, and legend headings
  xlab("PCoA 1 (42.2%)") + ylab("PCoA 2 (23.6%)") + 
  labs(shape = "Soil Source") +
  labs(colour="treatment") +
  ggtitle("Phylogenetic Composition") +
  theme(legend.position = "none")+
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))
  
plot1 

#Save a copy of the plot
ggsave("../figures/WRC18_Rhizo/16SrRNA_phylo_WRC2018Rhizo.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7.5, height=6, dpi=900, limitsize=TRUE, bg="transparent")

```


```{r BIOLOG Import data, include=FALSE}
#Read in biolog files 
#There is one file for each treatment and all timepoints (n=12) that includes 5 different plates: One nitro (n=31), two PM3 (95), two PM4 (94). 220 different substrates 

#Per treatment all timepoints and plates combined into one file
#Coalesce all files into one master file
getwd()

bulk <- read.csv("../data/WRC18_Rhizo/biolog_bulk.csv")
grass <- read.csv("../data/WRC18_Rhizo/biolog_grass.csv")
forb <- read.csv("../data/WRC18_Rhizo/biolog_forb.csv")
AWCD_design <- read.csv("../data/WRC18_Rhizo/biolog_AWCD_design.csv") #might can delete
biolog_design <- read.csv("../data/WRC18_Rhizo/biolog_design.csv", header=TRUE)
biolog_sub_groups <- read.csv("../data/WRC18_Rhizo/biolog_substrate_groups.csv")

#bulk + forb + grass
biolog_master <- rbind(bulk, grass, forb)

```

```{r BIOLOG eco plate - AWCD}
#Want to calculate the shannon diversity index @ T04 for a sample 
#And then combine all samples for a treatment 

#Subset master biolog file to include only nitro plates
eco <- biolog_master[(biolog_master$plate_type %in% "ECO"),]

#for each plate in eco_T, calculate adjusted abs for each substance in each rep, 3 reps per plate
#plate_num, sample, substance, rep 1 is well == 1:4, rep 2 is well == 5:8, rep 3 is well == 9:12 
#Each rep has its own negative control (water)

eco_rep1 <- eco %>%
 filter(Substance != "water") %>%
 filter(Well %in% c(1,2,3,4))

eco_rep2 <- eco %>%
 filter(Substance != "water") %>%
 filter(Well %in% c(5,6,7,8))

eco_rep3 <- eco %>%
 filter(Substance != "water") %>%
 filter(Well %in% c(9,10,11,12))
  
#get water value to substract from all other substances
eco_rep1_water <- eco %>%
  filter(Substance == "water",Well==c(1,2,3,4)) %>%
  dplyr::select(experiment, sample, timepoint, Substance, abs)

eco_rep2_water <- eco %>%
  filter(Substance == "water",Well==c(5,6,7,8 ))%>%
  dplyr::select(experiment, sample, timepoint, Substance, abs) 

eco_rep3_water <- eco %>%
  filter(Substance == "water",Well==c(9,10,11,12)) %>%
  dplyr::select(experiment, sample, timepoint, Substance, abs)

#join non water substrates and water substrates. Then substract water abs from substrate abs
eco_joined_adj_rep1 <- eco_rep1 %>% 
  left_join(eco_rep1_water, by=c("sample", "timepoint","experiment"))%>%
  mutate(adj_abs = abs.x-abs.y) %>%
  mutate(adj_abs = ifelse(adj_abs < 0, 0, adj_abs))

eco_joined_adj_rep2 <- eco_rep2 %>% 
  left_join(eco_rep2_water, by=c("sample", "timepoint","experiment"))%>%
  mutate(adj_abs = abs.x-abs.y) %>%
  mutate(adj_abs = ifelse(adj_abs < 0, 0, adj_abs))

eco_joined_adj_rep3 <- eco_rep3 %>% 
  left_join(eco_rep3_water, by=c("sample", "timepoint","experiment"))%>%
  mutate(adj_abs = abs.x-abs.y) %>%
  mutate(adj_abs = ifelse(adj_abs < 0, 0, adj_abs))

#For each rep calculate AWCD
eco_AWCD_rep1 <- eco_joined_adj_rep1 %>%
  group_by(sample, timepoint, experiment) %>%
  dplyr::summarise(sum_abs=sum(adj_abs))%>%
  mutate(AWCD = sum_abs/31) 

eco_join_AWCD_rep1 <- eco_joined_adj_rep1 %>% 
  left_join(eco_AWCD_rep1, by=c("sample","timepoint", "experiment"))

eco_AWCD_rep2 <- eco_joined_adj_rep2 %>%
  group_by(sample, timepoint, experiment) %>%
  dplyr::summarise(sum_abs=sum(adj_abs))%>%
  mutate(AWCD = sum_abs/31) 

eco_join_AWCD_rep2 <- eco_joined_adj_rep2 %>% 
  left_join(eco_AWCD_rep2, by=c("sample","timepoint", "experiment"))

eco_AWCD_rep3 <- eco_joined_adj_rep3 %>%
  group_by(sample, timepoint, experiment) %>%
  dplyr::summarise(sum_abs=sum(adj_abs))%>%
  mutate(AWCD = sum_abs/31) 

eco_join_AWCD_rep3 <- eco_joined_adj_rep3 %>% 
  left_join(eco_AWCD_rep3, by=c("sample","timepoint", "experiment")) 

#combine it all!
eco_joined_all <- bind_rows(eco_join_AWCD_rep1, eco_join_AWCD_rep2, eco_join_AWCD_rep3)
#Remove 201
eco_joined_all <- eco_joined_all %>%
  filter(sample != "201-M-EC-2") 

#Now we need a design file!
biolog_sub_groups_eco <- biolog_sub_groups %>% 
    filter(plate_type=="ECO") %>%
    slice(1:32)

eco_joined_all <- eco_joined_all %>% 
  dplyr::rename(Substance=Substance.x) %>%
  left_join(biolog_design, by=c("sample")) %>% 
  left_join(biolog_sub_groups_eco, by="Substance") %>%
  filter(sample!="Blank")

#need to get average for each treatment source combo
eco_AWCD_plot <- ggplot(data=eco_joined_all, aes(x=timepoint, y=AWCD, color=treatment)) +
  stat_summary(geom="point", fun.y=mean, na.rm=TRUE, size=3) +
  scale_color_manual(labels=c("Unfertilized", "Fertilized"), values=c("gray","darkgreen")) +
  facet_grid(~ source) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24,    face="bold"), panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.text=element_text(size=24, face="bold"), legend.title = element_text(size=24,        face="bold"))+
  theme(axis.text.x=element_blank()) +
    theme(strip.text.x = element_text(size = 24, face="bold")) +
  ggtitle("Carbon AWCD") +
  xlab("Timepoints T0-T11") +
  ylab("AWCD") +
  theme(legend.position = "none")
  
eco_AWCD_plot

ggsave("../figures/WRC18_Rhizo/Eco_plate_AWCD.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)

# run full parametric statistical model
###Not sure this makes sense bc of timepoints
eco_AWCD.lm <- lm(AWCD ~ treatment * source, data = eco_joined_all)
plot(eco_AWCD.lm)
anova(eco_AWCD.lm)

# run linear regression on diversity
#eco_AWCD.reg <- lm(AWCD~eco_shannon, data = eco_joined_all)
#summary(eco_AWCD.reg)
#Need more design details in eco_joined_all
 
```


```{r eco plate T04}
#subset to include only T04 
eco_T <- eco_joined_all %>% 
  filter(timepoint=="T04") 
#Subset design file
biolog_design <- biolog_design %>%
  filter(sample != "201-M-EC-2")

i <- with(eco_T, interaction(treatment, source))

eco_subs <- ggplot(eco_T, aes(treatment, Substance)) +
  geom_tile(aes(fill = adj_abs), colour = "black") + 
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity", x="", y="")+
  scale_fill_gradient(low = "white", high = "black") +
  scale_x_discrete(labels=c("UF","F","UF","F","UF","F"))+
  theme(axis.text.y = element_text(size=14))+
  theme(text = element_text(size=14, face="bold"))

eco_subs

ggsave("../figures/WRC18_Rhizo/eco_sub_overview1.jpg", plot=eco_subs, device=NULL, path=NULL, scale=1, width=11, height=7, dpi=900, limitsize=TRUE)

eco_T_grouped <- eco_T %>% 
  group_by(treatment, source, group)%>%
  dplyr::summarize(adj_abs_avg=mean(adj_abs))

eco_subs_groups <- ggplot(eco_T_grouped, aes(group, treatment)) +
  geom_tile(aes(fill = adj_abs_avg), color="black") + 
  scale_fill_gradient(low = "white", high = "#4b0082") +
  facet_grid(~ source) +
  labs(x="",y="", fill = "Absorbance \n Intensity")+
  coord_flip()+
  theme(axis.text.y = element_text(size=24))+
  theme(text = element_text(size=24, face="bold"), legend.position = "left")+
  scale_y_discrete(labels=c("UF","F","UF","F","UF","F"))

eco_subs_groups

ggsave("../figures/WRC18_Rhizo/eco_sub_overview2.jpg", plot=eco_subs_groups, device=NULL, path=NULL, scale=1, width=12, height=7, dpi=900, limitsize=TRUE)

eco_T_grouped_lm <- eco_T %>% filter(group=="sugar alcohols")
i <- with(eco_T_grouped_lm, interaction(treatment, source))
eco.abs.lm <- lm(adj_abs ~ treatment + source + i , data = eco_T_grouped_lm)
#plot(nitro.abs.lm)
anova(eco.abs.lm)

eco_T_grouped_lm <- eco_T %>% filter(group=="sugar acids")
i <- with(eco_T_grouped_lm, interaction(treatment, source))
eco.abs.lm <- lm(adj_abs ~ treatment + source + i , data = eco_T_grouped_lm)
#plot(nitro.abs.lm)
anova(eco.abs.lm)

eco_T_grouped_lm <- eco_T %>% filter(group=="polysaccharides")
i <- with(eco_T_grouped_lm, interaction(treatment, source))
eco.abs.lm <- lm(adj_abs ~ treatment + source  + i, data = eco_T_grouped_lm)
#plot(nitro.abs.lm)
anova(eco.abs.lm)

eco_T_grouped_lm <- eco_T %>% filter(group=="polymer")
i <- with(eco_T_grouped_lm, interaction(treatment, source))
eco.abs.lm <- lm(adj_abs ~ treatment + source  + i, data = eco_T_grouped_lm)
#plot(nitro.abs.lm)
anova(eco.abs.lm)

eco_T_grouped_lm <- eco_T %>% filter(group=="phospho-sugars")
i <- with(eco_T_grouped_lm, interaction(treatment, source))
eco.abs.lm <- lm(adj_abs ~ treatment + source  + i, data = eco_T_grouped_lm)
#plot(nitro.abs.lm)
anova(eco.abs.lm)

eco_T_grouped_lm <- eco_T %>% filter(group=="oligosaccharides")
i <- with(eco_T_grouped_lm, interaction(treatment, source))
eco.abs.lm <- lm(adj_abs ~ treatment + source , data = eco_T_grouped_lm)
#plot(nitro.abs.lm)
anova(eco.abs.lm)

eco_T_grouped_lm <- eco_T %>% filter(group=="oligopeptides")
i <- with(eco_T_grouped_lm, interaction(treatment, source))
eco.abs.lm <- lm(adj_abs ~ treatment + source  + i, data = eco_T_grouped_lm)
#plot(nitro.abs.lm)
anova(eco.abs.lm)

eco_T_grouped_lm <- eco_T %>% filter(group=="monosaccharides")
i <- with(eco_T_grouped_lm, interaction(treatment, source))
eco.abs.lm <- lm(adj_abs ~ treatment + source  + i, data = eco_T_grouped_lm)
#plot(nitro.abs.lm)
anova(eco.abs.lm)

eco_T_grouped_lm <- eco_T %>% filter(group=="modified sugars")
i <- with(eco_T_grouped_lm, interaction(treatment, source))
eco.abs.lm <- lm(adj_abs ~ treatment + source  + i, data = eco_T_grouped_lm)
#plot(nitro.abs.lm)
anova(eco.abs.lm)

eco_T_grouped_lm <- eco_T %>% filter(group=="modified carboxylic acids ")
i <- with(eco_T_grouped_lm, interaction(treatment, source))
eco.abs.lm <- lm(adj_abs ~ treatment + source + i , data = eco_T_grouped_lm)
#plot(nitro.abs.lm)
anova(eco.abs.lm)

eco_T_grouped_lm <- eco_T %>% filter(group=="L-Amino acid")
i <- with(eco_T_grouped_lm, interaction(treatment, source))
eco.abs.lm <- lm(adj_abs ~ treatment + source , data = eco_T_grouped_lm)
#plot(nitro.abs.lm)
anova(eco.abs.lm)

eco_T_grouped_lm <- eco_T %>% filter(group=="glycosides")
i <- with(eco_T_grouped_lm, interaction(treatment, source))
eco.abs.lm <- lm(adj_abs ~ treatment + source + i , data = eco_T_grouped_lm)
#plot(nitro.abs.lm)
anova(eco.abs.lm)

eco_T_grouped_lm <- eco_T %>% filter(group=="carboxylic acids")
i <- with(eco_T_grouped_lm, interaction(treatment, source))
eco.abs.lm <- lm(adj_abs ~ treatment + source  + i, data = eco_T_grouped_lm)
#plot(nitro.abs.lm)
anova(eco.abs.lm)

eco_T_grouped_lm <- eco_T %>% filter(group=="amine")
i <- with(eco_T_grouped_lm, interaction(treatment, source))
eco.abs.lm <- lm(adj_abs ~ treatment + source + i , data = eco_T_grouped_lm)
#plot(nitro.abs.lm)
anova(eco.abs.lm)



```
```{r Eco T4 AWCD}
eco_T_unique <- eco_T %>%
  distinct(plate_num, treatment, source, AWCD)

eco_subs <- ggplot(eco_T_unique, aes(x=source, y=AWCD, fill=treatment)) +
  geom_boxplot() + 
  scale_fill_manual(values=c("gray","darkgreen"), labels=c("Unfertilized","Fertilized")) +
  #scale_fill_discrete(labels=c("UF","F"),values=c("gray","darkgreen"))+
 theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
  annotate("text",x=0.5,y=1.5, label="A", size=4) +
  annotate("text",x=0.8,y=1.15, label="c", size=4) + #m-b
  annotate("text",x=1.2,y=1.2, label="c", size=4) + #mf-b
  annotate("text",x=1.8,y=1.25, label="c", size=4) + #m-f
  annotate("text",x=2.2,y=1.5, label="a", size=4) + #mf-f
  annotate("text",x=2.8,y=1.3, label="bc", size=4) + #m-g
  annotate("text",x=3.2,y=1.4, label="ab", size=4) + #mf-g
theme(axis.title=element_text(size=14),
          axis.text=element_text(size=14, color="black"), axis.text.x = element_blank(),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          #Set plot title textsize
          theme(plot.title=element_text(size=14)) +
          labs(x = "", y = "Carbon AWCD", title="", fill="Treatment") +
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))

eco_subs

ggsave("../figures/WRC18_Rhizo/pub/eco_awcd_boxplot.jpg", plot=eco_subs, device=NULL, path=NULL, scale=1, width=5, height=3.5, dpi=900, limitsize=TRUE)

i <- with(eco_T_unique, interaction(treatment, source))

eco.awcd.lm <- lm(AWCD ~ treatment + source + i, data = eco_T_unique)
#plot(nitro.abs.lm)
anova(eco.awcd.lm)

t <- HSD.test(eco.awcd.lm, "i", group=TRUE, alpha=0.05)
t
plot(t)

```

```{r biolog eco plate diversity}
#Transform so that substances are columns and sample is row?
#First select just the columns needed... then turn negatve values into 0
eco_joined_adj_sub <- eco_T %>%
  dplyr::select(Substance, sample,  adj_abs) #%>%
  #mutate(adj_abs = ifelse(adj_abs < 0, 0, adj_abs)) #negative values = 0

#I want Substance.x to be columns and sample to be rows and vales to be adj_abs
m <- melt(eco_joined_adj_sub)
eco.cast<-cast(data=m, formula = sample ~ Substance,fun.aggregate = sum)

#Make sample column rownames and remove sample column
rownames(eco.cast) = eco.cast$sample
eco.cast <- eco.cast[, -c(1)]

eco_shannon <- vegan::diversity(eco.cast, "shannon")
wrc.phylo <- wrc.phylo %>%
  filter(sample != "WRC18_201")
wrc.phylo <- cbind(wrc.phylo, eco_shannon)
wrc.phylo<-wrc.phylo[,-c(36)]


eco_shan <- ggplot(wrc.phylo, aes(x=source, y=eco_shannon, fill = treatment))+
  geom_boxplot() +
  #facet_grid(~ source) +
  scale_fill_manual(values=c("gray","darkgreen"), labels=c("Unfertilized","Fertilized")) +
   theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
  annotate("text",x=0.5,y=3.3, label="B", size=4) +
  annotate("text",x=1.0,y=3.2, label="______b______", size=4) + #m-b
  #annotate("text",x=1.2,y=3.0, label="c", size=4) + #mf-b
  #annotate("text",x=1.8,y=3.0, label="c", size=4) + #m-f
  annotate("text",x=2.5,y=3.3, label="_____________a___________", size=4) + #mf-f
  #annotate("text",x=2.8,y=3.0, label="bc", size=4) + #m-g
  #annotate("text",x=3.2,y=3.0, label="ab", size=4) + #mf-g
theme(axis.title=element_text(size=14),
          axis.text=element_text(size=14, color="black"), axis.text.x = element_blank(),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          #Set plot title textsize
          theme(plot.title=element_text(size=14)) +
          labs(x = "", y = "Carbon diversity H'", title="", fill="Treatment") +
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))


eco_shan

ggsave("../figures/WRC18_Rhizo/pub/eco_shan.jpg", plot=eco_shan, device=NULL, path=NULL, scale=1, width=5, height=3.5, dpi=900, limitsize=TRUE, bg="transparent")

# run full parametric statistical model
i <- with(wrc.phylo, interaction(treatment, source))

eco.lm <- lm(eco_shannon ~ treatment + source +i, data = wrc.phylo)
plot(eco.lm)
anova(eco.lm)

t <- HSD.test(eco.lm, "source", group=TRUE, alpha=.05)
t

# run linear regression on diversity
#eco.reg <- lm(PD~eco_shannon, data = wrc.phylo)
#summary(eco.reg)

eco.reg <- lm(fun.shannon~eco_shannon, data = wrc.phylo)
summary(eco.reg)

eco.reg <- lm(bac.shannon~eco_shannon, data = wrc.phylo)
summary(eco.reg)


```

```{r indicator species}

###NEED TO GET ECO_T DATA WITH SAMPLE NAMES AS ROWNAMES AND SUBUSTANCE NAMES AS COLUMNS
  #eco.cast
###TREATMENT AND SOURCE SHOULD BE COLUMNS WITHING THE ABOVE
###SHOULD HAVE A BIOLOG DESIGN TABLE


eco.cast.ind <- eco.cast  %>%
 rownames_to_column(var = "sample")

new.data <- eco.cast.ind %>%
  inner_join(biolog_design, eco.cast.ind , by="sample")

new.data <- new.data %>%
  column_to_rownames(var="sample")

head(new.data)

new.data<- new.data %>% unite(st, c("source", "treatment"))
new.data$st<- as.factor(new.data$st)

design.type <- as.factor(new.data$st)

levels(design.type) 
dataREL.2013 <- new.data[,-c(32:33)]
head(dataREL.2013)
set.seed(42)

#dataREL <- dataREL.2013[, colSums(dataREL.2013) > 0.01]
bac.ind <- indval(dataREL.2013, clustering =  design.type, numitr=1000, set.seed(42))

design.type
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.05)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
#colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$Substance <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

# Export Bacteria Indicator Table
write.table(bac.indicators[,c(5:8)], "../figures/WRC18_Rhizo/pub/BacterialIndicators_biolog_eco.txt",
            sep="\t", row.names = F, quote = F)

  
```

```{r biolog eco plate substance count}
#For each treatment*source get count of group (group is the category of substrate)
#Presence absence; set a threshold positive value
#Left join substrate group to biolog results to add group to each substrate?
eco_joined_adj <- eco_T %>%
  #select(Substance.x, sample,  adj_abs) %>%
 #rename(Substance = Substance.x) %>%
  mutate(PA = ifelse(adj_abs < 0.5, 0, 1) )

i <- with(eco_joined_adj, interaction(source, treatment))

ggplot(eco_joined_adj, aes(x=group, y=PA, fill=treatment)) +
  geom_bar(stat="identity") +
  facet_grid(~ source) +
  coord_flip()    

###Need to consider reps in ECO plate 

eco_joined_adj_total <- eco_joined_adj %>%
  #select(sample, treatment, source, group, PA)%>%
  group_by(treatment, source, sample, plate)%>%
  #count(treatment, source, group)
  dplyr::summarize(sum_PA=sum(PA))

eco_joined_adj_mean <- eco_joined_adj_total %>%
  #select(sample, treatment, source, group, PA)%>%
  group_by(treatment, source)%>%
  #count(treatment, source, group)
  dplyr::summarize(mean_PA=mean(sum_PA), sd_PA=sd(sum_PA))
  
eco_groups <- ggplot(eco_joined_adj_groups, aes(treatment, source)) +
  geom_tile(aes(fill = scale(sum, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) 

eco_groups1 <- ggplot(eco_joined_adj_groups, aes(treatment, sum)) +
  geom_bar(stat="identity")  +
  facet_grid(~ source) 
eco_groups1

ggsave("../figures/WRC18_Rhizo/eco_sub_overview.jpg", plot=eco_subs, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)
```

```{r BIOLOG nitro plate - AWCD}
#Want to calculate the shannon diversity index @ T04 for a sample 
#And then combine all samples for a treatment 

#Subset master biolog file to include only PM3B nitro plates
nitro <- biolog_master[(biolog_master$plate_type %in% "PM3B"),]

#2 reps (A&B) per sample. Each plate is a rep. 
#for each plate in nitro_T, calculate adjusted abs for each substance 
#Each plate/rep has its own negative control (water)
#then combine reps 

#nitro without water
nitro_subs <- nitro %>%
  filter(Substance != "negative control") 

#get water value to substract from all other substances
nitro_rep1_water <- nitro %>%
  filter(Substance == "negative control") %>%
  dplyr::select(experiment, sample, timepoint, Substance, rep, abs) 

#join non water substrates and water substrates. Then substract water abs from substrate abs
nitro_rep1_join <- nitro_subs %>% 
  left_join(nitro_rep1_water, by=c("sample", "timepoint", "experiment", "rep")) %>%
  mutate(adj_abs = abs.x-abs.y) %>% 
  mutate(adj_abs = ifelse(adj_abs < 0, 0, adj_abs))%>%
  filter(sample!="Blank")

#For each plate calculate AWCD 
nitro_AWCD <- nitro_rep1_join %>%
  dplyr::group_by(sample, timepoint, experiment, rep) %>%
  dplyr::summarise(sum_abs=sum(adj_abs))%>%
  mutate(AWCD = sum_abs/95)
#Combine AWCD per timepoint, sample, experiemnt, rep to working dataset
nitro_join_AWCD <- nitro_rep1_join %>% 
  left_join(nitro_AWCD, by=c("sample","timepoint", "experiment", "rep"))
#Now combine reps which are differnt plates

#Add design file
biolog_sub_groups_nitro <- biolog_sub_groups %>% 
    filter(plate_type=="PM3B") %>%
    dplyr::rename(Substance.x=Substance)

nitro_joined_all <- nitro_join_AWCD %>% 
  left_join(biolog_design, by=c("sample")) %>% 
  left_join(biolog_sub_groups_nitro, by="Substance.x")%>%   
  filter(sample != "201-M-EC-2")

i <- with(nitro_joined_all, interaction(treatment, source))

nitro_subs <- ggplot(nitro_joined_all, aes(treatment, Substance.x)) +
  geom_tile(aes(fill = scale(adj_abs, center=FALSE, scale=TRUE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~source) +
  labs(fill = "Absorbance \n Intensity")

nitro_subs

ggsave("../figures/WRC18_Rhizo/nitro_sub_overview.jpg", plot=nitro_subs, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)

#nitro_joined_all_f <- nitro_joined_all %>% filter(timepoint=="T0")
#need to get average for each treatment source combo
nitro_AWCD_plot <- ggplot(data=nitro_joined_all, aes(x=timepoint, y=AWCD, color=treatment)) +
  stat_summary(geom="point", fun.y=mean, na.rm=TRUE, size=3) +
  scale_color_manual(labels=c("Unfertilized", "Fertilized"), values=c("gray","darkgreen")) +
  facet_grid(~ source) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24,    face="bold"), panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.text=element_text(size=24, face="bold"), legend.title = element_text(size=24,        face="bold"))+
  theme(axis.text.x=element_blank()) +
    theme(strip.text.x = element_text(size = 24, face="bold")) +
  ggtitle("Nitrogen AWCD") +
  xlab("Timpoints T0-T11") +
  ylab("AWCD") +
  theme(legend.position = "none")

nitro_AWCD_plot

ggsave("../figures/WRC18_Rhizo/nitro_plate_AWCD.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)



```

```{r nitro plate T04}
#subset to include only T04 
nitro_T <- nitro_joined_all %>% 
  filter(timepoint=="T04") %>%
  filter(Substance.x != "hydroxylamine")

i <- with(nitro_T, interaction(treatment, source))

nitro_subs <- ggplot(nitro_T, aes(treatment, Substance.x)) +
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "#4b0082") +
  labs(fill = "Absorbance \n Intensity")+
  facet_grid(~ source) 

nitro_subs

ggsave("../figures/WRC18_Rhizo/nitro_sub_overview.jpg", plot=nitro_subs, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)

nitro_T_grouped <- nitro_T %>% 
  group_by(treatment, source, group)%>%
  dplyr::summarize(adj_abs_avg=mean(adj_abs))

nitro_subs_groups <- ggplot(nitro_T_grouped, aes(treatment, group)) +
  geom_tile(aes(fill = scale(adj_abs_avg, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "#4b0082") +
  labs(fill = "Absorbance \n Intensity")+
  facet_grid(~ source) 


nitro_subs_groups

ggsave("../figures/WRC18_Rhizo/nitro_group_heat.png", plot=nitro_subs_groups, device=NULL, path=NULL, scale=1, width=12, height=7, dpi=900, limitsize=TRUE, bg="transparent")

nitro_T_grouped_lm <- nitro_T_grouped
i <- with(nitro_T_grouped_lm, interaction(treatment, source))
nitro.abs.lm <- lm(adj_abs_avg ~ treatment + source + i , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(nitro.abs.lm)

t <- HSD.test(nitro.abs.lm, "i", group=TRUE, alpha=.05)
t
#plot(t)

```

```{r}

# run full parametric statistical model
####I THINK I HAVE TO DO THIS FOR EACH GROUP?
nitro_T_grouped_lm <- nitro_T %>% filter(group=="inorganic")
i <- with(nitro_T_grouped_lm, interaction(treatment, source))
nitro.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(nitro.abs.lm)

nitro_T_grouped_lm <- nitro_T %>% filter(group=="other organic ")
i <- with(nitro_T_grouped_lm, interaction(treatment, source))
nitro.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(nitro.abs.lm)

nitro_T_grouped_lm <- nitro_T %>% filter(group=="oligopeptides")
i <- with(nitro_T_grouped_lm, interaction(treatment, source))
nitro.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(nitro.abs.lm)

nitro_T_grouped_lm <- nitro_T %>% filter(group=="nucleosides and nucleotides")
i <- with(nitro_T_grouped_lm, interaction(treatment, source))
nitro.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(nitro.abs.lm)

nitro_T_grouped_lm <- nitro_T %>% filter(group=="nitrogen bases")
i <- with(nitro_T_grouped_lm, interaction(treatment, source))
nitro.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(nitro.abs.lm)

nitro_T_grouped_lm <- nitro_T %>% filter(group=="modified sugars")
i <- with(nitro_T_grouped_lm, interaction(treatment, source))
nitro.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(nitro.abs.lm)

nitro_T_grouped_lm <- nitro_T %>% filter(group=="modified carboxylic acids ")
i <- with(nitro_T_grouped_lm, interaction(treatment, source))
nitro.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(nitro.abs.lm)

nitro_T_grouped_lm <- nitro_T %>% filter(group=="modified amino acids")
i <- with(nitro_T_grouped_lm, interaction(treatment, source))
nitro.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(nitro.abs.lm)

nitro_T_grouped_lm <- nitro_T %>% filter(group=="L-amino acids")
i <- with(nitro_T_grouped_lm, interaction(treatment, source))
nitro.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(nitro.abs.lm)

nitro_T_grouped_lm <- nitro_T %>% filter(group=="D-amino acids")
i <- with(nitro_T_grouped_lm, interaction(treatment, source))
nitro.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(nitro.abs.lm)

nitro_T_grouped_lm <- nitro_T %>% filter(group=="carboxylic acids")
i <- with(nitro_T_grouped_lm, interaction(treatment, source))
nitro.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(nitro.abs.lm)

nitro_T_grouped_lm <- nitro_T %>% filter(group=="amines")
i <- with(nitro_T_grouped_lm, interaction(treatment, source))
nitro.abs.lm <- lm(adj_abs ~ treatment + source + i, data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(nitro.abs.lm)

nitro_T_grouped_lm <- nitro_T %>% filter(group=="allantoin")
i <- with(nitro_T_grouped_lm, interaction(treatment, source))
nitro.abs.lm <- lm(adj_abs ~ treatment + source, data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(nitro.abs.lm)
```

```{r}


n <- nitro_T %>% filter(group == "inorganic")

n.p <- ggplot(n, aes(Substance.x, treatment )) +
  facet_grid(~source)+
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  labs(fill = "Absorbance \n Intensity")+
  coord_flip()
#scale(adj_abs, scale=TRUE, center=FALSE)

n.p

p <- nitro_T %>% 
  filter(group == "inorganic")%>%
  filter(Substance.x != "hydroxylamine") 
  

nitro_subs_groups <- ggplot(p, aes(Substance.x, treatment)) +
  facet_grid(~source)+
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity")+
  coord_flip()

nitro_subs_groups

ggsave("../figures/WRC18_Rhizo/nitro_sub_overview.jpg", plot=nitro_subs_groups, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)


```

```{r N T4 AWCD}
nitro_T_unique <- nitro_T %>%
  distinct(plate_num, treatment, source, AWCD)

nitro_subs <- ggplot(nitro_T_unique, aes(x=source, y=AWCD, fill=treatment)) +
  geom_boxplot() + 
  scale_fill_manual(values=c("gray","darkgreen"), labels=c("Unfertilized","Fertilized")) +
  #scale_fill_discrete(labels=c("UF","F"),values=c("gray","darkgreen"))+
 theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
  annotate("text",x=0.5,y=1.2, label="C", size=4) +
  annotate("text",x=1.0,y=0.7, label="____b____", size=4) +
  annotate("text",x=2.5,y=0.9, label="___________a_________", size=4) +
  annotate("text",x=1.2,y=0.6, label="***", size=4) + #mf-b
  annotate("text",x=2.2,y=0.8, label="***", size=4) + #mf-f
  annotate("text",x=3.2,y=0.75, label="***", size=4) + #mf-g
theme(axis.title=element_text(size=14),
          axis.text=element_text(size=14, color="black"), axis.text.x = element_blank(),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          #Set plot title textsize
          theme(plot.title=element_text(size=14)) +
          labs(x = "", y = "Nitrogen AWCD", title="", fill="Treatment") +
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))

nitro_subs

ggsave("../figures/WRC18_Rhizo/pub/nitro_awcd_boxplot.jpg", plot=nitro_subs, device=NULL, path=NULL, scale=1, width=5, height=3.5, dpi=900, limitsize=TRUE)

i <- with(nitro_T_unique, interaction(treatment, source))

nitro.awcd.lm <- lm(AWCD ~ treatment + source + i, data = nitro_T_unique)
#plot(nitro.abs.lm)
anova(nitro.awcd.lm)

t <- HSD.test(nitro.awcd.lm, "source", group=TRUE, alpha=0.05)
t
plot(t)

```

```{r nitro plate diversity}
#Transform so that substances are columns and sample is row?
#First select just the columns needed... then turn negatve values into 0
nitro_joined_adj_sub <- nitro_T %>%
  dplyr::select(Substance.x, sample,  adj_abs )  #%>%

#I want Substance.x to be columns and sample to be rows and vales to be adj_abs
m <- melt(nitro_joined_adj_sub)
nitro.cast<-cast(data=m, formula = sample ~ Substance.x,fun.aggregate = sum)

#Make sample column rownames and remove sample column
rownames(nitro.cast) = nitro.cast$sample
nitro.cast <- nitro.cast[, -c(1)]

nitro_shannon <- diversity(nitro.cast, "shannon")

wrc.phylo<-wrc.phylo[,-c(35)]
wrc.phylo <- wrc.phylo %>%
  filter(sample != "WRC18_201")
wrc.phylo <- cbind(wrc.phylo, nitro_shannon)
wrc.phylo<-wrc.phylo[,-c(37:38)]



nitro_shan <- ggplot(wrc.phylo, aes(x=source, y=nitro_shannon, fill = treatment))+
  geom_boxplot()+
  #facet_grid(~source)+
 scale_fill_manual(values=c("gray","darkgreen"), labels=c("Unfertilized","Fertilized")) +
  #scale_fill_discrete(labels=c("UF","F"),values=c("gray","darkgreen"))+
 theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
  annotate("text",x=0.5,y=4.4, label="D", size=4) +
  annotate("text",x=1.2,y=4.3, label="***", size=4) + #mf-b
  annotate("text",x=2.2,y=4.4, label="***", size=4) + #mf-f
  annotate("text",x=3.2,y=4.35, label="***", size=4) + #mf-g
theme(axis.title=element_text(size=14),
          axis.text=element_text(size=14, color="black"), axis.text.x = element_blank(),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          #Set plot title textsize
          theme(plot.title=element_text(size=14)) +
          labs(x = "", y = "Nitrogen diversity H'", title="", fill="Treatment") +
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))

nitro_shan

ggsave("../figures/WRC18_Rhizo/pub/nitro_shan.png", plot=nitro_shan, device=NULL, path=NULL, scale=1, width=8, height=6.3, dpi=900, limitsize=TRUE ,bg="transparent")

# run full parametric statistical model
i <- with(wrc.phylo, interaction(treatment, source))
nitro.lm <- lm(nitro_shannon ~ treatment + source + i, data = wrc.phylo)
plot(nitro.lm)
anova(nitro.lm)

t <- HSD.test(nitro.lm, "source", group=TRUE, alpha=.05)
t

# run linear regression on diversity
nitro.reg <- lm(fun.shannon~nitro_shannon, data = wrc.phylo)
summary(nitro.reg)

nitro.reg <- lm(bac.shannon~nitro_shannon, data = wrc.phylo)
summary(nitro.reg)


```
```{r indicator analysis nitro}
nitro.cast.ind <- nitro.cast  %>%
 rownames_to_column(var = "sample")

new.data <- nitro.cast.ind %>%
  inner_join(biolog_design, nitro.cast.ind , by="sample")

new.data <- new.data %>%
  column_to_rownames(var="sample")

head(new.data)

new.data<- new.data %>% unite(st, c("source", "treatment"))
new.data$st<- as.factor(new.data$st)

design.type <- as.factor(new.data$st)

levels(design.type) 
dataREL.2013 <- new.data[,-c(95:96)]
head(dataREL.2013)
set.seed(42)

#dataREL <- dataREL.2013[, colSums(dataREL.2013) > 0.01]
bac.ind <- indval(dataREL.2013, clustering =  design.type, numitr=1000, set.seed(42))

design.type
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.05)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
#colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$Substance <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

# Export Bacteria Indicator Table
write.table(bac.indicators[,c(5:8)], "../figures/WRC18_Rhizo/pub/BacterialIndicators_biolog_nitro.txt",
            sep="\t", row.names = F, quote = F)

```

```{r biolog PM3B nitro plate substance count}
#For each treatment*source get count of group (group is the category of substrate)
#Presence absence; set a threshold positive value
#Left join substrate group to biolog results to add group to each substrate?
nitro_joined_adj <- nitro_joined_all %>%
  #dplyr::select(Substance.x, sample,  adj_abs) %>%
 #rename(Substance = Substance.x) %>%
  mutate(PA = ifelse(adj_abs < 0.5, 0, 1) )

i <- with(nitro_joined_adj, interaction(source, treatment))

ggplot(nitro_joined_adj, aes(x=group, y=PA, fill=treatment)) +
  geom_bar(stat="identity") +
  facet_grid(~ source) +
  coord_flip()   

nitro_joined_adj_groups <- nitro_joined_adj %>%
  #dplyr::select(sample, treatment, source, group, PA)%>%
  group_by(treatment, source, group)%>%
  #count(treatment, source, group)
  summarize(sum=sum(PA))
  
nitro_groups <- ggplot(nitro_joined_adj_groups, aes(treatment, source)) +
  geom_tile(aes(fill = scale(sum, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) 

nitro_groups1 <- ggplot(nitro_joined_adj_groups, aes(treatment, sum)) +
  geom_bar(stat="identity")  +
  facet_grid(~ source) 
nitro_groups1

```

```{r BIOLOG phosphate-use AWCD}
#Want to calculate the shannon diversity index @ T04 for a sample 
#And then combine all samples for a treatment 

#Subset master biolog file to include only nitro plates
po4 <- biolog_master[(biolog_master$plate_type %in% "PM4A"),]
#subset to include only T04

#for each plate in po4_T, calculate adjusted abs for each substance in each rep, 3 reps per plate
#plate_num, sample, substance, rep 1 is well == 1:4, rep 2 is well == 5:8, rep 3 is well == 9:12 
#Each rep has its own negative control (water)

#po4 A1-E12
po4_rep1 <- po4 %>%
 filter(Substance != "negative control") %>%
 filter(Well %in% c(1:12) & Row %in% c("A","B","C","D","E"))  

#sulfur F1-H12
sulfur_rep1 <- po4 %>%
 filter(Substance != "negative control") %>%
 filter(Well %in% c(1:12),Row %in% c("F","G","H")) 
  
#get water value to substract from all other substances
po4_rep1_water <- po4 %>%
  filter(Substance == "negative control",Row ==c("A")) %>%
  dplyr::select(experiment, sample, timepoint, Substance, rep, abs)  

sulfur_rep1_water <- po4 %>%
  filter(Substance == "negative control",Row ==c("F")) %>%
  dplyr::select(experiment, sample, timepoint, Substance, rep, abs)   

#Timepoint T04, All substances except water in rep1 (ie wells 1-4 and rows a-h) joined with water value
po4_rep1_join <- po4_rep1 %>% 
  left_join(nitro_rep1_water, by=c("sample", "timepoint", "experiment", "rep")) %>%
  mutate(adj_abs = abs.x-abs.y) %>% 
  mutate(adj_abs = ifelse(adj_abs < 0, 0, adj_abs))%>%
  filter(sample!="Blank")

sulfur_rep1_join <- sulfur_rep1 %>%
  left_join(nitro_rep1_water, by=c("sample", "timepoint", "experiment", "rep")) %>%
  mutate(adj_abs = abs.x-abs.y) %>% 
  mutate(adj_abs = ifelse(adj_abs < 0, 0, adj_abs))%>%
  filter(sample!="Blank")

#For each plate calculate AWCD 
po4_AWCD <- po4_rep1_join %>%
 dplyr::group_by(sample, timepoint, experiment, rep) %>%
  dplyr::summarise(sum_abs=sum(adj_abs))%>%
  mutate(AWCD = sum_abs/59)

sulfur_AWCD <- sulfur_rep1_join %>%
  group_by(sample, timepoint, experiment, rep) %>%
  summarise(sum_abs=sum(adj_abs))%>%
  mutate(AWCD = sum_abs/36)

  #Combine AWCD per timepoint, sample, experiemnt, rep to working dataset
po4_join_AWCD <- po4_rep1_join %>% 
  left_join(po4_AWCD, by=c("sample","timepoint", "experiment", "rep"))

sulfur_join_AWCD <- po4_rep1_join %>% 
  left_join(sulfur_AWCD, by=c("sample","timepoint", "experiment", "rep"))

#Now combine reps which are differnt plates
```

```{r P graph AWCD}
#Need to add a design file... PO4
biolog_sub_groups_po4 <- biolog_sub_groups %>% 
    filter(plate_type=="PM4A") %>%
    dplyr::rename(Substance.x=Substance)

po4_joined_all <- po4_join_AWCD %>% 
  left_join(biolog_design, by=c("sample")) %>% 
  left_join(biolog_sub_groups_po4, by="Substance.x") %>%
  filter(sample != "201-M-EC-2")

i <- with(po4_joined_all, interaction(treatment, source))

po4_subs <- ggplot(po4_joined_all, aes(treatment, Substance.x)) +
  geom_tile(aes(fill = scale(adj_abs, center=FALSE, scale = TRUE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity")

po4_subs

ggsave("../figures/WRC18_Rhizo/po4_sub_overview.jpg", plot=po4_subs, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)

#po4_joined_all_f <- po4_joined_all %>% filter(timepoint=="T0")
#need to get average for each treatment source combo
po4_AWCD_plot <- ggplot(data=po4_joined_all, aes(x=timepoint, y=AWCD, color=treatment)) +
  stat_summary(geom="point", fun.y=mean, na.rm=TRUE, size=3) +
  scale_color_manual(labels=c("Unfertilized", "Fertilized"), values=c("gray","darkgreen")) +
  facet_grid(~ source) + 
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24,    face="bold"), panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.text=element_text(size=24, face="bold"), legend.title = element_text(size=24,        face="bold"))+
  theme(axis.text.x=element_blank()) +
    theme(strip.text.x = element_text(size = 24, face="bold")) +
  ggtitle("Phosphorus AWCD") +
  xlab("Timpoints T0-T11") +
  ylab("AWCD") +
  theme(legend.position = "none")

po4_AWCD_plot

ggsave("../figures/WRC18_Rhizo/po4_plate_AWCD.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)

i<- with(po4_joined_all,(interaction(treatment,source)))

po4.lm <- lm(AWCD ~ treatment + source+i, data = po4_joined_all)
plot(po4.lm)
anova(po4.lm)

t <- HSD.test(po4.lm, "i", group=TRUE, alpha=.05)
t


po4_subs_groups <- ggplot(po4_joined_all, aes(group, treatment)) +
  geom_tile(aes(fill = scale(adj_abs, center=FALSE, scale = TRUE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity")+
  coord_flip()

po4_subs_groups

ggsave("../figures/WRC18_Rhizo/po4_groups_overview.jpg", plot=po4_subs_groups, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)

```

```{r P plate T04}
#subset to include only T04 
po4_T <- po4_joined_all %>% 
  filter(timepoint=="T04") 
#Subset design file
biolog_design <- biolog_design %>%
  filter(sample != "201-M-EC-2")

i <- with(po4_T, interaction(treatment, source))

po4_T_group_s <- po4_T %>% 
  filter(Substance.x!="triethylphosphate") %>%
  filter(Substance.x!="hypophosphite") %>%
  filter(Substance.x!="phosphono acetic acid")%>%
  filter(Substance.x!="methylene diphosphonic acid")

po4_T_grouped <- po4_T_group_s %>% 
  group_by(treatment, source, group)%>%
  summarize(adj_abs_avg=mean(adj_abs))
  
po4_subs <- ggplot(po4_T_grouped, aes(treatment, group)) +
  geom_tile(aes(fill = adj_abs_avg), colour = "black") + 
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity", x="", y="")+
  scale_fill_gradient(low = "white", high = "#4b0082") +
  scale_x_discrete(labels=c("UF","F","UF","F","UF","F"))+
  theme(axis.text.y = element_text(size=24))+
  theme(text = element_text(size=24, face="bold")) 
po4_subs

ggsave("../figures/WRC18_Rhizo/po4_sub_overview.jpg", plot=po4_subs, device=NULL, path=NULL, scale=1, width=12, height=7, dpi=900, limitsize=TRUE)

#plot(t)
```

```{r p}

# run full parametric statistical model
####I THINK I HAVE TO DO THIS FOR EACH GROUP?
po4_T_grouped_lm <- po4_T_grouped %>% filter(group=="phospho-sugars")
i <- with(po4_T_grouped_lm, interaction(treatment, source))
po4.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(po4.abs.lm)

po4_T_grouped_lm <- po4_T_grouped %>% filter(group=="other oganic ")
i <- with(po4_T_grouped_lm, interaction(treatment, source))
po4.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(po4.abs.lm)

po4_T_grouped_lm <- po4_T_grouped %>% filter(group=="nucleosides and nucleotides")
i <- with(po4_T_grouped_lm, interaction(treatment, source))
po4.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(po4.abs.lm)

po4_T_grouped_lm <- po4_T_grouped %>% filter(group=="modified amino acids ")
i <- with(po4_T_grouped_lm, interaction(treatment, source))
po4.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(po4.abs.lm)

po4_T_grouped_lm <- po4_T_grouped %>% filter(group=="inorganic")
i <- with(po4_T_grouped_lm, interaction(treatment, source))
po4.abs.lm <- lm(adj_abs ~ treatment + source , data = nitro_T_grouped_lm)
#plot(nitro.abs.lm)
anova(po4.abs.lm)




po4_subs_groups <- ggplot(po4_T, aes(group, treatment)) +
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "black") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity")+
  coord_flip()

po4_subs_groups

ggsave("../figures/WRC18_Rhizo/po4_group_overview.jpg", plot=po4_subs_groups, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)
```

```{r PO4 T4 AWCD}
po4_T_unique <- po4_T %>%
  distinct(plate_num, treatment, source, AWCD)

po4_subs <- ggplot(po4_T_unique, aes(x=source, y=AWCD, fill=treatment)) +
  geom_boxplot() + 
  scale_fill_manual(values=c("gray","darkgreen"), labels=c("Unfertilized","Fertilized")) +
  #scale_fill_discrete(labels=c("UF","F"),values=c("gray","darkgreen"))+
 theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
  annotate("text",x=0.5,y=1.0, label="E", size=4) +
  annotate("text",x=1.0,y=0.7, label="____b____", size=4) +
  annotate("text",x=2.5,y=0.7, label="___________a_________", size=4) +
  annotate("text",x=1.2,y=0.6, label="***", size=4) + #mf-b
  annotate("text",x=2.2,y=0.6, label="***", size=4) + #mf-f
  annotate("text",x=3.2,y=0.6, label="***", size=4) + #mf-g
theme(axis.title=element_text(size=14),
          axis.text=element_text(size=14, color="black"), #axis.text.x = element_blank(),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          #Set plot title textsize
          theme(plot.title=element_text(size=14)) +
          labs(x = "", y = "Phosphorus AWCD", title="", fill="Treatment") +
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))

po4_subs

ggsave("../figures/WRC18_Rhizo/pub/po4_awcd_boxplot.jpg", plot=po4_subs, device=NULL, path=NULL, scale=1, width=5, height=3.5, dpi=900, limitsize=TRUE)

i <- with(po4_T_unique, interaction(treatment, source))

po4.awcd.lm <- lm(AWCD ~ treatment + source + i, data = po4_T_unique)
#plot(po4.abs.lm)
anova(po4.awcd.lm)

t <- HSD.test(po4.awcd.lm, "source", group=TRUE, alpha=0.05)
t
plot(t)

```

```{r po4 by group}
po4_T_group <- po4_T %>% filter(group=="phospho-sugars") #%>%
po4_subs <- ggplot(po4_T_group, aes(treatment, Substance.x)) +
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "white") + 
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity", x="", y="")+
  scale_fill_gradient(low = "white", high = "black") +
  scale_x_discrete(labels=c("UF","F","UF","F","UF","F"))+
  theme(axis.text.y = element_text(size=14))+
  theme(text = element_text(size=14, face="bold")) 
po4_subs

ggsave("../figures/WRC18_Rhizo/po4_p_sugars_overview.jpg", plot=po4_subs, device=NULL, path=NULL, scale=1, width=8, height=6, dpi=900, limitsize=TRUE)

po4_T_group <- po4_T %>% filter(group=="inorganic") #%>%
po4_subs <- ggplot(po4_T_group, aes(treatment, Substance.x)) +
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "white") + 
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity", x="", y="")+
  scale_fill_gradient(low = "white", high = "black") +
  scale_x_discrete(labels=c("UF","F","UF","F","UF","F"))+
  theme(axis.text.y = element_text(size=14))+
  theme(text = element_text(size=14, face="bold")) 
po4_subs

ggsave("../figures/WRC18_Rhizo/po4_inorganic_overview.jpg", plot=po4_subs, device=NULL, path=NULL, scale=1, width=8, height=6, dpi=900, limitsize=TRUE)

po4_T_group <- po4_T %>% filter(group=="modified amino acids ") #%>%
po4_subs <- ggplot(po4_T_group, aes(treatment, Substance.x)) +
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "white") + 
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity", x="", y="")+
  scale_fill_gradient(low = "white", high = "black") +
  scale_x_discrete(labels=c("UF","F","UF","F","UF","F"))+
  theme(axis.text.y = element_text(size=14))+
  theme(text = element_text(size=14, face="bold")) 
po4_subs

ggsave("../figures/WRC18_Rhizo/po4_maa_overview.jpg", plot=po4_subs, device=NULL, path=NULL, scale=1, width=16, height=10, dpi=900, limitsize=TRUE)

po4_T_group <- po4_T %>% filter(group=="other organic ") #%>%
po4_subs <- ggplot(po4_T_group, aes(treatment, Substance.x)) +
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "white") + 
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity", x="", y="")+
  scale_fill_gradient(low = "white", high = "black") +
  scale_x_discrete(labels=c("UF","F","UF","F","UF","F"))+
  theme(axis.text.y = element_text(size=14))+
  theme(text = element_text(size=14, face="bold")) 
po4_subs

ggsave("../figures/WRC18_Rhizo/po4_othero_overview.jpg", plot=po4_subs, device=NULL, path=NULL, scale=1, width=16, height=10, dpi=900, limitsize=TRUE)

po4_T_group <- po4_T %>% filter(group=="nucleosides and nucleotides") #%>%
po4_subs <- ggplot(po4_T_group, aes(treatment, Substance.x)) +
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "white") + 
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity", x="", y="")+
  scale_fill_gradient(low = "white",high = "black") +
  scale_x_discrete(labels=c("UF","F","UF","F","UF","F"))+
  theme(axis.text.y = element_text(size=14))+
  theme(text = element_text(size=14, face="bold")) 
po4_subs

ggsave("../figures/WRC18_Rhizo/po4_nucleo_overview.jpg", plot=po4_subs, device=NULL, path=NULL, scale=1, width=16, height=10, dpi=900, limitsize=TRUE)
```

```{r P Diversity}
#Transform so that substances are columns and sample is row?
#First select just the columns needed... then turn negatve values into 0
po4_joined_adj_sub <- po4_T %>%
  dplyr::select(Substance.x, sample,  adj_abs )  #%>%

#I want Substance.x to be columns and sample to be rows and vales to be adj_abs
m <- melt(po4_joined_adj_sub)
po4.cast<-cast(data=m, formula = sample ~ Substance.x,fun.aggregate = sum)
#Make sample column rownames and remove sample column
rownames(po4.cast) = po4.cast$sample
po4.cast <- po4.cast[, -c(1)]

po4_shannon <- diversity(po4.cast, "shannon")

wrc.phylo<- wrc.phylo[,-c(38)]
wrc.phylo <- wrc.phylo %>%
  filter(sample != "WRC18_201")
wrc.phylo <- cbind(wrc.phylo, po4_shannon)
wrc.phylo<- wrc.phylo[,-c(38)]


po4_shan <- ggplot(wrc.phylo, aes(x=source, y=po4_shannon, fill = treatment))+
  geom_boxplot()+
  scale_fill_manual(values=c("gray","darkgreen"), labels=c("Unfertilized","Fertilized")) +
  #scale_fill_discrete(labels=c("UF","F"),values=c("gray","darkgreen"))+
 theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
  annotate("text",x=0.5,y=3.85, label="F", size=4) +
  annotate("text",x=1.0,y=3.85, label="_____b____", size=4) +
  annotate("text",x=2.0,y=3.85, label="_____a____", size=4) +
  annotate("text",x=3.0,y=3.85, label="____ab____", size=4) + 
theme(axis.title=element_text(size=14),
          axis.text=element_text(size=14, color="black"), #axis.text.x = element_blank(),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          #Set plot title textsize
          theme(plot.title=element_text(size=14)) +
          labs(x = "", y = "Phosphorus diversity H'", title="", fill="Treatment") +
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))

po4_shan

ggsave("../figures/WRC18_Rhizo/pub/po4_shan.jpg", plot=po4_shan, device=NULL, path=NULL, scale=1, width=4, height=3.5, dpi=900, limitsize=TRUE, bg="transparent")

# run full parametric statistical model
po4.lm <- lm(po4_shannon ~ treatment * source, data = wrc.phylo)
plot(po4.lm)
anova(po4.lm)

t <- HSD.test(po4.lm, "source", group=TRUE, alpha=.05)
t

# run linear regression on diversity
po4.reg <- lm(fun.shannon~po4_shannon, data = wrc.phylo)
summary(po4.reg)

po4.reg <- lm(bac.shannon~po4_shannon, data = wrc.phylo)
summary(po4.reg)


```
```{r indicator po4}
po4.cast.ind <- po4.cast  %>%
 rownames_to_column(var = "sample")

new.data <- po4.cast.ind %>%
  inner_join(biolog_design, po4.cast.ind , by="sample")

new.data <- new.data %>%
  column_to_rownames(var="sample")

head(new.data)

new.data<- new.data %>% unite(st, c("source", "treatment"))
new.data$st<- as.factor(new.data$st)

design.type <- as.factor(new.data$st)

levels(design.type) 
dataREL.2013 <- new.data[,-c(60:61)]
head(dataREL.2013)
set.seed(42)

#dataREL <- dataREL.2013[, colSums(dataREL.2013) > 0.01]
bac.ind <- indval(dataREL.2013[,-c(2,4,7,8,15,31,33:38,44,52)], clustering =  design.type, numitr=1000, set.seed(42))

design.type
summary(bac.ind)

inds <- which(bac.ind$pval <= 0.05)
bac.indicators <- as.data.frame(matrix(NA, nrow = length(inds), ncol = 4))
#colnames(bac.indicators) <- c("OTU", "Cluster", "IndVal", "Prob")

bac.indicators$Substance <- names(inds)
bac.indicators$Cluster <- bac.ind$maxcls[inds]
bac.indicators$IndVal <- bac.ind$indcls[inds]
bac.indicators$Prob <- bac.ind$pval[inds]

# Export Bacteria Indicator Table
write.table(bac.indicators[,c(5:8)], "../figures/WRC18_Rhizo/pub/BacterialIndicators_biolog_po4.txt",
            sep="\t", row.names = F, quote = F)
```

```{r combine CNP AWCD and diversity}
g <- ggarrange(eco_subs, eco_shan, nitro_subs, nitro_shan, po4_subs, po4_shan, ncol = 2, nrow = 3, heights=c(3.5,3.5,3.5,3.5,3.5,3.5), common.legend = TRUE, legend = "bottom")

g

ggsave("../figures/WRC18_Rhizo/pub/biolog_boxplots.pdf", plot=g, device=NULL, path=NULL, scale=1, width=10, height=10.5, dpi=300, limitsize=TRUE)

ggsave("../figures/WRC18_Rhizo/pub/biolog_boxplots.jpg", plot=g, device=NULL, path=NULL, scale=1, width=10, height=10.5, dpi=300, limitsize=TRUE)
```

```{r otu div correlate to c n p div}
wrc.phylo.div <- wrc.phylo %>%
  dplyr::select(treatment, source, sample, shannon, eco_shannon, nitro_shannon, po4_shannon.1)
wrc.phylo.div.g <- gather(wrc.phylo.div, diversity, value, eco_shannon:po4_shannon.1)
div_correlation <- ggplot(wrc.phylo.div.g, aes(x=shannon, y=value, fill=source, shape=diversity))+
   geom_smooth(method='lm',formula=y~x) +
  geom_point(aes(color=treatment, stroke=3))
div_correlation

mods<- wrc.phylo.div.g %>% group_by(diversity,source) %>% do(model=lm(shannon~value, data=.))
library(broom)
mods$model
mods %>% tidy(model)
mods%>% glance(model)

div.lm <- lm(shannon ~ value + treatment + source, data = wrc.phylo.div.g)
summary(div.lm)
anova(div.lm)

t <- HSD.test(div.lm, "source", group=TRUE, alpha=.05)
t

i<-with(wrc.phylo.div.g,(interaction(treatment, source)))

p1 <- ggplot(wrc.phylo.div.g, aes(x = diversity, y = value, fill = source), size=16) +
  geom_flat_violin(aes(fill = source),position = position_nudge(x = .1, y = 0), adjust = 1.5, trim = FALSE, alpha = .5, colour = NA)+
  geom_point(aes(x = diversity, y = value, colour = source),position = position_jitter(width = .05), size = 1.5, shape = 20)+
  geom_boxplot(aes(x = diversity, y = value, fill = source),outlier.shape = NA, alpha = .5, width = .1, colour = "black") + labs(x = "Nutrient-use diveristy", y = "") 
p1
```

```{r S AWCD graph}
#Need to add a design file... sulfur
biolog_sub_groups_sulfur <- biolog_sub_groups %>% 
    filter(plate_type=="PM4A") %>%
    rename(Substance.x=Substance)

sulfur_joined_all <- sulfur_join_AWCD %>% 
  left_join(biolog_design, by=c("sample")) %>% 
  left_join(biolog_sub_groups_sulfur, by="Substance.x") %>%
  filter(sample != "201-M-EC-2")

i <- with(sulfur_joined_all, interaction(treatment, source))

sulfur_subs <- ggplot(sulfur_joined_all, aes(treatment, Substance.x)) +
  geom_tile(aes(fill = scale(adj_abs, center=FALSE, scale = TRUE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity")

sulfur_subs

ggsave("../figures/WRC18_Rhizo/sulfur_sub_overview.jpg", plot=sulfur_subs, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)

#sulfur_joined_all_f <- sulfur_joined_all %>% filter(timepoint=="T0")
#need to get average for each treatment source combo
sulfur_AWCD_plot <- ggplot(data=sulfur_joined_all, aes(x=timepoint, y=AWCD, color=treatment)) +
  stat_summary(geom="point", fun.y=mean, na.rm=TRUE, size=3) +
  scale_color_manual(labels=c("Unfertilized", "Fertilized"), values=c("gray","darkgreen")) +
  facet_grid(~ source) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24,    face="bold"), panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.text=element_text(size=24, face="bold"), legend.title = element_text(size=24,        face="bold"))+
  theme(axis.text.x=element_blank()) +
    theme(strip.text.x = element_text(size = 24, face="bold")) +
  ggtitle("Sulfur AWCD") +
  xlab("Timpoints T0-T11") +
  ylab("AWCD") +
  theme(legend.position = "none")

sulfur_AWCD_plot

ggsave("../figures/WRC18_Rhizo/sulfur_plate_AWCD.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)

i<-with(sulfur_joined_all, interaction(treatment, source))
sulfur.lm <- lm(AWCD ~ treatment + source + i, data = sulfur_joined_all)
plot(sulfur.lm)
anova(sulfur.lm)

t <- HSD.test(sulfur.lm, "i", group=TRUE, alpha=.05)
t


ggsave("../figures/WRC18_Rhizo/sulfur_groups_overview.jpg", plot=po4_subs_groups, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)
```

```{r S plate T04}
#subset to include only T04 
sulfur_T <- sulfur_joined_all %>% 
  filter(timepoint=="T04") 
#Subset design file
biolog_design <- biolog_design %>%
  filter(sample != "201-M-EC-2")

i <- with(sulfur_T, interaction(treatment, source))

sulfur_subs <- ggplot(sulfur_T, aes(treatment, Substance.x)) +
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  labs(fill = "Absorbance \n Intensity")+
  facet_grid(~ source) 

sulfur_subs

ggsave("../figures/WRC18_Rhizo/sulfur_sub_overview.jpg", plot=sulfur_subs, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)

sulfur_subs_groups <- ggplot(sulfur_T, aes(group, treatment)) +
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity")+
  coord_flip()

sulfur_subs_groups

ggsave("../figures/WRC18_Rhizo/sulfur_sub_overview.jpg", plot=sulfur_subs_groups, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)
```

```{r sulfur diversity}
#SULFUR
#Transform so that substances are columns and sample is row?
#First dplyr::select just the columns needed... then turn negatve values into 0
sulfur_joined_adj_sub <- sulfur_joined_all %>%
  select(Substance.x, sample,  adj_abs ) 

#I want Substance.x to be columns and sample to be rows and vales to be adj_abs
m <- melt(sulfur_joined_adj_sub)
sulfur.cast<-cast(data=m, formula = sample ~ Substance.x,fun.aggregate = sum)
#Make sample column rownames and remove sample column
rownames(sulfur.cast) = sulfur.cast$sample
sulfur.cast <- sulfur.cast[, -c(1)]

sulfur_shannon <- diversity(sulfur.cast, "shannon")

wrc.phylo<- wrc.phylo[,-c(38)]
wrc.phylo <- wrc.phylo %>%
  filter(sample != "WRC18_201")
wrc.phylo <- cbind(wrc.phylo, sulfur_shannon)

sulfur_shan <- ggplot(wrc.phylo, aes(x=treatment, y=sulfur_shannon, fill = treatment))+
  geom_boxplot()+
  facet_grid(~source)+
  scale_fill_manual(labels=c("Unfertilized", "Fertilized"), values=c("gray","darkgreen")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24,      face="bold"), panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.position = "none")+
  theme(axis.text.x=element_blank()) +
    theme(strip.text.x = element_text(size = 24, face="bold")) +
  ggtitle("Sulfur Substrate-Use Diversity") +
  xlab("") +
  ylab("Shannon H'") +
  theme(rect = element_rect(fill = "transparent"))

sulfur_shan

# run full parametric statistical model
sulfur.lm <- lm(sulfur_shannon ~ treatment * source, data = wrc.phylo)
plot(sulfur.lm)
anova(sulfur.lm)

# run linear regression on diversity
sulfur.reg <- lm(PD~sulfur_shannon, data = wrc.phylo)
summary(sulfur.reg)

sulfur.reg <- lm(shannon~sulfur_shannon, data = wrc.phylo)
summary(sulfur.reg)

ggsave("../figures/WRC18_Rhizo/sulfur_shan.png", plot=sulfur_shan, device=NULL, path=NULL, scale=1, width=8, height=8, dpi=900, limitsize=TRUE)
```

```{r biolog PM4A P substance count}
#For each treatment*source get count of group (group is the category of substrate)
#Presence absence; set a threshold positive value
#Left join substrate group to biolog results to add group to each substrate?

po4_joined_adj <- po4_joined_all %>% 
  #rename(Substance = Substance.x) %>%
  #left_join(biolog_sub_groups, by="Substance") %>%
  mutate(PA = ifelse(adj_abs < 0.5, 0, 1) )
  
i <- with(po4_joined_adj, interaction(treatment, source))

ggplot(po4_joined_adj, aes(x=group, y=PA, fill=treatment)) +
  geom_bar(stat="identity") +
  facet_grid(~ source) +
  coord_flip()   

po4_joined_adj_groups <- po4_joined_adj %>%
  #select(sample, treatment, source, group, PA)%>%
  group_by(treatment, source, group)%>%
  #count(treatment, source, group)
  summarize(sum=sum(PA))
  
po4_groups <- ggplot(po4_joined_adj_groups, aes(treatment, source)) +
  geom_tile(aes(fill = scale(sum, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) 

po4_groups1 <- ggplot(po4_joined_adj_groups, aes(treatment, sum)) +
  geom_bar(stat="identity")  +
  facet_grid(~ source) 
po4_groups1

```

```{r biolog sulfur plate substance count}
#For each treatment*source get count of group (group is the category of substrate)
#Presence absence; set a threshold positive value
#Left join substrate group to biolog results to add group to each substrate?
sulfur_joined_adj <- sulfur_T %>% 
 # rename(Substance = Substance.x) %>%
  #left_join(biolog_sub_groups, by="Substance") %>%
  mutate(PA = ifelse(adj_abs < 0.5, 0, 1) )

#So my sample labels are diff in deisgn file and biolog file
#extract unique sample ids from biolog and then add column to meta

  
i <- with(sulfur_joined_adj, interaction(treatment, source))

ggplot(sulfur_joined_adj, aes(x=group, y=PA, fill=treatment)) +
  geom_bar(stat="identity") +
  facet_grid(~ source) +
  coord_flip()  

sulfur_joined_adj_groups <- sulfur_joined_adj %>%
  #select(sample, treatment, source, group, PA)%>%
  group_by(treatment, source, group)%>%
  #count(treatment, source, group)
  summarize(sum=sum(PA))
  
sulfur_groups <- ggplot(sulfur_joined_adj_groups, aes(treatment, source)) +
  geom_tile(aes(fill = scale(sum, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) 

sulfur_groups1 <- ggplot(sulfur_joined_adj_groups, aes(treatment, sum)) +
  geom_bar(stat="identity")  +
  facet_grid(~ source) 
sulfur_groups1

```

```{r AWCD_ALL}

eco_all <- eco_joined_all %>%
  dplyr::select(experiment, timepoint, plate_num, sample, plate_type.x, Substance.x=Substance, adj_abs, AWCD, treatment, group) 

nitro_all <- nitro_joined_all %>%
  dplyr::select(experiment, timepoint, plate_num, sample, plate_type.x, Substance.x, adj_abs, AWCD, treatment, group)

po4_all <- po4_joined_all %>%
  dplyr::select(experiment, timepoint, plate_num, sample, plate_type.x, Substance.x, adj_abs, AWCD, treatment, group)

AWCD_all <- bind_rows(eco_all, nitro_all, po4_all)
#head(AWCD_all)

AWCD_all_T04 <- AWCD_all %>% 
  filter(timepoint=="T04")

i <- with(AWCD_all_T04, interaction(treatment, experiment))

AWCD_all_T04$plate_type.x <- factor(AWCD_all_T04$plate_type.x, labels=c("Carbon","Nitrogen","Phosphorus"))

all <- ggplot(AWCD_all_T04, aes(x=i, y=AWCD, color=treatment)) + 
  geom_boxplot() +
  geom_point(size=2)+
  facet_grid(~ plate_type.x) +
  scale_color_manual(values=c("gray", "darkgreen"), labels=c("Unfertilized", "Fertilized"))   +
  scale_x_discrete(labels=c("Bulk","","Forb","","Grass","")) +
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=18,face="bold"),
          axis.text=element_text(size=18, color="black"),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1),strip.text = element_text(size = 14), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "", y = "AWCD") +
  theme(rect=element_rect(fill="transparent"), text=element_text(size=18, color="black"))+
  theme(plot.background = element_rect(color=NA))

all

ggsave("../figures/WRC18_Rhizo/AWCD_all.png", plot=all, device=NULL, path=NULL, scale=1, width=8, height=4.5, dpi=900, limitsize=TRUE ,bg="transparent")

```
```{r t4}
eco_AWCD_T04 <- AWCD_all_T04 %>% filter(plate_type.x=="Carbon") %>%
dplyr::select(experiment, treatment, timepoint,sample,AWCD) %>%
  distinct()

head(eco_AWCD_T04)

i <- with(eco_AWCD_T04, interaction(treatment, experiment))

eco_awcd.lm <- lm(AWCD ~ treatment + experiment + i, data = eco_AWCD_T04)
plot(eco_awcd.lm)
anova(eco_awcd.lm)

t <- HSD.test(eco_awcd.lm, "i", group=TRUE, alpha=.01)
t

# run linear regression on diversity
#eco_awcd.reg <- lm(AWCD~shannon, data = wrc.phylo)
#summary(eco_awcd.reg)

AWCD_carbon <- ggplot(eco_AWCD_T04, aes(x=treatment, y=AWCD, color=treatment)) + 
  geom_boxplot() +
  geom_point(size=3)+
  facet_grid(~ experiment) +
  scale_fill_manual(values=c("gray", "darkgreen"), labels=c("Unfertilized", "Fertilized"))   +
  scale_x_discrete(labels=c("",""))+
  #scale_x_discrete(labels=c("Bulk","","Forb","","Grass","")) +
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=14,face="bold"),
          axis.text=element_text(size=14),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1),strip.text = element_text(size = 14), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "", y = "AWCD", title="Carbon") +
  theme(rect=element_rect(fill="transparent"))+
  theme(plot.background = element_rect(color=NA))

AWCD_carbon

ggsave("../figures/WRC18_Rhizo/AWCD_carbon.png", plot=AWCD_carbon, device=NULL, path=NULL, scale=1, width=8, height=6.3, dpi=900, limitsize=TRUE ,bg="transparent")

minutes <- read.csv("../data/WRC18_Rhizo/timepoint_minutes.csv")
eco_all_minutes <- eco_all %>% left_join(minutes, by="timepoint")

i <- with(eco_all_minutes, interaction(treatment, experiment))

AWCD_carbon_line <- ggplot(eco_all_minutes, aes(x=minutes, y=AWCD, color=i),group=interaction(treatment, experiment))+
  geom_smooth(size=2) +
  geom_vline(xintercept=72, linetype="dashed", size=2)+
  scale_color_manual(values=c("steelblue4","steelblue1","yellow4","yellow1","tomato4","tomato1"), labels=c("bulk-unfertilzed","bulk-fertilized","grass-unfertilized","grass-fertilized","forb-unfertilized", "forb-fertilized"))+
  theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=24,face="bold"),
          axis.text=element_text(size=24), plot.title = element_text(size=24, face="bold"),
          axis.title.y=element_text(margin=margin(r=10)), legend.title = element_text(size=24, face="bold"),
          panel.border = element_rect(colour = "black",size=1),strip.text = element_text(size = 14), legend.position="none") + 
          theme(legend.title = element_blank())+
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "", y = "", title="Carbon-Use Rates") +
  theme(rect=element_rect(fill="transparent"))+
  theme(plot.background = element_rect(color=NA))

AWCD_carbon_line

ggsave("../figures/WRC18_Rhizo/AWCD_carbon_line.png", plot=AWCD_carbon_line, device=NULL, path=NULL, scale=1, width=8, height=6.3, dpi=900, limitsize=TRUE ,bg="transparent")

```
```{r nitrogen AWCD}
nitro_AWCD_T04 <- AWCD_all_T04 %>% 
  filter(plate_type.x=="Nitrogen") %>%
  dplyr::select(experiment, treatment, timepoint,sample,AWCD) %>%
  distinct()

head(nitro_AWCD_T04)

i <- with(nitro_AWCD_T04, interaction(treatment, experiment))

nitro_awcd.lm <- lm(AWCD ~ treatment + experiment + i, data = nitro_AWCD_T04)
plot(nitro_awcd.lm)
anova(nitro_awcd.lm)

t <- HSD.test(nitro_awcd.lm, "i", group=TRUE, alpha=.01)
t

# run linear regression on diversity
#nitro_awcd.reg <- lm(AWCD~shannon, data = wrc.phylo)
#summary(nitro_awcd.reg)

AWCD_nitrogen <- ggplot(nitro_AWCD_T04, aes(x=treatment, y=AWCD, fill=treatment)) + 
  geom_boxplot() +
  facet_grid(~ experiment) +
  scale_fill_manual(values=c("gray", "darkgreen"), labels=c("Unfertilized", "Fertilized"))   +
  scale_x_discrete(labels=c("",""))+
  #scale_x_discrete(labels=c("Bulk","","Forb","","Grass","")) +
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=24,face="bold"),
          axis.text=element_text(size=24),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1),strip.text = element_text(size = 24), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "Minutes", y = "AWCD", title="Carbon") +
  theme(rect=element_rect(fill="transparent"))+
  theme(plot.background = element_rect(color=NA))

AWCD_nitrogen

ggsave("../figures/WRC18_Rhizo/AWCD_nitrogen.png", plot=AWCD_nitrogen, device=NULL, path=NULL, scale=1, width=8, height=6.3, dpi=900, limitsize=TRUE ,bg="transparent")

minutes <- read.csv("../data/WRC18_Rhizo/timepoint_minutes.csv")
nitro_all_minutes <- nitro_all %>% left_join(minutes, by="timepoint")

i <- with(nitro_all_minutes, interaction(treatment, experiment))

AWCD_nitrogen_line <- ggplot(nitro_all_minutes, aes(x=minutes, y=AWCD, color=i),group=interaction(treatment, experiment))+
  geom_smooth(size=2) +
   geom_vline(xintercept=72, linetype="dashed", size=2)+
  scale_color_manual(values=c("steelblue4","steelblue1","yellow4","yellow1","tomato4","tomato1"), labels=c("bulk-unfertilzed","bulk-fertilized","grass-unfertilized","grass-fertilized","forb-unfertilized", "forb-fertilized"))+
  theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=24,face="bold"),
          axis.text=element_text(size=24), plot.title = element_text(size=24, face="bold"),
          axis.title.y=element_text(margin=margin(r=10)), legend.title = element_text(size=24, face="bold"),
          panel.border = element_rect(colour = "black",size=1),strip.text = element_text(size = 24), legend.position="none") + 
          theme(legend.title = element_blank())+
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "Hours", y = "AWCD", title="Nitrogen-Use Rates") +
  theme(rect=element_rect(fill="transparent"))+
  theme(plot.background = element_rect(color=NA))

AWCD_nitrogen_line

ggsave("../figures/WRC18_Rhizo/AWCD_nitrogen_line.png", plot=AWCD_nitrogen_line, device=NULL, path=NULL, scale=1, width=8, height=6.3, dpi=900, limitsize=TRUE ,bg="transparent")

```

```{r phosphorus AWCD}
po4_AWCD_T04 <- AWCD_all_T04 %>% filter(plate_type.x=="Phosphorus")%>%
  dplyr::select(experiment, treatment, timepoint,sample,AWCD) %>%
  distinct()

head(po4_AWCD_T04)

i <- with(po4_AWCD_T04, interaction(treatment, experiment))

po4_awcd.lm <- lm(AWCD ~ treatment + experiment + i, data = po4_AWCD_T04)
plot(po4_awcd.lm)
anova(po4_awcd.lm)

t <- HSD.test(po4_awcd.lm, "i", group=TRUE, alpha=.01)
t

# run linear regression on diversity
#po4_awcd.reg <- lm(AWCD~shannon, data = wrc.phylo)
#summary(po4_awcd.reg)

AWCD_phosphorus <- ggplot(po4_AWCD_T04, aes(x=treatment, y=AWCD, fill=treatment)) + 
  geom_boxplot() +
  facet_grid(~ experiment) +
  scale_fill_manual(values=c("gray", "darkgreen"), labels=c("Unfertilized", "Fertilized"))   +
  scale_x_discrete(labels=c("",""))+
  #scale_x_discrete(labels=c("Bulk","","Forb","","Grass","")) +
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=14,face="bold"),
          axis.text=element_text(size=14),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1),strip.text = element_text(size = 14), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "", y = "AWCD", title="Carbon") +
  theme(rect=element_rect(fill="transparent"))+
  theme(plot.background = element_rect(color=NA))

AWCD_phosphorus

ggsave("../figures/WRC18_Rhizo/AWCD_phosphorus.png", plot=AWCD_phosphorus, device=NULL, path=NULL, scale=1, width=8, height=6.3, dpi=900, limitsize=TRUE ,bg="transparent")

minutes <- read.csv("../data/WRC18_Rhizo/timepoint_minutes.csv")
po4_all_minutes <- po4_all %>% left_join(minutes, by="timepoint")

i <- with(po4_all_minutes, interaction(treatment, experiment))

AWCD_phosphorus_line <- ggplot(po4_all_minutes, aes(x=minutes, y=AWCD, color=i),group=interaction(treatment, experiment))+
  geom_smooth(size=2) +
  geom_vline(xintercept=72, linetype="dashed", size=2)+
  scale_color_manual(values=c("steelblue4","steelblue1","yellow4","yellow1","tomato4","tomato1"), labels=c("bulk-unfertilzed","bulk-fertilized","grass-unfertilized","grass-fertilized","forb-unfertilized", "forb-fertilized"))+
  theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=24,face="bold"),
          axis.text=element_text(size=24), plot.title = element_text(size=24, face="bold"),
          axis.title.y=element_text(margin=margin(r=10)), legend.title = element_text(size=24, face="bold"),
          panel.border = element_rect(colour = "black",size=1),strip.text = element_text(size = 24), legend.position="none") + 
          theme(legend.title = element_blank())+
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "", y = "", title="Phosphorus-Use Rates") +
  theme(rect=element_rect(fill="transparent"))+
  theme(plot.background = element_rect(color=NA))

AWCD_phosphorus_line

ggsave("../figures/WRC18_Rhizo/AWCD_phosphorus_line.png", plot=AWCD_phosphorus_line, device=NULL, path=NULL, scale=1, width=8, height=6.3, dpi=900, limitsize=TRUE ,bg="transparent")

```

```{r Mantel Tests}
#need to remove WRC-201. Unfortunately this plate fell off shelf and is unreliable
otu.201 <- otu.2[-c(17),]
otu.dist <- vegdist(otu.201, method = "bray")
fungal_otu.rel.201 <- fungal_otu.rel[-c(17),]
fungal.dist <- vegdist(fungal_otu.rel.201, method = "bray")

meta.soilonly.201 <- meta.soilonly[-c(17),]
soil.dist.201 <- vegdist(x = scale(meta.soilonly.201[, -c(1:5,13)]), na.rm=TRUE, "euclid")

#unifrac.dist.201 <- unifrac[-c(17),-c(17)]
#unifrac.dist.201 <- as.dist(unifrac.dist.201, upper = T, diag = T)

eco.dist <- vegdist(eco.cast, method = "euclidean")
nitro.dist <-vegdist(nitro.cast, method = "euclidean")
po4.dist <- vegdist(po4.cast, method="euclidean")
#sulfur.dist <- vegdist(sulfur.cast, method = "euclidean")

mantel.rtest(otu.dist, eco.dist, nrepet = 999)
mantel.rtest(otu.dist, nitro.dist, nrepet = 999)
mantel.rtest(otu.dist, po4.dist, nrepet = 999)
#mantel.rtest(otu.dist, sulfur.dist, nrepet = 999)

mantel.rtest(fungal_otu.rel.dist, otu.2.rel.dist, nrepet=999)

mantel.rtest(eco.dist, nitro.dist, nrepet = 999)
mantel.rtest(po4.dist, nitro.dist, nrepet = 999)
mantel.rtest(po4.dist, eco.dist, nrepet = 999)

mantel.rtest(fungal.dist, eco.dist, nrepet = 999)
mantel.rtest(fungal.dist, nitro.dist, nrepet = 999)
mantel.rtest(fungal.dist, po4.dist, nrepet = 999)

mantel.rtest(soil.dist.201, eco.dist, nrepet = 999)
mantel.rtest(soil.dist.201, nitro.dist, nrepet = 999)
mantel.rtest(soil.dist.201, po4.dist, nrepet = 999)
#mantel.rtest(po4.dist, sulfur.dist, nrepet = 999)
#mantel.rtest(sulfur.dist, eco.dist, nrepet = 999)
#mantel.rtest(sulfur.dist, nitro.dist, nrepet = 999)

#mantel.rtest(unifrac.dist.201,eco.dist, nrepet = 999)
#mantel.rtest(unifrac.dist.201, nitro.dist, nrepet = 999)
#mantel.rtest(unifrac.dist.201, po4.dist, nrepet = 999)
#mantel.rtest(unifrac.dist.201, sulfur.dist, nrepet = 999)

#mantel.rtest(otu.dist, unifrac.dist.201, nrepet = 999)
```

```{r biolog eco plate ORDINATION GRAPH}
#eco.dist <- vegdist(eco.cast, method = "euclidean")
#biolog_design is design file

# Principal Coordinates Analysis
dataREL.dist <- eco.dist

pcoa <- cmdscale(dataREL.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1b <- round(pcoa$eig[1] / sum(pcoa$eig), 3) * 100
explainvar2b <- round(pcoa$eig[2] / sum(pcoa$eig), 3) * 100
sum.eigb <- sum(explainvar1b, explainvar2b)

explainvar1b #39.2
explainvar2b #16.1

eco.meta <- cbind(as.data.frame(eco.cast), biolog_design)

metaotu.ad <- adonis(eco.meta[,-c(32:35)] ~ source*treatment, method = "bray", data = eco.meta, perm=1000, set.seed=42)

metaotu.ad 
```

```{r Ordination (PCoA) - biolog ecoplate}
# Principal Coordinates Analysis - bulk and teas
new.data <-cbind(biolog_design,eco.cast)
pcoa.groups <- paste(new.data$treatment, new.data$source, new.data$ditch, sep = "_")
pcoa.points <- data.frame(pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
trt.c <- c("M","M","M","MF","MF","MF")
source.c <- c("bulk soil","forb","grass","bulk soil","forb","grass")
pcoa.cent.dataframe.trts <- as.data.frame(pcoa.cent.dataframe)
pcoa.cent.dataframe.trts$trt <- as.factor(trt.c)
pcoa.cent.dataframe.trts$source <- as.factor(source.c)

#Plot
df1a <- as.data.frame(pcoa.cent.dataframe.trts)
plot1a <- ggplot(df1a, aes(x=V1, y=V2, colour=trt, shape = source,
                           group = interaction(trt, source))) + theme_bw()   
plot1a <- plot1a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
#theme(panel.background = element_blank()) + 
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + geom_point(aes(colour=trt.c), size=5, stroke = 1.25, show.legend = TRUE) +
scale_colour_manual(labels = c("mowed","mowed/fertilized"), 
                    values = c("darkgray", "darkgreen")) + 
scale_shape_manual(labels = c("bulk soil", 
                              "forb",
                              "grass"), 
                   values = c(16,22,15)) +
theme(axis.title = element_text(size=24), axis.text = element_text(size=24),
      axis.text.x = element_text(size=24, color="black"), axis.text.y=element_text(color="black"),
      panel.border = element_rect(colour = "black", size = 1.25)) + 
theme(axis.ticks.length = unit(0.3, "cm")) + 
  #Set plot title textsize
  theme(plot.title=element_text(size=24)) +
  #Set legend text size
  theme(legend.text=element_text(size=24), legend.title = element_text(size=24))+
xlab("PCoA 1 (39.2%)") + ylab("PCoA 2 (16.1%)") +
  labs(colour = "treatment", shape = "source") +
  guides(colour = guide_legend(override.aes = list(pch=16, size = 10)),
         shape = guide_legend(override.aes = list(size = 10)))+ggtitle("Carbon-Use Beta Diversity")+
  theme(legend.position="none")+
  theme(rect=element_rect(fill="transparent"))+
  theme(plot.background = element_rect(color=NA))
plot1a

ggsave("../figures/WRC18_Rhizo/Ordination_ecoplate.png", plot=plot1a, device=NULL, path=NULL, scale=1, width=5, height=5, dpi=300, limitsize=TRUE, bg="transparent")
```

```{r biolog NITRO plate ORDINATION GRAPH}
#biolog_design is design file
#make sure to run nitro ecodist before proceeding

# Principal Coordinates Analysis
dataREL.dist <- nitro.dist

pcoa <- cmdscale(dataREL.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1b <- round(pcoa$eig[1] / sum(pcoa$eig), 3) * 100
explainvar2b <- round(pcoa$eig[2] / sum(pcoa$eig), 3) * 100
sum.eigb <- sum(explainvar1b, explainvar2b)

explainvar1b #50.3
explainvar2b #7.5

nitro.meta <- cbind(as.data.frame(nitro.cast), biolog_design)

metaotu.ad <- adonis(nitro.meta[,-c(95:98)] ~ source*treatment, method = "bray", data = nitro.meta, perm=1000, set.seed=42)

metaotu.ad 
```

```{r Ordination (PCoA) - biolog Nitrogen plate}
# Principal Coordinates Analysis - bulk and teas
biolog_design <- biolog_design %>%
  filter(sample != "201-M-EC-2")
new.data <-cbind(biolog_design,nitro.cast)
pcoa.groups <- paste(new.data$treatment, new.data$source, new.data$ditch, sep = "_")
pcoa.points <- data.frame(pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
trt.c <- c("M","M","M","MF","MF","MF")
source.c <- c("bulk soil","forb","grass","bulk soil","forb","grass")
pcoa.cent.dataframe.trts <- as.data.frame(pcoa.cent.dataframe)
pcoa.cent.dataframe.trts$trt <- as.factor(trt.c)
pcoa.cent.dataframe.trts$source <- as.factor(source.c)

#Plot
df2a <- as.data.frame(pcoa.cent.dataframe.trts)
plot2a <- ggplot(df2a, aes(x=V1, y=V2, colour=trt, shape = source,
                           group = interaction(trt, source))) + theme_bw()  + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
#theme(panel.background = element_blank()) + 
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + geom_point(aes(colour=trt.c), size=5, stroke = 1.25, show.legend = TRUE) +
scale_colour_manual(labels = c("mowed","mowed/fertilized"), 
                    values = c("darkgray", "darkgreen")) + 
scale_shape_manual(labels = c("bulk soil", 
                              "forb",
                              "grass"), 
                   values = c(16,22,15)) +
theme(axis.title = element_text(size=24), axis.text = element_text(size=24),
      axis.text.x = element_text(size=24, color="black"), axis.text.y=element_text(color="black"),
      panel.border = element_rect(colour = "black", size = 1.25)) + 
theme(axis.ticks.length = unit(0.3, "cm")) + 
   #Set plot title textsize
  theme(plot.title=element_text(size=24)) +
  #Set legend text size
  theme(legend.text=element_text(size=24), legend.title = element_text(size=24))+
xlab("PCoA 1 (50.3%)") + ylab("PCoA 2 (7.5%)") +
  labs(colour = "treatment", shape = "source") +
  guides(colour = guide_legend(override.aes = list(pch=16, size = 4)),
         shape = guide_legend(override.aes = list(size = 4)))+ggtitle("Nitrogen-Use Beta Diversity")+
  theme(legend.position = "none")+
  theme(rect=element_rect(fill="transparent"))+
  theme(plot.background = element_rect(color=NA))
plot2a 

ggsave("../figures/WRC18_Rhizo/Ordination_nitroplate.png", plot=plot2a, device=NULL, path=NULL, scale=1, width=5, height=5, dpi=300, limitsize=TRUE, bg="transparent")
```

```{r biolog PHOSPHORUS plate ORDINATION GRAPH}
#biolog_design is design file
#make sure to run PM3 ecodist before proceeding
phos.dist <- vegdist(po4.cast, method = "euclidean")
# Principal Coordinates Analysis
dataREL.dist <- phos.dist

pcoa <- cmdscale(dataREL.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1b <- round(pcoa$eig[1] / sum(pcoa$eig), 3) * 100
explainvar2b <- round(pcoa$eig[2] / sum(pcoa$eig), 3) * 100
sum.eigb <- sum(explainvar1b, explainvar2b)

explainvar1b #48.3
explainvar2b #12.5

po4.meta <- cbind(as.data.frame(po4.cast), biolog_design)

metaotu.ad <- adonis(po4.meta[,-c(60:63)] ~ source*treatment, method = "bray", data = po4.meta, perm=1000, set.seed=42)

metaotu.ad
```

```{r Ordination (PCoA) - biolog phosphorus plate}
# Principal Coordinates Analysis - bulk and teas
new.data <-cbind(biolog_design,po4.cast)
pcoa.groups <- paste(new.data$treatment, new.data$source, new.data$ditch, sep = "_")
pcoa.points <- data.frame(pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
trt.c <- c("M","M","M","MF","MF","MF")
source.c <- c("bulk soil","forb","grass","bulk soil","forb","grass")
pcoa.cent.dataframe.trts <- as.data.frame(pcoa.cent.dataframe)
pcoa.cent.dataframe.trts$trt <- as.factor(trt.c)
pcoa.cent.dataframe.trts$source <- as.factor(source.c)

#Plot
df3a <- as.data.frame(pcoa.cent.dataframe.trts)
plot3a <- ggplot(df3a, aes(x=V1, y=V2, colour=trt, shape = source,
                           group = interaction(trt, source))) + theme_bw() 
plot3a <- plot3a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
#theme(panel.background = element_blank()) + 
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + geom_point(aes(colour=trt.c), size=5, stroke = 1.25, show.legend = TRUE) +
scale_colour_manual(labels = c("mowed","mowed/fertilized"), 
                    values = c("darkgray", "darkgreen")) + 
scale_shape_manual(labels = c("bulk soil", 
                              "forb",
                              "grass"), 
                   values = c(16,22,15)) +
theme(axis.title = element_text(size=24), axis.text = element_text(size=24),
      axis.text.x = element_text(size=24, color="black"), axis.text.y=element_text(color="black"),
      panel.border = element_rect(colour = "black", size = 1.25)) + 
theme(axis.ticks.length = unit(0.3, "cm")) + 
   #Set plot title textsize
  theme(plot.title=element_text(size=24)) +
  #Set legend text size
  theme(legend.text=element_text(size=24), legend.title = element_text(size=24))+
xlab("PCoA 1 (57.2%)") + ylab("PCoA 2 (9.9%)") +
  labs(colour = "treatment", shape = "source") +
  guides(colour = guide_legend(override.aes = list(pch=16, size = 4)),
         shape = guide_legend(override.aes = list(size = 4)))+ggtitle("Phosphorus-Use Beta Div.")+
  theme(legend.position="none")+
  theme(rect = element_rect(fill = "transparent")) +
  theme(plot.background = element_rect(color=NA))

plot3a
ggsave("../figures/WRC18_Rhizo/Ordination_PO4plate.png", plot=plot3a, device=NULL, path=NULL, scale=1, width=5.2, height=5, dpi=300, limitsize=TRUE, bg="transparent")
```

```{r biolog SULFUR plate ORDINATION GRAPH}
#biolog_design is design file
#make sure to run SULFUR ecodist before proceeding
sulfur.dist <- vegdist(sulfur.cast, method = "euclidean")
# Principal Coordinates Analysis
dataREL.dist <- sulfur.dist

pcoa <- cmdscale(dataREL.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1b <- round(pcoa$eig[1] / sum(pcoa$eig), 3) * 100
explainvar2b <- round(pcoa$eig[2] / sum(pcoa$eig), 3) * 100
sum.eigb <- sum(explainvar1b, explainvar2b)

explainvar1b #39.6
explainvar2b #13

sulfur.meta <- cbind(as.data.frame(sulfur.cast), biolog_design)

metaotu.ad <- adonis(sulfur.meta[,-c(60:63)] ~ source*treatment, method = "bray", data = sulfur.meta, perm=1000, set.seed=42)

metaotu.ad
```

```{r Ordination (PCoA) - biolog sulfur plate}
# Principal Coordinates Analysis - bulk and teas
new.data <-cbind(biolog_design,sulfur.cast)
pcoa.groups <- paste(new.data$treatment, new.data$source, new.data$ditch, sep = "_")
pcoa.points <- data.frame(pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
trt.c <- c("M","M","M","MF","MF","MF")
source.c <- c("bulk soil","forb","grass","bulk soil","forb","grass")
pcoa.cent.dataframe.trts <- as.data.frame(pcoa.cent.dataframe)
pcoa.cent.dataframe.trts$trt <- as.factor(trt.c)
pcoa.cent.dataframe.trts$source <- as.factor(source.c)

#Plot
df4a <- as.data.frame(pcoa.cent.dataframe.trts)
plot4a <- ggplot(df4a, aes(x=V1, y=V2, colour=trt, shape = source,
                           group = interaction(trt, source))) + theme_bw() 

plot4a <- plot4a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + geom_point(aes(colour=trt.c), size=5, stroke = 1.25, show.legend = TRUE) +
scale_colour_manual(labels = c("mowed","mowed/fertilized"), 
                    values = c("darkgray", "darkgreen")) + 
scale_shape_manual(labels = c("bulk soil", 
                              "forb",
                              "grass"), 
                   values = c(16,22,15)) +
theme(axis.title = element_text(size=24), axis.text = element_text(size=24),
      axis.text.x = element_text(size=24), 
      panel.border = element_rect(colour = "black", size = 1.25)) + 
theme(axis.ticks.length = unit(0.3, "cm")) + 
   #Set plot title textsize
  theme(plot.title=element_text(size=24)) +
  #Set legend text size
  theme(legend.text=element_text(size=24), legend.title = element_text(size=24))+
xlab("PCoA 1 (39.6%)") + ylab("PCoA 2 (13.0%)") +
  labs(colour = "treatment", shape = "source") +
  guides(colour = guide_legend(override.aes = list(pch=16, size = 4)),
         shape = guide_legend(override.aes = list(size = 4)))+ggtitle("Sulfur Substrate-Use")+
  theme(legend.position = "none")+
  theme(rect = element_rect(fill = "transparent")) 

plot4a

ggsave("../figures/WRC18_Rhizo/Ordination_sulfurplate.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=5, dpi=300, limitsize=TRUE, bg="transparent")


```

```{r plant cover}
plant_cover <- read.csv("../data/WRC18_Rhizo/Summary_2018_WRC_plants.csv")

ggplot(plant_cover, aes(x=treatment, y=cover, fill= plant)) +
  geom_boxplot() + 
  facet_grid(~ year)

#group by label and rep 
#summarise --- just want percent for poster
plant_cover_reps <- plant_cover %>%
  group_by(year, block, treatment, plant)%>%
  summarise(mean_cover=mean(cover))

ggplot(plant_cover_reps, aes(x=treatment, y=mean_cover, fill=plant)) +
  geom_boxplot() + 
  facet_grid(~ year)

plant_cover_reps_2018 <- plant_cover_reps %>% 
  filter(year=="2018")
  
plant <- ggplot(plant_cover_reps_2018, aes(x=plant, y=mean_cover, fill=treatment)) +
  geom_boxplot()  + 
  scale_fill_manual(values=c("gray", "darkgreen"), labels=c("Unfertilized", "Fertilized"))   +
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=24,face="bold"),
          axis.text=element_text(size=18), plot.title = element_text(size=24, face="bold"),
          axis.title.y=element_text(margin=margin(r=10)), legend.title = element_text(size=24, face="bold"),
          panel.border = element_rect(colour = "black",size=1),strip.text = element_text(size = 24), legend.position="bottom")+ 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "", y = "Average Percent Cover", title="2018")
plant

ggsave("../figures/WRC18_Rhizo/plant_cover.jpg", plot=plant, device=NULL, path=NULL, scale=1, width=7, height=4.4, dpi=900, limitsize=TRUE)
  
```

```{r plant_full}
#All years of experiment
#What is pattern of forbs versus grasses each year?
#What about for AV and EC specifically (each year)?
#x=year, y=percent cover, color = grass/forb or AV/EC
all_plants <- read.csv("../data/WRC18_Rhizo/Plant_WRC_all.csv")
plant_type <- read.csv("../data/WRC18_Rhizo/plant_type.csv")

#Need to melt --- re-order all_plants data so can join
all_plants_type_gather <- gather(all_plants, species, value, Agalinus.purpurea:Xyris.sp)
head(all_plants_type_gather)
all_plants_type <- all_plants_type_gather %>% left_join(plant_type, by="species") %>%
  filter(treatment!="C") %>%
  filter(treatment!="F") %>%
  filter(type!="tree_shrub") %>%
  filter(type!="NA") %>%
  filter(ditch==0) %>%
  filter(mowing==1)%>%
  filter(block=="2"|block=="4"|block=="6"|block=="8")%>%
  filter(data_type=="p cover")
  #group_by(treatment, type, year)%>%
  #summarize_each(avg_pcover=mean(value), avg_sd=sd(value))
  

head(all_plants_type)
#i <- with(all_plants_type, interaction(treatment, type))


#####   summarize then plot
plant_all <- ggplot(all_plants_type, aes(x=year, y=value,shape=type, group=interaction(treatment,type)))+
  geom_point(aes(colour=treatment), size=3, stroke = 1.25)+
  stat_smooth(method="loess", formula=y~x)+ 
  scale_color_manual(values=c("gray", "darkgreen"), labels=c("Unfertilized", "Fertilized"))   +
  scale_fill_manual(values=c("gray", "darkgreen"), labels=c("Unfertilized", "Fertilized"))  +
  scale_shape_manual(values=c(22,15))+
  
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  #Set legend text size
theme(legend.text=element_text(size=14, face="bold"), legend.title = element_text(size=14,        face="bold"))+
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=24,face="bold"),
          axis.text=element_text(size=24), plot.title = element_text(size=24, face="bold"),
          axis.title.y=element_text(margin=margin(r=10)), legend.title = element_text(size=20, face="bold"),
          panel.border = element_rect(colour = "black",size=1),strip.text = element_text(size = 24), legend.position="bottom") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "Year", y = "Average Percent Cover", title="Forb/Grass Cover (2004-2018)")
plant_all

ggsave("../figures/WRC18_Rhizo/plant_all_cover.jpg", plot=plant_all, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)
  #facet_grid(~ treatment)


###HOW DO I TEST IF THE DIFFERENCE BETWEEN FORB M/MF AND GRASS M/MF FOR IS DIFFERENT?
```

```{r rain clouds, include=FALSE}
#Rainclouds for repeated measures
source("../bin/R_rainclouds.R")

#Rainclouds for repeated measures - denitrification
p1 <- ggplot(all_plants_type, aes(x = year, y = avg_pcover, fill = i), size=16) +
  geom_flat_violin(aes(fill = i),position = position_nudge(x = .1, y = 0), adjust = 1.5, trim = FALSE, alpha = .5, colour = NA)#+
  geom_point(aes(x = year, y = avg_pcover, colour = i),position = position_jitter(width = .05), size = 1.5, shape = 20)#+
  geom_boxplot(aes(x = year, y = avg_pcover, fill = i),outlier.shape = NA, alpha = .5, width = .1, colour = "black") + labs(x = "Year", y = "Percent cover") 
  #+ scale_fill_manual(name="Site", values=c("darkgreen","cyan"))+ scale_colour_manual(name="Site", values=c("darkgreen","cyan"))

p1

ggsave("../figures/WRC18_Rhizo/plant_rainclouds.png", plot=last_plot(), device=NULL, path=NULL, scale=1,  width=6, height=6, dpi=300, limitsize=TRUE)
```


```{r load taxonomy data into phyloseq object}
# Import otu and tax into phyloseq
otu <- otu_table(t(otu.2), taxa_are_rows = T)
tax <- import_mothur(mothur_constaxonomy_file =  taxfile)
#rownames(meta) <- meta$group
sample <- sample_data(meta)
#import into phyloseq
phy<-phyloseq(otu,tax)
row.names(sample)=meta$sample
sample_names(sample)
#Assign rownames to be Sample ID's
# Merge mothurdata object with sample metadata
meta_merge <- merge_phyloseq(phy,sample)

#meta_merge

#assign column names
colnames(tax_table(meta_merge))
colnames(tax_table(meta_merge)) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus")
colnames(tax_table(meta_merge))
rank_names(meta_merge)

#Inspecting phyloseq object
#sample_variables(meta_merge)
#sample_names(meta_merge)
#tax_table(meta_merge)[1:5, 2:6]

#treat_merge = merge_samples(meta_merge, sample_data(meta_merge)$pxt)
#otu_table(treat_merge)[1:5,1:5]
```

```{r phyloseq heat maps of taxa composition - phylum}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample
wrc_phylum <- meta_merge %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01) %>%                        # Filter out low abundance taxa
  arrange(Phylum)                                      # Sort data frame alphabetically by phylum

# Set colors for plotting
phylum_colors <- c("brown","orange", "blue", "red","yellow","purple")
  
  #c("#CBD588", "#5F7FC7","#DA5724", "#508578", "#CD9BCD",
  # "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  #"#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861", "#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

# Plot 
#s <- as.factor(wrc_phylum$Sample)
#wrc_phylum$Sample <- ordered(s,levels=c("bulkM","avM", "ecM","bulkMF","avMF", "ecMF"))
#s <- revalue(s, c("bulkM"="bulk unfertilized", "avM"="grass unfertilized", "ecM"="forb unfertilized","bulkMF"="bulk fertilized", "avMF"="grass fertilized", "ecMF"="forb fertilized"))
#s <- mapvalues(s, from = c("bulkM","avM", "ecM","bulkMF","avMF", "ecMF"), to = c("bulk unfertilized","grass unfertilized", "forb unfertilized","bulk fertilized", "grass fertilized", "forb fertilized"))
i <- with(wrc_phylum,interaction(source, treatment))

wrc_plot<- ggplot(wrc_phylum, aes(x = Phylum, y = Abundance, fill = i)) + 
  #facet_grid(Type ~ .) +
  #####UPDATE THIS ON EACH GRAPH ######
  geom_bar(stat = "summary", fun.y = "mean") +
  scale_fill_manual(values = phylum_colors) +
  # Remove x axis title
  theme(axis.title.x = element_blank()) + 
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 1.0%) \n") +
  ggtitle("Phylum Composition of Bacterial Communities By Site") 

wrc_plot + coord_flip()

ggsave(file="../figures/phylum_stack.pdf", width=9, height=9, dpi=300)

# heat map of rel abundance
heat_rel_abund <- ggplot(wrc_phylum, aes(treatment, Phylum)) +
  geom_tile(aes(fill = Abundance), colour = "white") + 
  scale_fill_gradient(low = "gray", high = "black") +
  facet_grid(~source) +
  labs(fill = "Relative Abundance") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
    panel.background = element_blank(), axis.line = element_line(colour = "black"))

heat_rel_abund

ggsave("../figures/WRC18_Rhizo/phylum_heat.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=3, dpi=300, limitsize=TRUE)
```

```{r phyloseq Class stacked graphs of taxa composition}
# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample
wrc_class <- meta_merge %>%
  tax_glom(taxrank = "Class") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01) %>%                        # Filter out low abundance taxa
  arrange(Class)                                     # Sort data frame alphabetically by phylum

heat_rel_abund <- ggplot(wrc_class, aes(treatment, Class)) +
  geom_tile(aes(fill = (Abundance)), colour = "white") + 
  scale_fill_gradient(low = "gray", high = "black") +
  facet_grid(~source) +
  labs(fill = "Relative Abundance") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
    panel.background = element_blank(), axis.line = element_line(colour = "black"))

heat_rel_abund

ggsave("../figures/WRC18_Rhizo/Class_heat.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=3, dpi=300, limitsize=TRUE)
```

```{r phyloseq Family stacked graphs of taxa composition}
# melt to long format (for ggploting) 
# prune out phyla below 1% in each sample
wrc_family <- meta_merge %>%
  tax_glom(taxrank = "Family") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01, Class == "Actinobacteria") %>%                        # Filter out low abundance taxa
  arrange(Family)                                     # Sort data frame alphabetically by phylum

heat_rel_abund <- ggplot(wrc_family, aes(treatment, Family)) +
  geom_tile(aes(fill = Abundance), colour = "white") + 
  scale_fill_gradient(low = "gray", high = "black") +
  facet_grid(~source) +
  labs(fill = "Relative Abundance") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
    panel.background = element_blank(), axis.line = element_line(colour = "black"))

heat_rel_abund

ggsave("../figures/WRC18_Rhizo/family_heat.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=3, dpi=300, limitsize=TRUE)

```

```{r phyloseq Family - Actinobacteria  stacked graphs of taxa composition}
# melt to long format (for ggploting) 
# prune out phyla below 1% in each sample
wrc_family <- meta_merge %>%
  tax_glom(taxrank = "Family") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01, Class == "Alphaproteobacteria") %>%                        # Filter out low abundance taxa
  arrange(Family)                                     # Sort data frame alphabetically by phylum

heat_rel_abund <- ggplot(wrc_family, aes(treatment, Family)) +
  geom_tile(aes(fill = Abundance), colour = "white") + 
  scale_fill_gradient(low = "gray", high = "black") +
  facet_grid(~source) +
  labs(fill = "Relative Abundance") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
    panel.background = element_blank(), axis.line = element_line(colour = "black"))

heat_rel_abund

ggsave("../figures/WRC18_Rhizo/alpha_heat.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=3, dpi=300, limitsize=TRUE)

```

```{r phyloseq Genus stacked graphs of taxa composition}
# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample
wrc_genus <- meta_merge %>%
  tax_glom(taxrank = "Genus") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01) %>%                        # Filter out low abundance taxa
  arrange(Genus)                                     # Sort data frame alphabetically by phylum

heat_rel_abund <- ggplot(wrc_genus, aes(treatment, Genus)) +
  geom_tile(aes(fill = Abundance), colour = "white") + 
  scale_fill_gradient(low = "gray", high = "black") +
  facet_grid(~source) +
  labs(fill = "Relative Abundance") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
    panel.background = element_blank(), axis.line = element_line(colour = "black"))

heat_rel_abund

ggsave("../figures/WRC18_Rhizo/Genus_heat.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=3, dpi=300, limitsize=TRUE)
```