---
title: "WRC2018_Rhizo"
author: "Regina A. B. Bledsoe, Ariane L. Peralta"
date: "May 8, 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning=FALSE)
#use to set working directory 
knitr::opts_knit$set(root.dir="../GitHub/WRC18_RhizoTeaDnitromp/analyses")

```

```{r load packages, include=FALSE}
getwd()

#Set source R tools
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")
source("../bin/R_rainclouds.R")

#load req'd packages 
require("vegan")
require("dplyr")
require("tidyverse")
require("ggplot2")
require("phyloseq")
require("agricolae")
require("ape")
require("picante")
require("reshape2")
require("reshape")
require("ade4")

#Set Std Err and Conf Int
se <- function(x, ...) {
  sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))
}
ci <- function(x, ...) {
  1.96 * sd(x, na.rm = TRUE)
}

set.seed(42)
```

```{r load otu tax and design data, include=FALSE}
getwd()
#Load data files 
#Assign file names to variables
sharedfile = "../data/WRC18_Rhizo/WRC15_18Rhizo.opti_mcc.shared"
taxfile = "../data/WRC18_Rhizo/WRC15_18Rhizo.opti_mcc.0.03.cons.taxonomy"
metafile = "../data/WRC18_Rhizo/2015_2018_WestResearchCampus_Sampling_Summary.csv"

#Read in design file
meta <- read.csv(metafile)
#head(meta)

#Read in OTU file
otu <- read.otu(sharedfile)

#Read in taxonomy file
#tax <- read.tax(taxfile)
#read.tax does not work and I was unable to solve issue. I use phyloseq import_mothur as a work around. 
tax <- import_mothur(mothur_constaxonomy_file =  taxfile)
tax.df <- as.data.frame(tax)
colnames(tax.df)=c("Domain","Phylum","Class", "Order","Family","Genus")
tax <- tax_table(tax.df)
```

```{r filter meta and otu datasets and rarefy - wrc18_Rhizo only , include=FALSE}
#Remove OTUs with less than 2 occurences across all sites
otu.2 <- otu[, which(colSums(otu) >2)]
#If needed can try to reduce tax dataset by removing otus removed from otu file
#otu.1 <- otu[, which(colSums(otu) <2)]
#list <- colnames(otu.1)
#This bit not working atm
#tax.trim <- tax[-c(list),]

#Remove all RP samples
#In the original WRC2015 sampling I collected bulk soils, and rhizospehre soils. Soil loosley adhering to roots was shaken into a bag (RP) for DNA and soil NH4/NO3 extraction. Roots and adhering soils collected for DNA processing (RS). Removed RP from DNA analysis.
#WRC15_65-WRC15_80
rownames(otu.2)
otu.2 <-otu.2[-c(33:48),]
#rownames(otu.2)

#FROM OTU - remove blocks 1, 3, 5, 7 and treatments C & F
#First join meta and otu. Filter and then remove extra columns
otu.2.filter <- as.data.frame(cbind(block = meta$block, treatment = meta$treatment, sample = meta$sample, year = meta$year, otu.2))
otu.2.filter <- otu.2.filter[which(otu.2.filter$block!=1 & otu.2.filter$block!=3 & otu.2.filter$block!=5 & otu.2.filter$block!=7 & meta$treatment!="C" & meta$treatment!="F" & meta$year!=2015),]
otu.2 <- otu.2.filter[,-c(1:4)]
rownames(otu.2.filter)

#FROM META - remove blocks 1, 3, 5, 7 and treatments C & F
meta <- meta[which(meta$block!=1 & meta$block!=3 & meta$block!=5 & meta$block!=7 & meta$treatment!="C" & meta$treatment!="F" & meta$year!=2015),]

#What sample has lowest read count? WRC18_201 16668, next lowest is WRC18_200 43192 highest WRC15_167 68819
otu2<- colSums(t(otu.2))
otu2
otu2[which.min(colSums(t(otu.2)))]
otu2[which.max(colSums(t(otu.2)))]
#Did not remove lowest sample
#rownames(otu.2)
#otu.2 <- otu.2[-c(17),]
```

```{r sequence read counts by sample}
#graph of read counts, need to work on y axis
c<-as.data.frame(otu.2)
c$colsum <- colSums(t(c))
c$group <- rownames(c)
p<-ggplot(c, aes(x=group,y=colsum))+geom_bar(stat="identity")+coord_flip()
p

#Rarefy
#Set min sample number
min.N <- min(rowSums(otu.2))
#rrarefy returns a randomly rarefied community data frame or vector of selected size
#This is the otu set that should be used for beta diversity metrics
otu.rare <- rrarefy(otu.2, min.N)
```

```{r rarefaction curve}
#rarefy returns a species richness estimate in a random subsample of selected size
otu.rare.rich <- rarefy(x = otu.2, sample=min.N, se=T)
#Plot rarefaction curve
###I am not sure how to interpret this. I think it suggests that I did not capture all of the diversity? 
rarecurve(x = otu.2, step = 20, col="blue", cex=0.6, las=1)
abline(0,1,col='red')
text(1000,1000,"1:1",pos=2, col='red')
```

```{r Relative abundance and PERMANOVA}
#transforms read counts to relative abundance
#Using rarefied OTU dataset
otu.2.rel <- otu.rare
for(i in 1:dim(otu.rare)[1]){
  otu.2.rel[i,] <- otu.rare[i,]/sum(otu.rare[i,])
}

#Combine experimental deisng meta data with OTU relative abundance data
metaotu <- cbind(meta, otu.2.rel)
#Where does otu data begin, remove for PERMANOVA
#metaotu[1,20:21]

#PERMANOVA with source and treatment as factors
#Among all treatments source (soil source) and treatment
metaotu.ad <- adonis(metaotu[,-c(1:23)] ~source*treatment, method = "bray", data = metaotu, perm=1000, set.seed=42)

metaotu.ad 

```

```{r Calculate distance matrix and PCoA components - Taxa/Otus, include=FALSE }

otu.2.rel.dist<-vegdist(otu.2.rel, method="bray")

# Principal Coordinates Analysis
otu.2.rel.dist.pcoa <- cmdscale(otu.2.rel.dist, k=3, eig=TRUE, add=FALSE)
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1a <- round(otu.2.rel.dist.pcoa$eig[1] / sum(otu.2.rel.dist.pcoa$eig), 3) * 100
explainvar2a <- round(otu.2.rel.dist.pcoa$eig[2] / sum(otu.2.rel.dist.pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)

explainvar1a #29.1
explainvar2a #12.2

```

```{r PCoA Plot based on Taxa/Otus}
rownames(meta)<-meta$sample
all.equal(rownames(meta), rownames(otu.2.rel))
# Set treatments
treat1 <- as.factor(meta$source)
levels(treat1) <- ordered(c("bulk","grass", "forb"))
treat2 <- as.factor(meta$treatment)
#levels(treat2) <- c("M", "MF")

pcoa.groups <- paste(meta$source, meta$treatment, sep = "_")
pcoa.points <- data.frame(otu.2.rel.dist.pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # treatment
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # source

levels(pcoa.col) <- c("Unfertilized","Fertilized")

#Run envfit to see which soil varibales are predicators 
#These are just the environmental parameters have values for bulk and rhizospehre soils
##fit<-envfit(otu.2.rel.dist.pcoa$points, meta[,-c(1:23)], perm=1000, na.rm=T, set.seed=42)
##fit

##A <-as.list(fit$vectors)
##vec<-as.data.frame(fit$vectors$arrows*sqrt(fit$vectors$r)*0.35)
##p<- as.data.frame(A$pvals)
##vec <- cbind(vec, p) 
##vec <-  subset(vec, A$pvals<=0.05)

##vec$parm<-rownames(vec)
##colnames(vec)
#vec$parm<-c("Moisture","pH","C%","N%")

#plot PCoA
df1a <- as.data.frame(pcoa.cent.dataframe)
plot1 <- ggplot(df1a, aes(x=V1, y=V2), 
                 group = interaction(pcoa.col, pcoa.shape)) + theme_bw() +
#group = interaction(pcoa.col, pcoa.shape), colour=pcoa.col2))
  geom_point(aes(shape = pcoa.shape, colour = pcoa.col), size=7) + 
  #geom_point(aes(colour=pcoa.col2), size =4.0)+
  #removes gridlines from plot
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  #Plot soil property vectors from envfit
  ##geom_segment(data=vec, aes(x=0,xend=Dim1,y=0,yend=Dim2), size=1, 
  ##   arrow = arrow(length = unit(0.2, "cm")),colour="black",inherit.aes=F)+ 
  #Label soil vectors. Using annotate because I can't figure out how to move segment text
  #Note when re-run should double check envfit stats to make sure that soil parameters are the same
  ##geom_text(data=vec,aes(x=Dim1,y=Dim2,label=vec$parm),size=5, inherit.aes=F)+
  #annotate("text",x=-0.02,y=0.2, label="C%") +
  #annotate("text",x=0.02,y=0.2, label="N%") +
  #annotate("text",x=0,y=-.2, label="moisture") +
  #annotate("text",x=0.1,y=-.225, label="pH") +
  #Set error bacrs for geom_point
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e), colour="black") + 
  #Set colors for treatments
  #scale_colour_manual(values = c("#999999", "#000000")) +
  scale_colour_manual(values = c("gray", "darkgreen","purple","red","yellow","blue","gray","pink","magenta")) +
  #Would have liked to set shapes but couldn't get this to work
  scale_shape_manual(values = c(16,22,15)) +
  #Sets map coordinates
  coord_cartesian(xlim = c(-0.2, 0.2), ylim = c(-0.2, 0.2)) +
  #Sets axis text and put border around plot
  theme(axis.title = element_text(size=24), 
        axis.text.x = element_text(size=24),  axis.text.y = element_text(size=24),
        panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24)) +
  #Set legend text size
  theme(legend.text=element_text(size=24), legend.title = element_text(size=24))+
  #Sets size of tick marks on axis
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  #Sets labels for plot title, axis titles, and legend headings
  xlab("PCoA 1 (29.1%)") + ylab("PCoA 2 (12.2%)") + 
  labs(shape = "Soil Source") +
  labs(colour="treatment") +
  ggtitle("16S rRNA Taxonomic Composition") 
  
plot1 

#Save a copy of the plot
ggsave("../figures/WRC18_Rhizo/16SrRNA_WRC2018Rhizo.jpg", plot=plot1, device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)

```

```{r Presence_Absence and Species richness, include =FALSE }
#Transform data to reflect only presence or absence ie 0 or 1
otu.2.PA<-(otu.rare>0*1)
#sanity check 
#otu.2.PA[,1:5]
#Add meta data
metaotu.2 <- cbind(meta, otu.2.PA)
#metaotu.2[1:15]
#Transform True/False to 0/1 
cols <- sapply(metaotu.2, is.logical)
metaotu.2[,cols]<-lapply(metaotu.2[,cols],as.numeric)
#head(metaotu.2)
```

```{r diversity metrics: richness}
#chao1 richness index
chao1 <- estimateR(otu.2)
c <- as.data.frame(chao1)
ct <- t(c)
otu.div <- cbind(meta, ct)

#interaction variable so can use Tukey
i <- with(otu.div, interaction(treatment, source))

richness.lm <- aov(S.chao1 ~ treatment + source + i, data = otu.div)
plot(richness.lm)
anova(richness.lm)

#Get Tukey post hoc groups p<=0.01*
t <- HSD.test(richness.lm, "i", group=TRUE, alpha=.01)
t
plot(t)

#Plot
chao<- ggplot(otu.div, aes(x=i, y=S.chao1, fill=treatment))  +
  geom_boxplot(binaxis='y', stackdir='center') + scale_fill_manual(values=c("#999999", "#000000")) +
#Set labels for Tukey groups
#annotate("text",x=0.82,y=5200, label="b") +
#annotate("text",x=1.81,y=5400, label="b") +
#annotate("text",x=2.81,y=5725, label="b") +
#annotate("text",x=1.1815,y=7050, label="a") +
#annotate("text",x=2.19,y=6600, label="a") + 
#annotate("text",x=3.19,y=6250, label="a") + 
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=14,face="bold"),
          axis.text=element_text(size=14),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="bottom") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "", y = "Chao1 species richness")
chao
#Save a copy of the plot
ggsave("../figures/WRC18_Rhizo/chao1.jpg", plot=chao, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)


```

```{r diversity metrics - calculate diversity and evenness}
# Calculate diversity metrics
shannon <- diversity(otu.rare, "shannon")
simp <- diversity(otu.rare, "simpson")
invsimp <-diversity(otu.rare, "invsimpson")
# Pielous evenness (uses presence/absence data)
J <- shannon/log(specnumber(otu.2.PA[,-c(1:1)]))

#Adding diversity fields to dataset
otu.div$shannon <- NULL
otu.div$J <- NULL
otu.div$simp <- NULL
otu.div$invsimp <- NULL
otu.div <- cbind(otu.div,shannon,J, simp, invsimp)
meta <- cbind(otu.div,shannon,J, simp, invsimp)
```

```{r diversity metrics - display shannon }
#Run richness first. interaction, levels, and treatments for plotting set there
shannon.lm <- aov(shannon ~ treatment + source + i, data = otu.div)
plot(shannon.lm)
anova(shannon.lm)

#Tukey groups at p<=0.01*
t <- HSD.test(shannon.lm, "i", group=TRUE, alpha=0.05)
t
plot(t)

#p is set in richness chuck
shan<- ggplot(otu.div, aes(x=i, y=shannon, fill=treatment))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000"))  + #annotate("text",x=0.82,y=6.1, label="bc") +
#annotate("text",x=1.81,y=5.9, label="c") +
#annotate("text",x=2.81,y=6.05, label="c") +
#annotate("text",x=1.1815,y=6.5, label="a") +
#annotate("text",x=2.19,y=6.3, label="ab") + 
#annotate("text",x=3.19,y=6.35, label="ab") + 
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "", y = "Shannon diversity H'")

shan

ggsave("../figures/WRC18_Rhizo/shannon.jpg", plot=shan, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)
```

```{r diversity metrics - Pielous evennes J}
J.lm <- aov(J ~ treatment + source + i , data = otu.div)
plot(J.lm)
anova(J.lm)
#TukeyHSD(J.lm)
t <- HSD.test(J.lm, "i", group=TRUE, alpha=0.05)
t
plot(t)

even<- ggplot(otu.div, aes(x=i, y=J, fill=treatment))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000")) +
#Set Tukey groups
#annotate("text",x=0.82,y=0.77, label="ab") +
#annotate("text",x=1.81,y=0.75, label="b") +
#annotate("text",x=2.81,y=0.76, label="b") +
#annotate("text",x=1.1815,y=0.80, label="a") +
#annotate("text",x=2.19,y=0.78, label="a")  + 
#annotate("text",x=3.19,y=0.80, label="a")  + 
#Set plot properties
theme_bw()  +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)), 
          axis.title.x=element_text(margin=margin(t=10))) +
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(size=14), panel.border = 
          element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Soil Source", y = "Pileou's eveness") + 
    theme(legend.text=element_text(size=14), legend.title = element_text(size=14), 
          legend.position = "bottom")

even 

ggsave("../figures/WRC18_Rhizo/even.jpg", plot=even, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

```
#This section is IN-PROGRESS Bear ?? .???.?? with me :)
```{r environmental parameters}
meta.env <- meta 

#SUMMARY TABLES
by_plant_treatment <- meta.env %>%
  group_by(source, treatment) %>%
  summarise_all(funs(mean,sd), na.rm=T)

write.csv(by_plant_treatment, file = "by_plant_treatment.csv",row.names=TRUE)

by_treatment <- meta.env %>%
  group_by(treatment) %>%
  summarise_all(funs(mean,sd), na.rm=T)

#i <- with(meta.env, interaction(Treatment, plant))

####STOPPED HERE --- need to update column names and add phosphorus

#NP ratio 
bio_NP <- aov(meta.env$bio_NP ~ treatment, meta.env)
plot(bio_NP)
anova(bio_NP)

bio_NP_p<- ggplot(meta.env[-c(3,17),], aes(x=source, y=bio_NP, fill=treatment))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333","444444"))
bio_NP_p

ggsave("../WRC18_Rhizo/figures/bio_NP.jpg", plot=bio_NP_p, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

tot_NP <- aov(meta.env$tot_NP ~ treatment, meta.env)
plot(tot_NP)
anova(tot_NP)

tot_NP_p<- ggplot(meta.env, aes(x=source, y=tot_NP, fill=treatment))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333","444444"))
tot_NP_p

ggsave("../WRC18_Rhizo/figures/tot_NP.jpg", plot=tot_NP_p, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

metafile.env = "../data/WRC18_Rhizo/2018_WestResearchCampus_Sampling_Summary.csv"

#Read in design file
meta.env <- read.csv(metafile.env)

by_treatment <- meta.env %>%
  group_by(treatment) %>%
  summarise_all(funs(mean,sd), na.rm=T)

write.csv(t(by_treatment), file = "../figures/WRC18_Rhizo/by_treatment.csv",row.names=TRUE)


#KCl extractable Ammonium and Nitrate
meta.env <- meta.env %>%
  filter(source=="bulk")

nh4 <- aov(meta.env$n_nh4_ug_gsoil ~ treatment*ditch, meta.env)
plot(nh4)
anova(nh4)

nh4<- ggplot(meta.env, aes(x=source, y=n_nh4_ug_gsoil, fill=treatment))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333", "444444"))+
  facet_grid(meta.env$ditch)
nh4


pH <- aov(pH ~ treatment , meta.env)
plot(pH)
anova(pH)

ppH<- ggplot(meta.env, aes(x=source, y=pH, fill=treatment))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000", "333333", "444444"))+
  facet_grid(meta.env$ditch)
ppH

#t <- HSD.test(pH, "i" , group=TRUE, alpha=0.05)
#t
#plot(t)

no3 <- aov(no3_mg.l ~ treatment , data=meta.env) 
plot(no3)
anova(no3)

avg_temp_C <- aov(avg_temp_C ~ treatment , data=meta.env) 
plot(avg_temp_C)
anova(avg_temp_C)

moist_percent <- aov(moist_percent ~ treatment , data=meta.env) 
plot(moist_percent)
anova(moist_percent)

TN <- aov(TN ~ treatment , data=meta.env) 
plot(TN)
anova(TN)

c_percent <- aov(c_percent ~ treatment , data=meta.env) 
plot(c_percent)
anova(c_percent)

n_percent <- aov(n_percent ~ treatment , data=meta.env) 
plot(n_percent)
anova(n_percent)

C_N <- aov(C_N ~ treatment , data=meta.env) 
plot(C_N)
anova(C_N)

```


```{r Phylogenetic Diversity - load data, include=FALSE}
# Phylogenetic Diversity - analysis by Mario Muscarella modified by Gina Bledsoe
# Load Tree
tree <- read.tree("../data/WRC18_Rhizo/WRC15-18Rhizo.bac.rename.tree")
tree$tip.label <- gsub("\\|", "", tree$tip.label)
sum(tree$tip.label %in% colnames(otu.2) == FALSE)
# Small Branches
sum(tree$edge.length < 0.0000001)
# Import Unifrac Matrix
unifrac.raw <- read.delim("../data/WRC18_Rhizo/WRC15-18Rhizo.bac.tree1.weighted.phylip.dist", skip = 1, header = F, row.names = 1)
colnames(unifrac.raw) <- rownames(unifrac.raw)
rownames(unifrac.raw)
#Remove spaces from rownames
require("stringr")
rownames(unifrac.raw) <- str_replace_all(rownames(unifrac.raw), fixed(" "), "")
rownames(unifrac.raw)
#narrow unifrac to 2018 Rhizopshere samples
unifrac <- unifrac.raw[which(row.names(unifrac.raw) %in%
                                   row.names(otu.2)),
                            which(row.names(unifrac.raw) %in%
                                   row.names(otu.2))]
dim(unifrac)
# Make into Distance Matrix
unifrac.dist <- as.dist(unifrac, upper = T, diag = T)

# Calculate Phylo Diversity
phylo_final <- pd(otu.2, tree, include.root = F)

rownames(meta)=meta$sample

all.equal(rownames(meta), rownames(phylo_final))

dim(phylo_final)
dim(meta)

wrc.phylo <- cbind(meta,phylo_final)
###At some point diversity measures get added twice
wrc.phylo <- wrc.phylo[,-c(33:36)]
```

```{r phylogenetic diversity - stats and graphs}
# run full parametric statistical model
PD.lm <- lm(PD ~ treatment * source, data = wrc.phylo)
plot(PD.lm)
anova(PD.lm)

# run linear regression on diversity
PD.reg <- lm(PD~shannon, data = wrc.phylo)
summary(PD.reg)

ggplot(wrc.phylo,aes(x=shannon,y=PD))+stat_summary(fun.data=mean_cl_normal) + 
  geom_smooth(method='lm',formula=y~x) +
  geom_point(aes(color=treatment))

ggplot(wrc.phylo,aes(x=shannon,y=PD, color=treatment))+stat_summary(fun.data=mean_cl_normal) + 
  geom_smooth(method='lm',formula=y~x) +
  geom_point(aes(color=treatment))

ggplot(wrc.phylo,aes(x=shannon,y=PD, color=source))+stat_summary(fun.data=mean_cl_normal) + 
  geom_smooth(method='lm',formula=y~x) +
  geom_point(aes(color=source))

p1 <- ggplot(wrc.phylo, aes(x = source, y = PD, fill = treatment), size=16) +
  geom_flat_violin(aes(fill = treatment),position = position_nudge(x = .1, y = 0), adjust = 1.5, trim = FALSE, alpha = .5, colour = NA)+
  geom_point(aes(x = source, y = PD, colour = treatment),position = position_jitter(width = .05), size = 1.5, shape = 20)+
  geom_boxplot(aes(x = source, y = PD, fill = treatment),outlier.shape = NA, alpha = .5, width = .1, colour = "black") + labs(x = "Sampling Date", y = "Phylogenetic Diversity") 
p1

p2 <- ggplot(wrc.phylo, aes(x = treatment, y = PD, fill = source), size=16) +
  geom_flat_violin(aes(fill = source),position = position_nudge(x = .1, y = 0), adjust = 1.5, trim = FALSE, alpha = .5, colour = NA)+
  geom_point(aes(x = treatment, y = PD, colour = source),position = position_jitter(width = .05), size = 1.5, shape = 20)+
  geom_boxplot(aes(x = treatment, y = PD, fill = source),outlier.shape = NA, alpha = .5, width = .1, colour = "black") + labs(x = "Sampling Date", y = "Phylogenetic Diversity") 
p2
```

```{r Calculate distance matrix and PCoA components, inlcued = FALSE }
# Principal Coordinates Analysis
otu.2.rel.dist.pcoa <- cmdscale(unifrac.dist, k=3, eig=TRUE, add=FALSE)
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1a <- round(otu.2.rel.dist.pcoa$eig[1] / sum(otu.2.rel.dist.pcoa$eig), 3) * 100
explainvar2a <- round(otu.2.rel.dist.pcoa$eig[2] / sum(otu.2.rel.dist.pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)

explainvar1a #42.2
explainvar2a #23.6

```
```{r Unifrac - PERMANOVA}
#PERMANOVA with source and treatment as factors
#Among all treatments source (soil source) and treatment
####I need to combine meta and unifrac.dist. Then run adonis 
m <- as.matrix(unifrac.dist)
rownames(meta) <- meta$sample
all.equal(rownames(m), rownames(meta))
unifrac.dist.meta<- cbind(meta, m)
head(unifrac.dist.meta[,-c(1:36)])

metaotu.ad <- adonis(unifrac.dist.meta[,-c(1:32)] ~source*treatment, method = "bray", data = unifrac.dist.meta, perm=1000, set.seed=42)

metaotu.ad 
```

```{r unifrac - PCoA}
all.equal(rownames(meta), rownames(phylo_final))
# Set treatments
treat1 <- as.factor(meta$source)
levels(treat1) <- ordered(c("bulk","grass", "forb"))
treat2 <- as.factor(meta$treatment)
#treat3 <-as.factor(meta$year)
#levels(treat2) <- c("M", "MF")

#pcoa.groups <- paste(meta$source, meta$treatment, meta$year, sep = "_")
pcoa.groups <- paste(meta$source, meta$treatment, sep = "_")


pcoa.points <- data.frame(otu.2.rel.dist.pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # treatment
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # source
#pcoa.col2 <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 3))

levels(pcoa.col) <- c("Unfertilized","Fertilized")

#Removed RS samples becuase only have soil porperty values for RP samples
#Run envfit to see which soil varibales are predicators 
#colnames(meta)

###Are these just the environmental parameters have values for bulk and rhizospehre soils?
#fit<-envfit(otu.2.rel.dist.pcoa$points, meta[,-c(1:12)], perm=1000, na.rm=T, set.seed=42)
#fit

#A <-as.list(fit$vectors)
#vec<-as.data.frame(fit$vectors$arrows*sqrt(fit$vectors$r)*0.35)
#p<- as.data.frame(A$pvals)
#vec <- cbind(vec, p) 
#vec <-  subset(vec, A$pvals<=0.05)

#vec$parm<-rownames(vec)
#colnames(vec)
#vec$parm<-c("Moisture","pH","C%","N%")

#plot PCoA
df1a <- as.data.frame(pcoa.cent.dataframe)
plot1 <- ggplot(df1a, aes(x=V1, y=V2), 
                 group = interaction(pcoa.col, pcoa.shape)) + theme_bw() +
#group = interaction(pcoa.col, pcoa.shape), colour=pcoa.col2))
  geom_point(aes(shape = pcoa.shape, colour = pcoa.col), size=7) + 
  #geom_point(aes(colour=pcoa.col2), size =4.0)+
  #removes gridlines from plot
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  #Plot soil property vectors from envfit
  #geom_segment(data=vec, aes(x=0,xend=Dim1,y=0,yend=Dim2), size=1, 
  #   arrow = arrow(length = unit(0.2, "cm")),colour="black",inherit.aes=F)+ 
  #Label soil vectors. Using annotate because I can't figure out how to move segment text
  #Note when re-run should double check envfit stats to make sure that soil parameters are the same
  #geom_text(data=vec,aes(x=Dim1,y=Dim2,label=vec$parm),size=5, inherit.aes=F)+
  #annotate("text",x=-0.02,y=0.2, label="C%") +
  #annotate("text",x=0.02,y=0.2, label="N%") +
  #annotate("text",x=0,y=-.2, label="moisture") +
  #annotate("text",x=0.1,y=-.225, label="pH") +
  #Set error bacrs for geom_point
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e), colour="black") + 
  #Set colors for treatments
  #scale_colour_manual(values = c("#999999", "#000000")) +
  scale_colour_manual(values = c("gray", "darkgreen","purple","red","yellow","blue","gray","pink","magenta")) +
  #Would have liked to set shapes but couldn't get this to work
  scale_shape_manual(values = c(16,22,15)) +
  #Sets map coordinates
  coord_cartesian(xlim = c(-.1, 0.1), ylim = c(-0.1, 0.1)) +
  #Sets axis text and put border around plot
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24, face="bold"),
        panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.text=element_text(size=24, face="bold"), legend.title = element_text(size=24, face="bold"))+
  #Sets size of tick marks on axis
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  #Sets labels for plot title, axis titles, and legend headings
  xlab("PCoA 1 (42.2%)") + ylab("PCoA 2 (23.6%)") + 
  labs(shape = "Soil Source") +
  labs(colour="treatment") +
  ggtitle("Phylogenetic Composition") 
  
plot1 

#Save a copy of the plot
ggsave("../figures/WRC18_Rhizo/16SrRNA_phylo_WRC2018Rhizo.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)

```

```{r BIOLOG Import data, include=FALSE}
#Read in biolog files 
#There is one file for each treatment and all timepoints (n=12) that includes 5 different plates: One nitro (n=31), two PM3 (95), two PM4 (94). 220 different substrates 

#Per treatment all timepoints and plates combined into one file
#Coalesce all files into one master file
getwd()

bulk <- read.csv("../data/WRC18_Rhizo/biolog_bulk.csv")
grass <- read.csv("../data/WRC18_Rhizo/biolog_grass.csv")
forb <- read.csv("../data/WRC18_Rhizo/biolog_forb.csv")
AWCD_design <- read.csv("../data/WRC18_Rhizo/biolog_AWCD_design.csv") #might can delete
biolog_design <- read.csv("../data/WRC18_Rhizo/biolog_design.csv", header=TRUE)
biolog_sub_groups <- read.csv("../data/WRC18_Rhizo/biolog_substrate_groups.csv")

#bulk + forb + grass
biolog_master <- rbind(bulk, grass, forb)

```

```{r BIOLOG eco plate - AWCD}
#Want to calculate the shannon diversity index @ T04 for a sample 
#And then combine all samples for a treatment 

#Subset master biolog file to include only nitro plates
eco <- biolog_master[(biolog_master$plate_type %in% "ECO"),]

#for each plate in eco_T, calculate adjusted abs for each substance in each rep, 3 reps per plate
#plate_num, sample, substance, rep 1 is well == 1:4, rep 2 is well == 5:8, rep 3 is well == 9:12 
#Each rep has its own negative control (water)

eco_rep1 <- eco %>%
 filter(Substance != "water") %>%
 filter(Well %in% c(1,2,3,4))

eco_rep2 <- eco %>%
 filter(Substance != "water") %>%
 filter(Well %in% c(5,6,7,8))

eco_rep3 <- eco %>%
 filter(Substance != "water") %>%
 filter(Well %in% c(9,10,11,12))
  
#get water value to substract from all other substances
eco_rep1_water <- eco %>%
  filter(Substance == "water",Well==c(1,2,3,4)) %>%
  select(experiment, sample, timepoint, Substance, abs)

eco_rep2_water <- eco %>%
  filter(Substance == "water",Well==c(5,6,7,8)) %>%
  select(experiment, sample, timepoint, Substance, abs) 

eco_rep3_water <- eco %>%
  filter(Substance == "water",Well==c(9,10,11,12)) %>%
  select(experiment, sample, timepoint, Substance, abs)

#join non water substrates and water substrates. Then substract water abs from substrate abs
eco_joined_adj_rep1 <- eco_rep1 %>% 
  left_join(eco_rep1_water, by=c("sample", "timepoint","experiment"))%>%
  mutate(adj_abs = abs.x-abs.y) %>%
  mutate(adj_abs = ifelse(adj_abs < 0, 0, adj_abs))

eco_joined_adj_rep2 <- eco_rep2 %>% 
  left_join(eco_rep2_water, by=c("sample", "timepoint","experiment"))%>%
  mutate(adj_abs = abs.x-abs.y) %>%
  mutate(adj_abs = ifelse(adj_abs < 0, 0, adj_abs))

eco_joined_adj_rep3 <- eco_rep3 %>% 
  left_join(eco_rep3_water, by=c("sample", "timepoint","experiment"))%>%
  mutate(adj_abs = abs.x-abs.y) %>%
  mutate(adj_abs = ifelse(adj_abs < 0, 0, adj_abs))

#For each rep calculate AWCD
eco_AWCD_rep1 <- eco_joined_adj_rep1 %>%
  group_by(sample, timepoint, experiment) %>%
  summarise(sum_abs=sum(adj_abs))%>%
  mutate(AWCD = sum_abs/31) 

eco_join_AWCD_rep1 <- eco_joined_adj_rep1 %>% 
  left_join(eco_AWCD_rep1, by=c("sample","timepoint", "experiment"))

eco_AWCD_rep2 <- eco_joined_adj_rep2 %>%
  group_by(sample, timepoint, experiment) %>%
  summarise(sum_abs=sum(adj_abs))%>%
  mutate(AWCD = sum_abs/31) 

eco_join_AWCD_rep2 <- eco_joined_adj_rep2 %>% 
  left_join(eco_AWCD_rep2, by=c("sample","timepoint", "experiment"))

eco_AWCD_rep3 <- eco_joined_adj_rep3 %>%
  group_by(sample, timepoint, experiment) %>%
  summarise(sum_abs=sum(adj_abs))%>%
  mutate(AWCD = sum_abs/31) 

eco_join_AWCD_rep3 <- eco_joined_adj_rep3 %>% 
  left_join(eco_AWCD_rep3, by=c("sample","timepoint", "experiment")) 

#combine it all!
eco_joined_all <- bind_rows(eco_join_AWCD_rep1, eco_join_AWCD_rep2, eco_join_AWCD_rep3)
#Remove 201
eco_joined_all <- eco_joined_all %>%
  filter(sample != "201-M-EC-2") 

#Now we need a design file!
biolog_sub_groups_eco <- biolog_sub_groups %>% 
    filter(plate_type=="ECO") %>%
    slice(1:32)

eco_joined_all <- eco_joined_all %>% 
  rename(Substance=Substance.x) %>%
  left_join(biolog_design, by=c("sample")) %>% 
  left_join(biolog_sub_groups_eco, by="Substance") %>%
  filter(sample!="Blank")

#need to get average for each treatment source combo
eco_AWCD_plot <- ggplot(data=eco_joined_all, aes(x=timepoint, y=AWCD, color=treatment)) +
  stat_summary(geom="point", fun.y=mean, na.rm=TRUE, size=3) +
  scale_color_manual(labels=c("Unfertilized", "Fertilized"), values=c("gray","darkgreen")) +
  facet_grid(~ source) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24,    face="bold"), panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.text=element_text(size=24, face="bold"), legend.title = element_text(size=24,        face="bold"))+
  theme(axis.text.x=element_blank()) +
    theme(strip.text.x = element_text(size = 24, face="bold")) +
  ggtitle("Carbon Plate AWCD") +
  xlab("Timepoints T0-T11") +
  ylab("AWCD") +
  theme(legend.position = "none")
  
eco_AWCD_plot

ggsave("../figures/WRC18_Rhizo/Eco_plate_AWCD.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)

# run full parametric statistical model
eco_AWCD.lm <- lm(AWCD ~ treatment * source, data = eco_joined_all)
plot(eco_AWCD.lm)
anova(eco_AWCD.lm)

# run linear regression on diversity
#eco_AWCD.reg <- lm(AWCD~eco_shannon, data = eco_joined_all)
#summary(eco_AWCD.reg)
#Need more design details in eco_joined_all
 
```

```{r eco plate T04}
#subset to include only T04 
eco_T <- eco_joined_all %>% 
  filter(timepoint=="T04") 
#Subset design file
biolog_design <- biolog_design %>%
  filter(sample != "201-M-EC-2")

i <- with(eco_T, interaction(treatment, source))

eco_subs <- ggplot(eco_T, aes(treatment, Substance)) +
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  labs(fill = "Absorbance \n Intensity")+
  facet_grid(~ source) 

eco_subs

ggsave("../figures/WRC18_Rhizo/eco_sub_overview.jpg", plot=eco_subs, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)

eco_subs_groups <- ggplot(eco_T, aes(group, treatment)) +
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity")+
  coord_flip()

eco_subs_groups

ggsave("../figures/WRC18_Rhizo/eco_sub_overview.jpg", plot=eco_subs_groups, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)
```

```{r biolog eco plate diversity}
#Transform so that substances are columns and sample is row?
#First select just the columns needed... then turn negatve values into 0
eco_joined_adj_sub <- eco_T %>%
  select(Substance, sample,  adj_abs ) #%>%
  #mutate(adj_abs = ifelse(adj_abs < 0, 0, adj_abs)) #negative values = 0

#I want Substance.x to be columns and sample to be rows and vales to be adj_abs
m <- melt(eco_joined_adj_sub)
eco.cast<-cast(data=m, formula = sample ~ Substance,fun.aggregate = sum)

#Make sample column rownames and remove sample column
rownames(eco.cast) = eco.cast$sample
eco.cast <- eco.cast[, -c(1)]

eco_shannon <- diversity(eco.cast, "shannon")
wrc.phylo <- wrc.phylo %>%
  filter(sample != "WRC18_201")
wrc.phylo <- cbind(wrc.phylo, eco_shannon)
wrc.phylo<-wrc.phylo[,-c(36)]


eco_shan <- ggplot(wrc.phylo, aes(x=treatment, y=eco_shannon, fill = treatment))+
  geom_boxplot() +
  facet_grid(~ source) +
  scale_fill_manual(labels=c("Unfertilized", "Fertilized"), values=c("gray","darkgreen")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24,    face="bold"), panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.position = "none")+
  theme(axis.text.x=element_blank()) +
    theme(strip.text.x = element_text(size = 24, face="bold")) +
  ggtitle("Carbon Substrate-Use Diversity") +
  xlab("") +
  ylab("Shannon H'") 


eco_shan

# run full parametric statistical model
eco.lm <- lm(eco_shannon ~ treatment * source, data = wrc.phylo)
plot(eco.lm)
anova(eco.lm)

# run linear regression on diversity
eco.reg <- lm(PD~eco_shannon, data = wrc.phylo)
summary(eco.reg)

eco.reg <- lm(shannon~eco_shannon, data = wrc.phylo)
summary(eco.reg)

ggsave("../figures/WRC18_Rhizo/eco_shan.jpg", plot=eco_shan, device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)
```

```{r biolog eco plate substance count}
#For each treatment*source get count of group (group is the category of substrate)
#Presence absence; set a threshold positive value
#Left join substrate group to biolog results to add group to each substrate?
eco_joined_adj <- eco_T %>%
  #select(Substance.x, sample,  adj_abs) %>%
 #rename(Substance = Substance.x) %>%
  mutate(PA = ifelse(adj_abs < 0.5, 0, 1) )

i <- with(eco_joined_adj, interaction(source, treatment))

ggplot(eco_joined_adj, aes(x=group, y=PA, fill=treatment)) +
  geom_bar(stat="identity") +
  facet_grid(~ source) +
  coord_flip()    

####group by treatment, mutate count(PA), then graph with heat?
###This looks wierd do not use
eco_joined_adj_groups <- eco_joined_adj %>%
  #select(sample, treatment, source, group, PA)%>%
  group_by(treatment, source, group)%>%
  #count(treatment, source, group)
  summarize(sum=mean(PA))
  
eco_groups <- ggplot(eco_joined_adj_groups, aes(treatment, source)) +
  geom_tile(aes(fill = scale(sum, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) 

eco_groups1 <- ggplot(eco_joined_adj_groups, aes(treatment, sum)) +
  geom_bar(stat="identity")  +
  facet_grid(~ source) 
eco_groups1

ggsave("../figures/WRC18_Rhizo/eco_sub_overview.jpg", plot=eco_subs, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)
```

```{r BIOLOG nitro plate - AWCD}
#Want to calculate the shannon diversity index @ T04 for a sample 
#And then combine all samples for a treatment 

#Subset master biolog file to include only PM3B nitro plates
nitro <- biolog_master[(biolog_master$plate_type %in% "PM3B"),]

#2 reps (A&B) per sample. Each plate is a rep. 
#for each plate in nitro_T, calculate adjusted abs for each substance 
#Each plate/rep has its own negative control (water)
#then combine reps 

#nitro without water
nitro_subs <- nitro %>%
  filter(Substance != "negative control") 

#get water value to substract from all other substances
nitro_rep1_water <- nitro %>%
  filter(Substance == "negative control") %>%
  select(experiment, sample, timepoint, Substance, rep, abs) 

#join non water substrates and water substrates. Then substract water abs from substrate abs
nitro_rep1_join <- nitro_subs %>% 
  left_join(nitro_rep1_water, by=c("sample", "timepoint", "experiment", "rep")) %>%
  mutate(adj_abs = abs.x-abs.y) %>% 
  mutate(adj_abs = ifelse(adj_abs < 0, 0, adj_abs))%>%
  filter(sample!="Blank")

#For each plate calculate AWCD 
nitro_AWCD <- nitro_rep1_join %>%
  group_by(sample, timepoint, experiment, rep) %>%
  summarise(sum_abs=sum(adj_abs))%>%
  mutate(AWCD = sum_abs/95)
#Combine AWCD per timepoint, sample, experiemnt, rep to working dataset
nitro_join_AWCD <- nitro_rep1_join %>% 
  left_join(nitro_AWCD, by=c("sample","timepoint", "experiment", "rep"))
#Now combine reps which are differnt plates

#Add design file
biolog_sub_groups_nitro <- biolog_sub_groups %>% 
    filter(plate_type=="PM3B") %>%
    rename(Substance.x=Substance)

nitro_joined_all <- nitro_join_AWCD %>% 
  left_join(biolog_design, by=c("sample")) %>% 
  left_join(biolog_sub_groups_nitro, by="Substance.x")%>%   
  filter(sample != "201-M-EC-2")

i <- with(nitro_joined_all, interaction(treatment, source))

nitro_subs <- ggplot(nitro_joined_all, aes(treatment, Substance.x)) +
  geom_tile(aes(fill = scale(adj_abs, center=FALSE, scale=TRUE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~source) +
  labs(fill = "Absorbance \n Intensity")

nitro_subs

ggsave("../figures/WRC18_Rhizo/nitro_sub_overview.jpg", plot=nitro_subs, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)

#nitro_joined_all_f <- nitro_joined_all %>% filter(timepoint=="T0")
#need to get average for each treatment source combo
nitro_AWCD_plot <- ggplot(data=nitro_joined_all, aes(x=timepoint, y=AWCD, color=treatment)) +
  stat_summary(geom="point", fun.y=mean, na.rm=TRUE, size=3) +
  scale_color_manual(labels=c("Unfertilized", "Fertilized"), values=c("gray","darkgreen")) +
  facet_grid(~ source) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24,    face="bold"), panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.text=element_text(size=24, face="bold"), legend.title = element_text(size=24,        face="bold"))+
  theme(axis.text.x=element_blank()) +
    theme(strip.text.x = element_text(size = 24, face="bold")) +
  ggtitle("Nitrogen Plate AWCD") +
  xlab("Timpoints T0-T11") +
  ylab("AWCD") +
  theme(legend.position = "none")

nitro_AWCD_plot

ggsave("../figures/WRC18_Rhizo/nitro_plate_AWCD.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)



```

```{r nitro plate T04}
#subset to include only T04 
nitro_T <- nitro_joined_all %>% 
  filter(timepoint=="T04") 

i <- with(nitro_T, interaction(treatment, source))

nitro_subs <- ggplot(nitro_T, aes(treatment, Substance.x)) +
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "gray", high = "black") +
  labs(fill = "Absorbance \n Intensity")+
  facet_grid(~ source) 

nitro_subs

ggsave("../figures/WRC18_Rhizo/nitro_sub_overview.jpg", plot=nitro_subs, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)

nitro_subs_groups <- ggplot(nitro_T, aes(group, treatment)) +
  geom_tile(aes(fill = adj_abs), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity")+
  coord_flip()

nitro_subs_groups

n <- nitro_joined_all %>% filter(group == "inorganic")

n.p <- ggplot(n, aes(Substance.x, treatment )) +
  facet_grid(~soure)+
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ treatment) +
  labs(fill = "Absorbance \n Intensity")+
  coord_flip()
#scale(adj_abs, scale=TRUE, center=FALSE)

n.p

p <- nitro_joined_all %>% 
  filter(group == "inorganic")%>%
  filter(Substance.x != "hydroxylamine") 
  

nitro_subs_groups <- ggplot(p, aes(Substance.x, treatment)) +
  facet_grid(~source)+
  geom_tile(aes(fill = scale(adj_abs, scale=TRUE, center=FALSE)), colour = "gray") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity")+
  coord_flip()

nitro_subs_groups

ggsave("../figures/WRC18_Rhizo/nitro_sub_overview.jpg", plot=nitro_subs_groups, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)


```

```{r nitro plate diversity}
#Transform so that substances are columns and sample is row?
#First select just the columns needed... then turn negatve values into 0
nitro_joined_adj_sub <- nitro_joined_all %>%
  select(Substance.x, sample,  adj_abs )  #%>%

#I want Substance.x to be columns and sample to be rows and vales to be adj_abs
m <- melt(nitro_joined_adj_sub)
nitro.cast<-cast(data=m, formula = sample ~ Substance.x,fun.aggregate = sum)

#Make sample column rownames and remove sample column
rownames(nitro.cast) = nitro.cast$sample
nitro.cast <- nitro.cast[, -c(1)]

nitro_shannon <- diversity(nitro.cast, "shannon")

wrc.phylo<-wrc.phylo[,-c(35)]
wrc.phylo <- wrc.phylo %>%
  filter(sample != "WRC18_201")
wrc.phylo <- cbind(wrc.phylo, nitro_shannon)
wrc.phylo<-wrc.phylo[,-c(37:38)]


nitro_shan <- ggplot(wrc.phylo, aes(x=treatment, y=nitro_shannon, fill = treatment))+
  geom_boxplot()+
  facet_grid(~source)+
  scale_fill_manual(labels=c("Unfertilized", "Fertilized"), values=c("gray","darkgreen")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24,    face="bold"), panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.position = "none")+
  theme(axis.text.x=element_blank()) +
    theme(strip.text.x = element_text(size = 24, face="bold")) +
  ggtitle("Nitrogen Substrate-Use Diversity") +
  xlab("") +
  ylab("Shannon H'") 

nitro_shan

# run full parametric statistical model
nitro.lm <- lm(nitro_shannon ~ treatment * source, data = wrc.phylo)
plot(nitro.lm)
anova(nitro.lm)

# run linear regression on diversity
nitro.reg <- lm(PD~nitro_shannon, data = wrc.phylo)
summary(nitro.reg)

nitro.reg <- lm(shannon~nitro_shannon, data = wrc.phylo)
summary(nitro.reg)

ggsave("../figures/WRC18_Rhizo/nitro_shan.jpg", plot=nitro_shan, device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)
```

```{r biolog PM3B nitro plate substance count}
#For each treatment*source get count of group (group is the category of substrate)
#Presence absence; set a threshold positive value
#Left join substrate group to biolog results to add group to each substrate?
nitro_joined_adj <- nitro_joined_all %>%
  #select(Substance.x, sample,  adj_abs) %>%
 #rename(Substance = Substance.x) %>%
  mutate(PA = ifelse(adj_abs < 0.5, 0, 1) )

i <- with(nitro_joined_adj, interaction(source, treatment))

ggplot(nitro_joined_adj, aes(x=group, y=PA, fill=treatment)) +
  geom_bar(stat="identity") +
  facet_grid(~ source) +
  coord_flip()   

nitro_joined_adj_groups <- nitro_joined_adj %>%
  #select(sample, treatment, source, group, PA)%>%
  group_by(treatment, source, group)%>%
  #count(treatment, source, group)
  summarize(sum=sum(PA))
  
nitro_groups <- ggplot(nitro_joined_adj_groups, aes(treatment, source)) +
  geom_tile(aes(fill = scale(sum, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) 

nitro_groups1 <- ggplot(nitro_joined_adj_groups, aes(treatment, sum)) +
  geom_bar(stat="identity")  +
  facet_grid(~ source) 
nitro_groups1

```

```{r BIOLOG phosphate-use AWCD}
#Want to calculate the shannon diversity index @ T04 for a sample 
#And then combine all samples for a treatment 

#Subset master biolog file to include only nitro plates
po4 <- biolog_master[(biolog_master$plate_type %in% "PM4A"),]
#subset to include only T04

#for each plate in po4_T, calculate adjusted abs for each substance in each rep, 3 reps per plate
#plate_num, sample, substance, rep 1 is well == 1:4, rep 2 is well == 5:8, rep 3 is well == 9:12 
#Each rep has its own negative control (water)

#po4 A1-E12
po4_rep1 <- po4 %>%
 filter(Substance != "negative control") %>%
 filter(Well %in% c(1:12) & Row %in% c("A","B","C","D","E"))  

#sulfur F1-H12
sulfur_rep1 <- po4 %>%
 filter(Substance != "negative control") %>%
 filter(Well %in% c(1:12),Row %in% c("F","G","H")) 
  
#get water value to substract from all other substances
po4_rep1_water <- po4 %>%
  filter(Substance == "negative control",Row ==c("A")) %>%
  select(experiment, sample, timepoint, Substance, rep, abs)  

sulfur_rep1_water <- po4 %>%
  filter(Substance == "negative control",Row ==c("F")) %>%
  select(experiment, sample, timepoint, Substance, rep, abs)   

#Timepoint T04, All substances except water in rep1 (ie wells 1-4 and rows a-h) joined with water value
po4_rep1_join <- po4_rep1 %>% 
  left_join(nitro_rep1_water, by=c("sample", "timepoint", "experiment", "rep")) %>%
  mutate(adj_abs = abs.x-abs.y) %>% 
  mutate(adj_abs = ifelse(adj_abs < 0, 0, adj_abs))%>%
  filter(sample!="Blank")

sulfur_rep1_join <- sulfur_rep1 %>%
  left_join(nitro_rep1_water, by=c("sample", "timepoint", "experiment", "rep")) %>%
  mutate(adj_abs = abs.x-abs.y) %>% 
  mutate(adj_abs = ifelse(adj_abs < 0, 0, adj_abs))%>%
  filter(sample!="Blank")

#For each plate calculate AWCD 
po4_AWCD <- po4_rep1_join %>%
  group_by(sample, timepoint, experiment, rep) %>%
  summarise(sum_abs=sum(adj_abs))%>%
  mutate(AWCD = sum_abs/59)

sulfur_AWCD <- sulfur_rep1_join %>%
  group_by(sample, timepoint, experiment, rep) %>%
  summarise(sum_abs=sum(adj_abs))%>%
  mutate(AWCD = sum_abs/36)

  #Combine AWCD per timepoint, sample, experiemnt, rep to working dataset
po4_join_AWCD <- po4_rep1_join %>% 
  left_join(po4_AWCD, by=c("sample","timepoint", "experiment", "rep"))

sulfur_join_AWCD <- po4_rep1_join %>% 
  left_join(sulfur_AWCD, by=c("sample","timepoint", "experiment", "rep"))

#Now combine reps which are differnt plates
```

```{r P graph AWCD}
#Need to add a design file... PO4
biolog_sub_groups_po4 <- biolog_sub_groups %>% 
    filter(plate_type=="PM4A") %>%
    rename(Substance.x=Substance)

po4_joined_all <- po4_join_AWCD %>% 
  left_join(biolog_design, by=c("sample")) %>% 
  left_join(biolog_sub_groups_po4, by="Substance.x") %>%
  filter(sample != "201-M-EC-2")

i <- with(po4_joined_all, interaction(treatment, source))

po4_subs <- ggplot(po4_joined_all, aes(treatment, Substance.x)) +
  geom_tile(aes(fill = scale(adj_abs, center=FALSE, scale = TRUE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity")

po4_subs

ggsave("../figures/WRC18_Rhizo/po4_sub_overview.jpg", plot=po4_subs, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)

#po4_joined_all_f <- po4_joined_all %>% filter(timepoint=="T0")
#need to get average for each treatment source combo
po4_AWCD_plot <- ggplot(data=po4_joined_all, aes(x=timepoint, y=AWCD, color=treatment)) +
  stat_summary(geom="point", fun.y=mean, na.rm=TRUE, size=3) +
  scale_color_manual(labels=c("Unfertilized", "Fertilized"), values=c("gray","darkgreen")) +
  facet_grid(~ source) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24,    face="bold"), panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.text=element_text(size=24, face="bold"), legend.title = element_text(size=24,        face="bold"))+
  theme(axis.text.x=element_blank()) +
    theme(strip.text.x = element_text(size = 24, face="bold")) +
  ggtitle("Phosphorus AWCD") +
  xlab("Timpoints T0-T11") +
  ylab("AWCD") +
  theme(legend.position = "none")

po4_AWCD_plot

ggsave("../figures/WRC18_Rhizo/po4_plate_AWCD.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)


po4_subs_groups <- ggplot(po4_joined_all, aes(group, treatment)) +
  geom_tile(aes(fill = scale(adj_abs, center=FALSE, scale = TRUE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity")+
  coord_flip()

po4_subs_groups

ggsave("../figures/WRC18_Rhizo/po4_groups_overview.jpg", plot=po4_subs_groups, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)

```
```{r S AWCD graph}
#Need to add a design file... sulfur
biolog_sub_groups_sulfur <- biolog_sub_groups %>% 
    filter(plate_type=="PM4A") %>%
    rename(Substance.x=Substance)

sulfur_joined_all <- sulfur_join_AWCD %>% 
  left_join(biolog_design, by=c("sample")) %>% 
  left_join(biolog_sub_groups_sulfur, by="Substance.x") %>%
  filter(sample != "201-M-EC-2")

i <- with(sulfur_joined_all, interaction(treatment, source))

sulfur_subs <- ggplot(sulfur_joined_all, aes(treatment, Substance.x)) +
  geom_tile(aes(fill = scale(adj_abs, center=FALSE, scale = TRUE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity")

sulfur_subs

ggsave("../figures/WRC18_Rhizo/sulfur_sub_overview.jpg", plot=sulfur_subs, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)

#sulfur_joined_all_f <- sulfur_joined_all %>% filter(timepoint=="T0")
#need to get average for each treatment source combo
sulfur_AWCD_plot <- ggplot(data=sulfur_joined_all, aes(x=timepoint, y=AWCD, color=treatment)) +
  stat_summary(geom="point", fun.y=mean, na.rm=TRUE, size=3) +
  scale_color_manual(labels=c("Unfertilized", "Fertilized"), values=c("gray","darkgreen")) +
  facet_grid(~ source) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24,    face="bold"), panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.text=element_text(size=24, face="bold"), legend.title = element_text(size=24,        face="bold"))+
  theme(axis.text.x=element_blank()) +
    theme(strip.text.x = element_text(size = 24, face="bold")) +
  ggtitle("Sulfur AWCD") +
  xlab("Timpoints T0-T11") +
  ylab("AWCD") +
  theme(legend.position = "none")

sulfur_AWCD_plot

ggsave("../figures/WRC18_Rhizo/sulfur_plate_AWCD.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)


sulfur_subs_groups <- ggplot(sulfur_joined_all, aes(group, treatment)) +
  geom_tile(aes(fill = scale(adj_abs, center=FALSE, scale = TRUE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) +
  labs(fill = "Absorbance \n Intensity")+
  coord_flip()

sulfur_subs_groups

ggsave("../figures/WRC18_Rhizo/sulfur_groups_overview.jpg", plot=po4_subs_groups, device=NULL, path=NULL, scale=1, width=8, height=5, dpi=900, limitsize=TRUE)
```


```{r P and S Diversity}
#Transform so that substances are columns and sample is row?
#First select just the columns needed... then turn negatve values into 0
po4_joined_adj_sub <- po4_joined_all %>%
  select(Substance.x, sample,  adj_abs )  #%>%

#I want Substance.x to be columns and sample to be rows and vales to be adj_abs
m <- melt(po4_joined_adj_sub)
po4.cast<-cast(data=m, formula = sample ~ Substance.x,fun.aggregate = sum)
#Make sample column rownames and remove sample column
rownames(po4.cast) = po4.cast$sample
po4.cast <- po4.cast[, -c(1)]

po4_shannon <- diversity(po4.cast, "shannon")

wrc.phylo<- wrc.phylo[,-c(38)]
wrc.phylo <- wrc.phylo %>%
  filter(sample != "WRC18_201")
wrc.phylo <- cbind(wrc.phylo, po4_shannon)
wrc.phylo<- wrc.phylo[,-c(38)]


po4_shan <- ggplot(wrc.phylo, aes(x=treatment, y=po4_shannon, fill = treatment))+
  geom_boxplot()+
  facet_grid(~source)+
  scale_fill_manual(labels=c("Unfertilized", "Fertilized"), values=c("gray","darkgreen")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24,    face="bold"), panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.position = "none")+
  theme(axis.text.x=element_blank()) +
    theme(strip.text.x = element_text(size = 24, face="bold")) +
  ggtitle("Phosphorus Substrate-Use Diversity") +
  xlab("") +
  ylab("Shannon H'") 

po4_shan

# run full parametric statistical model
po4.lm <- lm(po4_shannon ~ treatment * source, data = wrc.phylo)
plot(po4.lm)
anova(po4.lm)

# run linear regression on diversity
po4.reg <- lm(PD~po4_shannon, data = wrc.phylo)
summary(po4.reg)

po4.reg <- lm(shannon~po4_shannon, data = wrc.phylo)
summary(po4.reg)

ggsave("../figures/WRC18_Rhizo/po4_shan.jpg", plot=po4_shan, device=NULL, path=NULL, scale=1, width=5, height=5, dpi=900, limitsize=TRUE)
```
```{r sulfur diversity}
#SULFUR
#Transform so that substances are columns and sample is row?
#First select just the columns needed... then turn negatve values into 0
sulfur_joined_adj_sub <- sulfur_joined_all %>%
  select(Substance.x, sample,  adj_abs ) 

#I want Substance.x to be columns and sample to be rows and vales to be adj_abs
m <- melt(sulfur_joined_adj_sub)
sulfur.cast<-cast(data=m, formula = sample ~ Substance.x,fun.aggregate = sum)
#Make sample column rownames and remove sample column
rownames(sulfur.cast) = sulfur.cast$sample
sulfur.cast <- sulfur.cast[, -c(1)]

sulfur_shannon <- diversity(sulfur.cast, "shannon")

wrc.phylo<- wrc.phylo[,-c(38)]
wrc.phylo <- wrc.phylo %>%
  filter(sample != "WRC18_201")
wrc.phylo <- cbind(wrc.phylo, sulfur_shannon)

sulfur_shan <- ggplot(wrc.phylo, aes(x=treatment, y=sulfur_shannon, fill = treatment))+
  geom_boxplot()+
  facet_grid(~source)+
  scale_fill_manual(labels=c("Unfertilized", "Fertilized"), values=c("gray","darkgreen")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  theme(axis.title = element_text(size=24, face="bold"), 
        axis.text.x = element_text(size=24, face="bold"),  axis.text.y = element_text(size=24,    face="bold"), panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=24, face="bold")) +
  #Set legend text size
  theme(legend.position = "none")+
  theme(axis.text.x=element_blank()) +
    theme(strip.text.x = element_text(size = 24, face="bold")) +
  ggtitle("Sulfur Substrate-Use Diversity") +
  xlab("") +
  ylab("Shannon H'") 

sulfur_shan

# run full parametric statistical model
sulfur.lm <- lm(sulfur_shannon ~ treatment * source, data = wrc.phylo)
plot(sulfur.lm)
anova(sulfur.lm)

# run linear regression on diversity
sulfur.reg <- lm(PD~sulfur_shannon, data = wrc.phylo)
summary(sulfur.reg)

sulfur.reg <- lm(shannon~sulfur_shannon, data = wrc.phylo)
summary(sulfur.reg)

ggsave("../figures/WRC18_Rhizo/sulfur_shan.jpg", plot=sulfur_shan, device=NULL, path=NULL, scale=1, width=8, height=8, dpi=900, limitsize=TRUE)
```

```{r biolog PM4A P substance count}
#For each treatment*source get count of group (group is the category of substrate)
#Presence absence; set a threshold positive value
#Left join substrate group to biolog results to add group to each substrate?

po4_joined_adj <- po4_joined_all %>% 
  #rename(Substance = Substance.x) %>%
  #left_join(biolog_sub_groups, by="Substance") %>%
  mutate(PA = ifelse(adj_abs < 0.5, 0, 1) )
  
i <- with(po4_joined_adj, interaction(treatment, source))

ggplot(po4_joined_adj, aes(x=group, y=PA, fill=treatment)) +
  geom_bar(stat="identity") +
  facet_grid(~ source) +
  coord_flip()   

po4_joined_adj_groups <- po4_joined_adj %>%
  #select(sample, treatment, source, group, PA)%>%
  group_by(treatment, source, group)%>%
  #count(treatment, source, group)
  summarize(sum=sum(PA))
  
po4_groups <- ggplot(po4_joined_adj_groups, aes(treatment, source)) +
  geom_tile(aes(fill = scale(sum, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) 

po4_groups1 <- ggplot(po4_joined_adj_groups, aes(treatment, sum)) +
  geom_bar(stat="identity")  +
  facet_grid(~ source) 
po4_groups1

```


```{r biolog sulfur plate substance count}
#For each treatment*source get count of group (group is the category of substrate)
#Presence absence; set a threshold positive value
#Left join substrate group to biolog results to add group to each substrate?
sulfur_joined_adj <- sulfur_T %>% 
 # rename(Substance = Substance.x) %>%
  #left_join(biolog_sub_groups, by="Substance") %>%
  mutate(PA = ifelse(adj_abs < 0.5, 0, 1) )

#So my sample labels are diff in deisgn file and biolog file
#extract unique sample ids from biolog and then add column to meta

  
i <- with(sulfur_joined_adj, interaction(treatment, source))

ggplot(sulfur_joined_adj, aes(x=group, y=PA, fill=treatment)) +
  geom_bar(stat="identity") +
  facet_grid(~ source) +
  coord_flip()  

sulfur_joined_adj_groups <- sulfur_joined_adj %>%
  #select(sample, treatment, source, group, PA)%>%
  group_by(treatment, source, group)%>%
  #count(treatment, source, group)
  summarize(sum=sum(PA))
  
sulfur_groups <- ggplot(sulfur_joined_adj_groups, aes(treatment, source)) +
  geom_tile(aes(fill = scale(sum, scale=TRUE, center=FALSE)), colour = "white") + 
  scale_fill_gradient(low = "white", high = "black") +
  facet_grid(~ source) 

sulfur_groups1 <- ggplot(sulfur_joined_adj_groups, aes(treatment, sum)) +
  geom_bar(stat="identity")  +
  facet_grid(~ source) 
sulfur_groups1

```

```{r Mantel Tests}
#need to remove WRC-201. Unfortunately this plate fell off shelf and is unreliable
otu.201 <- otu.2[-c(17),]
otu.dist <- vegdist(otu.201, method = "bray")

unifrac.dist.201 <- unifrac[-c(17),-c(17)]
unifrac.dist.201 <- as.dist(unifrac.dist.201, upper = T, diag = T)

eco.dist <- vegdist(eco.cast, method = "euclidean")
nitro.dist <-vegdist(nitro.cast, method = "euclidean")
po4.dist <- vegdist(po4.cast, method="euclidean")
sulfur.dist <- vegdist(sulfur.cast, method = "euclidean")

mantel.rtest(otu.dist, eco.dist, nrepet = 999)
mantel.rtest(otu.dist, nitro.dist, nrepet = 999)
mantel.rtest(otu.dist, po4.dist, nrepet = 999)
mantel.rtest(otu.dist, sulfur.dist, nrepet = 999)

mantel.rtest(eco.dist, nitro.dist, nrepet = 999)
mantel.rtest(po4.dist, nitro.dist, nrepet = 999)
mantel.rtest(po4.dist, eco.dist, nrepet = 999)
mantel.rtest(po4.dist, sulfur.dist, nrepet = 999)
mantel.rtest(sulfur.dist, eco.dist, nrepet = 999)
mantel.rtest(sulfur.dist, nitro.dist, nrepet = 999)

mantel.rtest(unifrac.dist.201,eco.dist, nrepet = 999)
mantel.rtest(unifrac.dist.201, nitro.dist, nrepet = 999)
mantel.rtest(unifrac.dist.201, po4.dist, nrepet = 999)
mantel.rtest(unifrac.dist.201, sulfur.dist, nrepet = 999)

mantel.rtest(otu.dist, unifrac.dist.201, nrepet = 999)
```




```{r biolog eco plate ORDINATION GRAPH}
#eco.dist <- vegdist(eco.cast, method = "euclidean")
#biolog_design is design file

# Principal Coordinates Analysis
dataREL.dist <- eco.dist

pcoa <- cmdscale(dataREL.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1b <- round(pcoa$eig[1] / sum(pcoa$eig), 3) * 100
explainvar2b <- round(pcoa$eig[2] / sum(pcoa$eig), 3) * 100
sum.eigb <- sum(explainvar1b, explainvar2b)

explainvar1b #39.2
explainvar2b #16.1
```

```{r Ordination (PCoA) - biolog ecoplate}
# Principal Coordinates Analysis - bulk and teas
new.data <-cbind(biolog_design,eco.cast)
pcoa.groups <- paste(new.data$treatment, new.data$source, new.data$ditch, sep = "_")
pcoa.points <- data.frame(pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
trt.c <- c("M","M","M","MF","MF","MF")
source.c <- c("bulk soil","forb","grass","bulk soil","forb","grass")
pcoa.cent.dataframe.trts <- as.data.frame(pcoa.cent.dataframe)
pcoa.cent.dataframe.trts$trt <- as.factor(trt.c)
pcoa.cent.dataframe.trts$source <- as.factor(source.c)

#Plot
df1a <- as.data.frame(pcoa.cent.dataframe.trts)
plot1a <- ggplot(df1a, aes(x=V1, y=V2, colour=trt, shape = source,
                           group = interaction(trt, source))) + theme_bw() 
plot1a <- plot1a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + geom_point(aes(colour=trt.c), size=5, stroke = 1.25, show.legend = TRUE) +
scale_colour_manual(labels = c("mowed","mowed/fertilized"), 
                    values = c("darkgray", "darkgreen")) + 
scale_shape_manual(labels = c("bulk soil", 
                              "forb",
                              "grass"), 
                   values = c(16,22,15)) +
theme(axis.title = element_text(size=24), axis.text = element_text(size=24),
      axis.text.x = element_text(size=24), 
      panel.border = element_rect(colour = "black", size = 1.25)) + 
theme(axis.ticks.length = unit(0.3, "cm")) + 
  #Set plot title textsize
  theme(plot.title=element_text(size=24)) +
  #Set legend text size
  theme(legend.text=element_text(size=24), legend.title = element_text(size=24))+
xlab("PCoA 1 (39.2%)") + ylab("PCoA 2 (16.1%)") +
  labs(colour = "treatment", shape = "source") +
  guides(colour = guide_legend(override.aes = list(pch=16, size = 10)),
         shape = guide_legend(override.aes = list(size = 10)))+ggtitle("Carbon Substrate-Use")+
  theme(legend.position="none")

plot1a

ggsave("../figures/WRC18_Rhizo/Ordination_ecoplate.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=5, dpi=300, limitsize=TRUE)
```


```{r biolog NITRO plate ORDINATION GRAPH}
#biolog_design is design file
#make sure to run nitro ecodist before proceeding

# Principal Coordinates Analysis
dataREL.dist <- nitro.dist

pcoa <- cmdscale(dataREL.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1b <- round(pcoa$eig[1] / sum(pcoa$eig), 3) * 100
explainvar2b <- round(pcoa$eig[2] / sum(pcoa$eig), 3) * 100
sum.eigb <- sum(explainvar1b, explainvar2b)

explainvar1b #38.3
explainvar2b #10.9
```

```{r Ordination (PCoA) - biolog Nitrogen plate}
# Principal Coordinates Analysis - bulk and teas
new.data <-cbind(biolog_design,nitro.cast)
pcoa.groups <- paste(new.data$treatment, new.data$source, new.data$ditch, sep = "_")
pcoa.points <- data.frame(pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
trt.c <- c("M","M","M","MF","MF","MF")
source.c <- c("bulk soil","forb","grass","bulk soil","forb","grass")
pcoa.cent.dataframe.trts <- as.data.frame(pcoa.cent.dataframe)
pcoa.cent.dataframe.trts$trt <- as.factor(trt.c)
pcoa.cent.dataframe.trts$source <- as.factor(source.c)

#Plot
df2a <- as.data.frame(pcoa.cent.dataframe.trts)
plot2a <- ggplot(df2a, aes(x=V1, y=V2, colour=trt, shape = source,
                           group = interaction(trt, source))) + theme_bw() 
plot2a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + geom_point(aes(colour=trt.c), size=5, stroke = 1.25, show.legend = TRUE) +
scale_colour_manual(labels = c("mowed","mowed/fertilized"), 
                    values = c("darkgray", "darkgreen")) + 
scale_shape_manual(labels = c("bulk soil", 
                              "forb",
                              "grass"), 
                   values = c(16,22,15)) +
theme(axis.title = element_text(size=24), axis.text = element_text(size=24),
      axis.text.x = element_text(size=24), 
      panel.border = element_rect(colour = "black", size = 1.25)) + 
theme(axis.ticks.length = unit(0.3, "cm")) + 
   #Set plot title textsize
  theme(plot.title=element_text(size=24)) +
  #Set legend text size
  theme(legend.text=element_text(size=24), legend.title = element_text(size=24))+
xlab("PCoA 1 (38.3%)") + ylab("PCoA 2 (10.7%)") +
  labs(colour = "treatment", shape = "source") +
  guides(colour = guide_legend(override.aes = list(pch=16, size = 4)),
         shape = guide_legend(override.aes = list(size = 4)))+ggtitle("Phosphorus Substrate-Use")+
  theme(legend.position = "none")

ggsave("../figures/WRC18_Rhizo/Ordination_nitroplate.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=5, dpi=300, limitsize=TRUE)
```

```{r biolog PHOSPHORUS plate ORDINATION GRAPH}
#biolog_design is design file
#make sure to run PM3 ecodist before proceeding
phos.dist <- vegdist(po4.cast, method = "euclidean")
# Principal Coordinates Analysis
dataREL.dist <- phos.dist

pcoa <- cmdscale(dataREL.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1b <- round(pcoa$eig[1] / sum(pcoa$eig), 3) * 100
explainvar2b <- round(pcoa$eig[2] / sum(pcoa$eig), 3) * 100
sum.eigb <- sum(explainvar1b, explainvar2b)

explainvar1b #48.3
explainvar2b #12.5
```

```{r Ordination (PCoA) - biolog phosphorus plate}
# Principal Coordinates Analysis - bulk and teas
new.data <-cbind(biolog_design,po4.cast)
pcoa.groups <- paste(new.data$treatment, new.data$source, new.data$ditch, sep = "_")
pcoa.points <- data.frame(pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
trt.c <- c("M","M","M","MF","MF","MF")
source.c <- c("bulk soil","forb","grass","bulk soil","forb","grass")
pcoa.cent.dataframe.trts <- as.data.frame(pcoa.cent.dataframe)
pcoa.cent.dataframe.trts$trt <- as.factor(trt.c)
pcoa.cent.dataframe.trts$source <- as.factor(source.c)

#Plot
df3a <- as.data.frame(pcoa.cent.dataframe.trts)
plot3a <- ggplot(df3a, aes(x=V1, y=V2, colour=trt, shape = source,
                           group = interaction(trt, source))) + theme_bw() 
plot3a <- plot3a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + geom_point(aes(colour=trt.c), size=5, stroke = 1.25, show.legend = TRUE) +
scale_colour_manual(labels = c("mowed","mowed/fertilized"), 
                    values = c("darkgray", "darkgreen")) + 
scale_shape_manual(labels = c("bulk soil", 
                              "forb",
                              "grass"), 
                   values = c(16,22,15)) +
theme(axis.title = element_text(size=14), axis.text = element_text(size=14),
      axis.text.x = element_text(size=14), 
      panel.border = element_rect(colour = "black", size = 1.25)) + 
theme(axis.ticks.length = unit(0.3, "cm")) + 
   #Set plot title textsize
  theme(plot.title=element_text(size=24)) +
  #Set legend text size
  theme(legend.text=element_text(size=24), legend.title = element_text(size=24))+
xlab("PCoA 1 (39.6%)") + ylab("PCoA 2 (13.0%)") +
  labs(colour = "treatment", shape = "source") +
  guides(colour = guide_legend(override.aes = list(pch=16, size = 4)),
         shape = guide_legend(override.aes = list(size = 4)))+ggtitle("Phosphorus Use Composition")+
  theme(legend.position="none")
plot3a
ggsave("../figures/WRC18_Rhizo/Ordination_PO4plate.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=5, dpi=300, limitsize=TRUE)
```

```{r biolog SULFUR plate ORDINATION GRAPH}
#biolog_design is design file
#make sure to run SULFUR ecodist before proceeding
sulfur.dist <- vegdist(sulfur.cast, method = "euclidean")
# Principal Coordinates Analysis
dataREL.dist <- sulfur.dist

pcoa <- cmdscale(dataREL.dist, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1b <- round(pcoa$eig[1] / sum(pcoa$eig), 3) * 100
explainvar2b <- round(pcoa$eig[2] / sum(pcoa$eig), 3) * 100
sum.eigb <- sum(explainvar1b, explainvar2b)

explainvar1b #45.6
explainvar2b #10.1
```

```{r Ordination (PCoA) - biolog sulfur plate}
# Principal Coordinates Analysis - bulk and teas
new.data <-cbind(biolog_design,sulfur.cast)
pcoa.groups <- paste(new.data$treatment, new.data$source, new.data$ditch, sep = "_")
pcoa.points <- data.frame(pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
trt.c <- c("M","M","M","MF","MF","MF")
source.c <- c("bulk soil","forb","grass","bulk soil","forb","grass")
pcoa.cent.dataframe.trts <- as.data.frame(pcoa.cent.dataframe)
pcoa.cent.dataframe.trts$trt <- as.factor(trt.c)
pcoa.cent.dataframe.trts$source <- as.factor(source.c)

#Plot
df4a <- as.data.frame(pcoa.cent.dataframe.trts)
plot4a <- ggplot(df4a, aes(x=V1, y=V2, colour=trt, shape = source,
                           group = interaction(trt, source))) + theme_bw() 
plot4a <- plot4a + theme(panel.grid.major = element_blank(), 
               panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + geom_point(aes(colour=trt.c), size=5, stroke = 1.25, show.legend = TRUE) +
scale_colour_manual(labels = c("mowed","mowed/fertilized"), 
                    values = c("darkgray", "darkgreen")) + 
scale_shape_manual(labels = c("bulk soil", 
                              "forb",
                              "grass"), 
                   values = c(16,22,15)) +
theme(axis.title = element_text(size=24), axis.text = element_text(size=24),
      axis.text.x = element_text(size=24), 
      panel.border = element_rect(colour = "black", size = 1.25)) + 
theme(axis.ticks.length = unit(0.3, "cm")) + 
   #Set plot title textsize
  theme(plot.title=element_text(size=24)) +
  #Set legend text size
  theme(legend.text=element_text(size=24), legend.title = element_text(size=24))+
xlab("PCoA 1 (39.6%)") + ylab("PCoA 2 (13.0%)") +
  labs(colour = "treatment", shape = "source") +
  guides(colour = guide_legend(override.aes = list(pch=16, size = 4)),
         shape = guide_legend(override.aes = list(size = 4)))+ggtitle("Sulfur Substrate-Use")+
  theme(legend.position = "none")

plot4a

ggsave("../figures/WRC18_Rhizo/Ordination_sulfurplate.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=5, dpi=300, limitsize=TRUE)


```

```{r plant cover}
plant_cover <- read.csv("../data/WRC18_Rhizo/Summary_2018_WRC_plants.csv")

ggplot(plant_cover, aes(x=treatment, y=cover, fill= plant)) +
  geom_boxplot() + 
  facet_grid(~ year)

#group by label and rep 
#summarise --- just want percent for poster
plant_cover_reps <- plant_cover %>%
  group_by(year, block, treatment, plant)%>%
  summarise(mean_cover=mean(cover))

ggplot(plant_cover_reps, aes(x=treatment, y=mean_cover, fill=plant)) +
  geom_boxplot() + 
  facet_grid(~ year)

plant_cover_reps_2018 <- plant_cover_reps %>% 
  filter(year=="2018")
  
ggplot(plant_cover_reps_2018, aes(x=treatment, y=mean_cover, fill=plant)) +
  geom_boxplot() 
###
  
```
```{r load taxonomy data into phyloseq object}
# Import otu and tax into phyloseq
otu <- otu_table(t(otu.2), taxa_are_rows = T)
tax <- import_mothur(mothur_constaxonomy_file =  taxfile)
#rownames(meta) <- meta$group
sample <- sample_data(meta)
#import into phyloseq
phy<-phyloseq(otu,tax)
row.names(sample)=meta$sample
sample_names(sample)
#Assign rownames to be Sample ID's
# Merge mothurdata object with sample metadata
meta_merge <- merge_phyloseq(phy,sample)

#meta_merge

#assign column names
colnames(tax_table(meta_merge))
colnames(tax_table(meta_merge)) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus")
colnames(tax_table(meta_merge))
rank_names(meta_merge)

#Inspecting phyloseq object
#sample_variables(meta_merge)
#sample_names(meta_merge)
#tax_table(meta_merge)[1:5, 2:6]

#treat_merge = merge_samples(meta_merge, sample_data(meta_merge)$pxt)
#otu_table(treat_merge)[1:5,1:5]
```

```{r phyloseq heat maps of taxa composition - phylum}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample
wrc_phylum <- meta_merge %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01) %>%                        # Filter out low abundance taxa
  arrange(Phylum)                                      # Sort data frame alphabetically by phylum

# Set colors for plotting
phylum_colors <- c("brown","orange", "blue", "red","yellow","purple")
  
  #c("#CBD588", "#5F7FC7","#DA5724", "#508578", "#CD9BCD",
  # "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", 
  #"#8569D5", "#5E738F","#D1A33D", "#8A7C64", "#599861", "#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

# Plot 
#s <- as.factor(wrc_phylum$Sample)
#wrc_phylum$Sample <- ordered(s,levels=c("bulkM","avM", "ecM","bulkMF","avMF", "ecMF"))
#s <- revalue(s, c("bulkM"="bulk unfertilized", "avM"="grass unfertilized", "ecM"="forb unfertilized","bulkMF"="bulk fertilized", "avMF"="grass fertilized", "ecMF"="forb fertilized"))
#s <- mapvalues(s, from = c("bulkM","avM", "ecM","bulkMF","avMF", "ecMF"), to = c("bulk unfertilized","grass unfertilized", "forb unfertilized","bulk fertilized", "grass fertilized", "forb fertilized"))
i <- with(wrc_phylum,interaction(source, treatment))

wrc_plot<- ggplot(wrc_phylum, aes(x = Phylum, y = Abundance, fill = i)) + 
  #facet_grid(Type ~ .) +
  #####UPDATE THIS ON EACH GRAPH ######
  geom_bar(stat = "summary", fun.y = "mean") +
  scale_fill_manual(values = phylum_colors) +
  # Remove x axis title
  theme(axis.title.x = element_blank()) + 
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 1.0%) \n") +
  ggtitle("Phylum Composition of Bacterial Communities By Site") 

wrc_plot + coord_flip()

ggsave(file="../figures/phylum_stack.pdf", width=9, height=9, dpi=300)

# heat map of rel abundance
heat_rel_abund <- ggplot(wrc_phylum, aes(treatment, Phylum)) +
  geom_tile(aes(fill = Abundance), colour = "white") + 
  scale_fill_gradient(low = "gray", high = "black") +
  facet_grid(~source) +
  labs(fill = "Relative Abundance") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
    panel.background = element_blank(), axis.line = element_line(colour = "black"))

heat_rel_abund

ggsave("../figures/WRC18_Rhizo/phylum_heat.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=3, dpi=300, limitsize=TRUE)
```

```{r phyloseq Class stacked graphs of taxa composition}
# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample
wrc_class <- meta_merge %>%
  tax_glom(taxrank = "Class") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01) %>%                        # Filter out low abundance taxa
  arrange(Class)                                     # Sort data frame alphabetically by phylum

heat_rel_abund <- ggplot(wrc_class, aes(treatment, Class)) +
  geom_tile(aes(fill = (Abundance)), colour = "white") + 
  scale_fill_gradient(low = "gray", high = "black") +
  facet_grid(~source) +
  labs(fill = "Relative Abundance") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
    panel.background = element_blank(), axis.line = element_line(colour = "black"))

heat_rel_abund

ggsave("../figures/WRC18_Rhizo/Class_heat.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=3, dpi=300, limitsize=TRUE)
```

```{r phyloseq Family stacked graphs of taxa composition}
# melt to long format (for ggploting) 
# prune out phyla below 1% in each sample
wrc_family <- meta_merge %>%
  tax_glom(taxrank = "Family") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01, Class == "Actinobacteria") %>%                        # Filter out low abundance taxa
  arrange(Family)                                     # Sort data frame alphabetically by phylum

heat_rel_abund <- ggplot(wrc_family, aes(treatment, Family)) +
  geom_tile(aes(fill = Abundance), colour = "white") + 
  scale_fill_gradient(low = "gray", high = "black") +
  facet_grid(~source) +
  labs(fill = "Relative Abundance") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
    panel.background = element_blank(), axis.line = element_line(colour = "black"))

heat_rel_abund

ggsave("../figures/WRC18_Rhizo/family_heat.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=3, dpi=300, limitsize=TRUE)

```

```{r phyloseq Family - Actinobacteria  stacked graphs of taxa composition}
# melt to long format (for ggploting) 
# prune out phyla below 1% in each sample
wrc_family <- meta_merge %>%
  tax_glom(taxrank = "Family") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01, Class == "Alphaproteobacteria") %>%                        # Filter out low abundance taxa
  arrange(Family)                                     # Sort data frame alphabetically by phylum

heat_rel_abund <- ggplot(wrc_family, aes(treatment, Family)) +
  geom_tile(aes(fill = Abundance), colour = "white") + 
  scale_fill_gradient(low = "gray", high = "black") +
  facet_grid(~source) +
  labs(fill = "Relative Abundance") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
    panel.background = element_blank(), axis.line = element_line(colour = "black"))

heat_rel_abund

ggsave("../figures/WRC18_Rhizo/alpha_heat.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=3, dpi=300, limitsize=TRUE)

```

```{r phyloseq Genus stacked graphs of taxa composition}
# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample
wrc_genus <- meta_merge %>%
  tax_glom(taxrank = "Genus") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.01) %>%                        # Filter out low abundance taxa
  arrange(Genus)                                     # Sort data frame alphabetically by phylum

heat_rel_abund <- ggplot(wrc_genus, aes(treatment, Genus)) +
  geom_tile(aes(fill = Abundance), colour = "white") + 
  scale_fill_gradient(low = "gray", high = "black") +
  facet_grid(~source) +
  labs(fill = "Relative Abundance") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
    panel.background = element_blank(), axis.line = element_line(colour = "black"))

heat_rel_abund

ggsave("../figures/WRC18_Rhizo/Genus_heat.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=5, height=3, dpi=300, limitsize=TRUE)
```