---
title: "WRC2015-2018_Rhizo"
author: "Regina A. B. Bledsoe, Ariane L. Peralta"
date: "May 8, 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

```{r load packages, include=FALSE}
getwd()

#Set source R tools
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")

#load req'd packages 
require("vegan")
require("dplyr")
require("tidyverse")
require("ecodist")
require("ggplot2")
require("phyloseq")
require("agricolae")

#Set Std Err and Conf Int
se <- function(x, ...) {
  sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))
}
ci <- function(x, ...) {
  1.96 * sd(x, na.rm = TRUE)
}

set.seed(42)
```

```{r load data, include=FALSE}
getwd()
#Load data files 
#Assign file names to variables
sharedfile = "../data/WRC18_Rhizo/WRC15_18Rhizo.opti_mcc.shared"
taxfile = "../data/WRC18_Rhizo/WRC15_18Rhizo.opti_mcc.0.03.cons.taxonomy"
metafile = "../data/WRC18_Rhizo/2015_2018_WestResearchCampus_Sampling_Summary.csv"

#Read in design file
meta <- read.csv(metafile)
#head(meta)

#Read in OTU file
otu <- read.otu(sharedfile)

#Read in taxonomy file
#tax <- read.tax(taxfile)
#read.tax does not work and I was unable to solve issue. I use phyloseq import_mothur as a work around. 
tax <- import_mothur(mothur_constaxonomy_file =  taxfile)
tax.df <- as.data.frame(tax)
colnames(tax.df)=c("Domain","Phylum","Class", "Order","Family","Genus")
tax <- tax_table(tax.df)
```

```{r filter meta and otu datasets and rarefy}
#Remove OTUs with less than 2 occurences across all sites
otu.2 <- otu[, which(colSums(otu) > 9)]
otu.2 <- as.data.frame(otu[-c(33:48),])

#FROM OTU - Select only bulk from 2018 
#First join meta and otu. Filter and then remove extra columns
otu.2.filter <- bind_cols(meta, otu.2) %>%
  filter(source=="bulk") %>%
  filter(year=="2018") 
head(otu.2.filter)
rownames(otu.2.filter)=otu.2.filter$sample
otu.2.filter <- otu.2.filter[,-c(1:26)]
head(otu.2.filter)

#FROM META - Select only bulk from 2018 
meta.filter <- meta %>%
  filter(source=="bulk") %>%
  filter(year=="2018")

#What sample has lowest read count? WRC18_174 24513, highest WRC15_167 69111
otu2<- colSums(t(otu.2.filter))
otu2[which.min(colSums(t(otu.2.filter)))] 
otu2[which.max(colSums(t(otu.2.filter)))]
#remove WRC18_201
#rownames(otu.2)
#otu.2 <- otu.2[-c(41),]
#meta$sample
#meta <- meta[-c(41),]

#graph of read counts, need to work on y axis
c<-as.data.frame(otu.2.filter)
c$colsum <- colSums(t(c))
c$group <- rownames(c)
p<-ggplot(c, aes(x=group,y=colsum))+geom_bar(stat="identity")+coord_flip()
p

#Rarefy
#Set min sample number
min.N <- min(rowSums(otu.2.filter))
#rrarefy returns a randomly rarefied community data frame or vector of selected size
#This is the otu set that should be used for beta diversity metrics
otu.rare <- rrarefy(otu.2.filter, min.N)
```

```{r rarefaction curve}
#rarefy returns a species richness estimate in a random subsample of selected size
otu.rare.rich <- rarefy(x = otu.2, sample=min.N, se=T)
#Plot rarefaction curve
###I am not sure how to interpret this. I think it suggests that I did not capture all of the diversity? 
rarecurve(x = otu.2, step = 20, col="blue", cex=0.6, las=1)
abline(0,1,col='red')
text(1000,1000,"1:1",pos=2, col='red')
```

```{r Relative abundance and PERMANOVA }
#transforms read counts to relative abundance
#Using rarefied OTU dataset
otu.2.rel <- otu.rare
for(i in 1:dim(otu.rare)[1]){
  otu.2.rel[i,] <- otu.rare[i,]/sum(otu.rare[i,])
}

#Combine experimental deisng meta data with OTU relative abundance data
metaotu <- cbind(meta.filter, otu.2.rel)
#Where does otu data begin, remove for PERMANOVA
#metaotu[1,20:21]

#PERMANOVA with source and treatment as factors
#Among all treatments source (soil source) and treatment
metaotu.ad <- adonis(metaotu[,-c(1:26)] ~ fertilized*mowed, method = "bray", data = metaotu, perm=1000, set.seed=42)
#metaotu.ad <- adonis(metaotu[,-c(1:19)] ~source*treatment*year, method = "bray", data = metaotu, perm=1000, set.seed=42)
metaotu.ad 

#Tries to make a 1GB vector and fails even with 1 permutation
#require("RVAideMemoire")
#metaotu.ad.1 <- metaotu[which(metaotu$treatment=='M'),]
#p <- pairwise.perm.manova(metaotu.ad.1[,-c(1:19)],metaotu.ad.1$source,nperm=1)

#Another attempt at pairwise adonis but not sure that this method has been well worked out by the authors
#require("pairwiseAdonis")
#data(metaotu)
#pairwise.adonis(metaotu[,-c(1:19)],metaotu$pxt)

#source (soil source) is significant but the PCoA suggests that it is only significant between source and rhizosphere and not source functionl type (grass, forb). Because I can't run a global pairwise analysis I will run individual pairwise analysis between source(soil sources) within a treatment. 
#source (soil source) within treatment

#Pairwise adonis/PERMANOVA of soil source within a treatment
#MOWED UNFERTILIZED
#metaotu.M <- metaotu[which(metaotu$treatment=='M'),]
#metaotu.M.GF <- metaotu.M[which(metaotu.M$source!='bulk'),]
#metaotu.M.BG <- metaotu.M[which(metaotu.M$source!='ec'),]
#metaotu.M.BF <- metaotu.M[which(metaotu.M$source!='av'),]

#metaotu.M.GF.ad <- adonis(metaotu.M.GF[,-c(1:19)] ~ source, method = "bray", data = metaotu.M.GF, perm=1000, set.seed=42)
#metaotu.M.GF.ad 

#metaotu.M.BG.ad <- adonis(metaotu.M.BG[,-c(1:19)] ~ source, method = "bray", data = metaotu.M.BG, perm=1000, set.seed=42)
#metaotu.M.BG.ad 

#metaotu.M.BF.ad <- adonis(metaotu.M.BF[,-c(1:19)] ~ source, method = "bray", data = metaotu.M.BF, perm=1000, set.seed=42)
#metaotu.M.BF.ad 

#MOWED FERTILIZED
#metaotu.MF <- metaotu[which(metaotu$treatment=='MF'),]
#metaotu.MF.GF <- metaotu.MF[which(metaotu.MF$source!='bulk'),]
#metaotu.MF.BG <- metaotu.MF[which(metaotu.MF$source!='ec'),]
#metaotu.MF.BF <- metaotu.MF[which(metaotu.MF$source!='av'),]

#metaotu.MF.GF.ad <- adonis(metaotu.MF.GF[,-c(1:19)] ~ source, method = "bray", data = metaotu.MF.GF, perm=1000, set.seed=42)
#metaotu.MF.GF.ad 

#metaotu.MF.BG.ad <- adonis(metaotu.MF.BG[,-c(1:19)] ~ source, method = "bray", data = metaotu.MF.BG, perm=1000, set.seed=42)
#metaotu.MF.BG.ad 

#metaotu.MF.BF.ad <- adonis(metaotu.MF.BF[,-c(1:19)] ~ source, method = "bray", data = metaotu.MF.BF, perm=1000, set.seed=42)
#metaotu.MF.BF.ad 

```
```{r Calculate distance matrix and PCoA components }

otu.2.rel.dist<-vegdist(otu.2.rel, method="bray")

# Principal Coordinates Analysis
otu.2.rel.dist.pcoa <- cmdscale(otu.2.rel.dist, k=3, eig=TRUE, add=FALSE)
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1a <- round(otu.2.rel.dist.pcoa$eig[1] / sum(otu.2.rel.dist.pcoa$eig), 3) * 100
explainvar2a <- round(otu.2.rel.dist.pcoa$eig[2] / sum(otu.2.rel.dist.pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)

explainvar1a
explainvar2a

```

```{r PCoA Plot }
all.equal(rownames(meta.filter), rownames(otu.2.rel))

#pcoa.groups <- paste(meta$source, meta$treatment, meta$year, sep = "_")
pcoa.groups <- paste(meta.filter$fertilized, meta.filter$mowed, sep = "_")


pcoa.points <- data.frame(otu.2.rel.dist.pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1)) # fert
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2))  # mow
#pcoa.col2 <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 3)) #year

#pcoa.shape <- ordered(c("A. virginicus", "Bulk Soil", "E. caroliniana"))
#pcoa.shape <- ordered(pcoa.shape,levels=c("bulk","av", "ec"))
#pcoa.shape <- revalue(pcoa.shape, c("bulk"="Bulk Soil", "av"="Grass", "ec"="Forb"))
#pcoa.shape <- mapvalues(pcoa.shape, from = c("bulk","av","ec"), to = c("Bulk Soil", "Grass", "Forb"))

levels(pcoa.col) <- c("Fertilized", "Unfertilized")
levels(pcoa.shape) <-c("Mowed", "Unmowed")

#Removed RS samples becuase only have soil porperty values for RP samples
#Run envfit to see which soil varibales are predicators 
colnames(meta)

##fit<-envfit(otu.2.rel.dist.pcoa$points, meta[,-c(1:3,8:9,11,18:19)], perm=1000, na.rm=T, set.seed=42)
##fit

##A <-as.list(fit$vectors)
##vec<-as.data.frame(fit$vectors$arrows*sqrt(fit$vectors$r)*0.35)
##p<- as.data.frame(A$pvals)
##vec <- cbind(vec, p) 
##vec <-  subset(vec, A$pvals<=0.05)

##vec$parm<-rownames(vec)
##colnames(vec)
#vec$parm<-c("Moisture","pH","C%","N%")

#plot PCoA
df1a <- as.data.frame(pcoa.cent.dataframe)
plot1 <- ggplot(df1a, aes(x=V1, y=V2,  color=pcoa.col, shape = pcoa.shape, 
                 group = interaction(pcoa.col, pcoa.shape))) + theme_bw() 
plot1 +
#group = interaction(pcoa.col, pcoa.shape), colour=pcoa.col2))
    geom_point(aes(shape = pcoa.shape, colour = pcoa.col), size=8) +
  #geom_point(aes(colour=pcoa.col2), size =5.0)+ 
  #geom_point(aes(colour=pcoa.col2), size =4.0)+
  #removes gridlines from plot
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  #Plot soil property vectors from envfit
  ##geom_segment(data=vec, aes(x=0,xend=Dim1,y=0,yend=Dim2), size=1, 
  ##    arrow = arrow(length = unit(0.2, "cm")),colour="black",inherit.aes=F)+ 
  #Label soil vectors. Using annotate because I can't figure out how to move segment text
  #Note when re-run should double check envfit stats to make sure that soil parameters are the same
  ##geom_text(data=vec,aes(x=Dim1,y=Dim2,label=vec$parm),size=5, inherit.aes=F)+
  #annotate("text",x=-0.02,y=0.2, label="C%") +
  #annotate("text",x=0.02,y=0.2, label="N%") +
  #annotate("text",x=0,y=-.2, label="moisture") +
  #annotate("text",x=0.1,y=-.225, label="pH") +
  #Set error bacrs for geom_point
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  #Set colors for treatments
  #scale_colour_manual(values = c("#999999", "#000000")) +
  scale_colour_manual(values = c("darkgreen", "gray")) +
  #Would have liked to set shapes but couldn't get this to work
  #scale_shape_manual(values = c(22, 21, 24)) +
  #Sets map coordinates
  ##coord_cartesian(xlim = c(-0.1, 0.1), ylim = c(-0.1, 0.1)) +
  #Sets axis text and put border around plot
  theme(axis.title = element_text(size=14), 
        axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
        panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=14)) +
  #Set legend text size
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  #Sets size of tick marks on axis
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  #Sets labels for plot title, axis titles, and legend headings
  xlab("PCoA 1 (25.1%)") + ylab("PCoA 2 (12.9%)") + 
  labs(shape = "Mowed") +
  labs(colour="Fertilized") +
  ggtitle("16S rRNA Microbial Community Analysis") 
  
#Save a copy of the plot
ggsave("../figures/WRC18_Rhizo/phylo_WRC2018all.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)

```

```{r Presence_Absence and Species richness }
#Transform data to reflect only presence or absence ie 0 or 1
otu.2.PA<-(otu.rare>0*1)
#sanity check 
#otu.2.PA[,1:5]
#Add meta data
metaotu.2 <- cbind(meta.filter, otu.2.PA)
#metaotu.2[1:15]
#Transform True/False to 0/1 
cols <- sapply(metaotu.2, is.logical)
metaotu.2[,cols]<-lapply(metaotu.2[,cols],as.numeric)
#head(metaotu.2)
```

```{r diversity metrics: richness}
#chao1 richness index
chao1 <- estimateR(otu.2.filter)
c <- as.data.frame(chao1)
ct <- t(c)
row.names(meta.filter)=row.names(ct)
otu.div <- cbind(meta.filter, ct)

#interaction variable so can use Tukey
levels(otu.div$fertilized)<-c("Unfertilized", "Fertilized")
i <- with(otu.div, interaction(fertilized, mowed))

richness.lm <- aov(S.chao1 ~ fertilized + mowed + i, data = otu.div)
plot(richness.lm)
anova(richness.lm)
#Get Tukey post hoc groups p<=0.01*
t <- HSD.test(richness.lm, "i", group=TRUE, alpha=.01)
t
plot(t)

#Set variable names for plot
#levels(meta$treatment) <- c("Unfertilized", "Fertilized")
#p <- meta$source
#p <- ordered(p,levels=c("bulk","av", "ec"))
#p <- revalue(p, c("bulk"="Bulk Soil", "av"="Grass", "ec"="Forb"))
#p <- mapvalues(p, from = c("bulk","av","ec"), to = c("Bulk Soil", "Grass", "Forb"))


#Plot
chao<- ggplot(otu.div, aes(x=treatment, y=S.chao1))  +
  geom_boxplot(binaxis='y', stackdir='center') + 
  scale_fill_manual(values=c("#999999", "#000000", "darkgreen", "gray")) +
#Set labels for Tukey groups
#annotate("text",x=0.82,y=5200, label="b") +
#annotate("text",x=1.81,y=5400, label="b") +
#annotate("text",x=2.81,y=5725, label="b") +
#annotate("text",x=1.1815,y=7050, label="a") +
#annotate("text",x=2.19,y=6600, label="bc") + 
#annotate("text",x=3.19,y=6250, label="ab") + 
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=14,face="bold"),
          axis.text=element_text(size=14),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="bottom") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "", y = "Chao1 species richness")
chao
#Save a copy of the plot
ggsave("../figures/chao1_all.jpg", plot=chao, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)


```

```{r diversity metrics - calculate diversity and evenness}
# Calculate diversity metrics
shannon <- diversity(otu.rare, "shannon")
simp <- diversity(otu.rare, "simpson")
invsimp <-diversity(otu.rare, "invsimpson")
# Pielous evenness (uses presence/absence data)
J <- shannon/log(specnumber(otu.2.PA[,-c(1:1)]))

#Adding diversity fields to dataset
otu.div$shannon <- NULL
otu.div$J <- NULL
otu.div$simp <- NULL
otu.div$invsimp <- NULL
otu.div <- cbind(otu.div,shannon,J, simp, invsimp)
```

```{r diversity metrics - display shannon }
#Run richness first. interaction, levels, and treatments for plotting set there
shannon.lm <- aov(shannon ~ fertilized + mowed + i, data = otu.div)
plot(shannon.lm)
anova(shannon.lm)
i = with(otu.div,(interaction(fertilized,mowed)))
#Tukey groups at p<=0.01*
t <- HSD.test(shannon.lm, "i", group=TRUE, alpha=0.05)
t
plot(t)

#p is set in richness chuck
shan<- ggplot(otu.div, aes(x=treatment, y=shannon))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000"))  + 
  #facet_grid(~ year) +
#annotate("text",x=0.82,y=6.1, label="bc") +
#annotate("text",x=1.81,y=5.9, label="c") +
#annotate("text",x=2.81,y=6.05, label="c") +
#annotate("text",x=1.1815,y=6.5, label="a") +
#annotate("text",x=2.19,y=6.3, label="ab") + 
#annotate("text",x=3.19,y=6.35, label="ab") + 
#Set plot elements
theme_bw()  +
#Remove plot gridlines
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
#Set axis title and text properties, tick marks, and labels
theme(axis.title=element_text(size=14,face="bold"),
          axis.text=element_text(size=14), #axis.text.x = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1), legend.position="none") + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "Treatment", y = "Shannon diversity H'")

shan

ggsave("../figures/shannon_all.jpg", plot=shan, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)
```

```{r diversity metrics - Pielous evennes J}
J.lm <- aov(J ~ fertilized + mowed + i , data = otu.div)
plot(J.lm)
anova(J.lm)
#TukeyHSD(J.lm)
t <- HSD.test(J.lm, "i", group=TRUE, alpha=0.05)
t
plot(t)

even<- ggplot(otu.div, aes(x=i, y=J, fill=treatment))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000")) +
#Set Tukey groups
annotate("text",x=0.82,y=0.77, label="ab") +
annotate("text",x=1.81,y=0.75, label="b") +
annotate("text",x=2.81,y=0.76, label="b") +
annotate("text",x=1.1815,y=0.80, label="a") +
annotate("text",x=2.19,y=0.78, label="a")  + 
annotate("text",x=3.19,y=0.80, label="a")  + 
#Set plot properties
theme_bw()  +
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.title.y=element_text(margin=margin(r=10)), 
          axis.title.x=element_text(margin=margin(t=10))) +
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(size=14), panel.border = 
          element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + labs(x = "Soil Source", y = "Pileou's eveness") + 
    theme(legend.text=element_text(size=14), legend.title = element_text(size=14), 
          legend.position = "bottom")

even 

ggsave("../figures/even_all.jpg", plot=even, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

```

```{r environmental parameters}
library(naniar)

meta.env <- meta %>%
  replace_with_na_at(.vars = c("n_no3_ug_gsoil"), condition = ~.x < 0) %>%
  mutate(avg_temp=(temp1_C + temp2_C + temp3_C)/3)

#SUMMARY TABLES
#2015
meta.env.2015 <- meta.env %>%
  filter(year=="2015")
meta.env.2015<- meta.env.2015[-c(39),]

meta.env.2015.means <- meta.env.2015 %>%
  group_by(source, treatment) %>%
  summarise_all(funs(mean,sd), na.rm=T)

meta.env.2015.rhizo <- meta.env.2015 %>%
  filter(block %in% c("2","4","6","8")) %>%
  filter(treatment %in% c("M","MF")) 

meta.env.2015.rhizo.means <- meta.env.2015.rhizo %>%
  group_by(source, treatment) %>%
  summarise_all(funs(mean,sd), na.rm=T)

#2018
meta.env.2018 <- meta.env %>%
  filter(year=="2018")

meta.env.2018.means <- meta.env.2018 %>%
  group_by(source, treatment) %>%
  summarise_all(funs(mean,sd), na.rm=T)

meta.env.2018.rhizo <- meta.env.2018 %>%
  filter(block %in% c("2","4","6","8")) %>%
  filter(treatment %in% c("M","MF")) 

meta.env.2018.rhizo.means <- meta.env.2018.rhizo %>%
  group_by(source, treatment) %>%
  summarise_all(funs(mean,sd), na.rm=T)


####STOPPED HERE --- 2018 analyses

i <- with(meta.filter, interaction(fertilized, mowed))

nh4 <- aov(n_nh4_ug_gsoil ~ fertilized+mowed + i, meta.filter)
plot(nh4)
anova(nh4)

nh4<- ggplot(meta.filter, aes(x=treatment, y=n_nh4_ug_gsoil))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333","222222"))
nh4

no3 <- aov(n_no3_ug_gsoil ~ fertilized+mowed + i, meta.filter)
plot(no3)
anova(no3)

no3<- ggplot(meta.filter, aes(x=treatment, y=n_nh4_ug_gsoil))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333","222222"))
no3

po4h <- aov(p_po4_hcl_ug_gsoil ~ fertilized+mowed + i, meta.filter)
plot(po4h)
anova(po4h)

po4h<- ggplot(meta.filter, aes(x=treatment, y=p_po4_hcl_ug_gsoil))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333","222222"))
po4h

po4w <- aov(p.po4_w_ug_gsoil ~ fertilized+mowed + i, meta.filter)
plot(po4w)
anova(po4w)

po4w<- ggplot(meta.filter, aes(x=treatment, y=p.po4_w_ug_gsoil))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333","222222"))
po4w

pH <- aov(pH ~ fertilized+mowed + i, meta.filter)
plot(pH)
anova(pH)

pH<- ggplot(meta.filter, aes(x=treatment, y=pH))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333","222222"))
pH

c_p <- aov(c_percent ~ fertilized+mowed + i, meta.filter)
plot(c_p)
anova(c_p)

c_p<- ggplot(meta.filter, aes(x=treatment, y=c_percent))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333","222222"))
c_p

n_p <- aov(n_percent ~ fertilized+mowed + i, meta.filter)
plot(n_p)
anova(n_p)

n_p<- ggplot(meta.filter, aes(x=treatment, y=n_percent))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333","222222"))
n_p

c_n <- aov(C_N ~ fertilized+mowed + i, meta.filter)
plot(c_n)
anova(c_n)

c_n<- ggplot(meta.filter, aes(x=treatment, y=C_N))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333","222222"))
c_n

temp <- aov(temp2_C ~ fertilized+mowed + i, meta.filter)
plot(temp)
anova(temp)

temp<- ggplot(meta.filter, aes(x=treatment, y=temp2_C))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333","222222"))
temp

moist <- aov(moist_percent ~ fertilized+mowed + i, meta.filter)
plot(moist)
anova(moist)

moist<- ggplot(meta.filter, aes(x=treatment, y=moist_percent))  +
  geom_boxplot() + scale_fill_manual(values=c("#999999", "#000000","333333","222222"))
moist

```


```{r phylogenetic analysis}
#Make unifrac distance matrix
#Run PERMANOVA
#Make PCoA plot
#Compare taxonomic div to phylogenetic diversity

```
```{r (Phylogenetic Diversity), include=FALSE}
# Phylogenetic Diversity - analysis by Mario Muscarella modified by Gina Bledsoe
# Load Tree
tree <- read.tree("../data/WRC18_Rhizo/WRC15-18Rhizo.bac.rename.tree")
tree$tip.label <- gsub("\\|", "", tree$tip.label)
sum(tree$tip.label %in% colnames(otu.2) == FALSE)
# Small Branches
sum(tree$edge.length < 0.0000001)
# Import Unifrac Matrix
unifrac.raw <- read.delim("../data/WRC18_Rhizo/WRC15-18Rhizo.bac.tree1.weighted.phylip.dist", skip = 1, header = F, row.names = 1)
colnames(unifrac.raw) <- rownames(unifrac.raw)
rownames(unifrac.raw)
#Remove spaces from rownames
require("stringr")
rownames(unifrac.raw) <- str_replace_all(rownames(unifrac.raw), fixed(" "), "")
rownames(unifrac.raw)
#narrow unifrac to 2018 Rhizopshere samples
unifrac <- unifrac.raw[which(row.names(unifrac.raw) %in%
                                   row.names(otu.2)),
                            which(row.names(unifrac.raw) %in%
                                   row.names(otu.2))]
dim(unifrac)
# Make into Distance Matrix
unifrac.dist <- as.dist(unifrac, upper = T, diag = T)

# Calculate Phylo Diversity
phylo_final <- pd(otu.2, tree, include.root = F)

rownames(meta)=meta$sample

all.equal(rownames(meta), rownames(phylo_final))

dim(phylo_final)
dim(meta)

wrc.phylo <- cbind(meta,phylo_final)
#wrc.phylo <- wrc.phylo[,-c(33:36)]

# run full parametric statistical model
PD.lm <- lm(PD ~ treatment * source * year, data = wrc.phylo)
plot(PD.lm)
anova(PD.lm)

# run linear regression on diversity
PD.reg <- lm(PD~shannon, data = wrc.phylo)
summary(PD.reg)

ggplot(wrc.phylo,aes(x=shannon,y=PD))+stat_summary(fun.data=mean_cl_normal) + 
  geom_smooth(method='lm',formula=y~x) +
  geom_point(aes(color=treatment)) +
  facet_grid(~ year)

ggplot(wrc.phylo,aes(x=shannon,y=PD, color=treatment))+stat_summary(fun.data=mean_cl_normal) + 
  geom_smooth(method='lm',formula=y~x) +
  geom_point(aes(color=treatment))+
  facet_grid(~ year)

ggplot(wrc.phylo,aes(x=shannon,y=PD, color=source))+stat_summary(fun.data=mean_cl_normal) + 
  geom_smooth(method='lm',formula=y~x) +
  geom_point(aes(color=source))+
  facet_grid(~ year)

p1 <- ggplot(wrc.phylo, aes(x = source, y = PD, fill = treatment), size=16) +
  geom_flat_violin(aes(fill = treatment),position = position_nudge(x = .1, y = 0), adjust = 1.5, trim = FALSE, alpha = .5, colour = NA)+
  geom_point(aes(x = source, y = PD, colour = treatment),position = position_jitter(width = .05), size = 1.5, shape = 20)+
  geom_boxplot(aes(x = source, y = PD, fill = treatment),outlier.shape = NA, alpha = .5, width = .1, colour = "black") + labs(x = "Sampling Date", y = "Phylogenetic Diversity") 
p1

p2 <- ggplot(wrc.phylo, aes(x = treatment, y = PD, fill = source), size=16) +
  geom_flat_violin(aes(fill = source),position = position_nudge(x = .1, y = 0), adjust = 1.5, trim = FALSE, alpha = .5, colour = NA)+
  geom_point(aes(x = treatment, y = PD, colour = source),position = position_jitter(width = .05), size = 1.5, shape = 20)+
  geom_boxplot(aes(x = treatment, y = PD, fill = source),outlier.shape = NA, alpha = .5, width = .1, colour = "black") + labs(x = "Sampling Date", y = "Phylogenetic Diversity") 
p2
```

```{r Calculate distance matrix and PCoA components }
#dim(otu.2.rel.dist)
#otu.2.rel.dist<-vegdist(unifrac.dist, method="bray")

# Principal Coordinates Analysis
otu.2.rel.dist.pcoa <- cmdscale(unifrac.dist, k=3, eig=TRUE, add=FALSE)
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1a <- round(otu.2.rel.dist.pcoa$eig[1] / sum(otu.2.rel.dist.pcoa$eig), 3) * 100
explainvar2a <- round(otu.2.rel.dist.pcoa$eig[2] / sum(otu.2.rel.dist.pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)

explainvar1a
explainvar2a

```

```{r PCoA Plot Phylogenetic}

#PERMANOVA with source and treatment as factors
#Among all treatments source (soil source) and treatment
####I need to combine meta and unifrac.dist. Then run adonis 
m <- as.matrix(unifrac.dist)
rownames(meta) <- meta$sample
all.equal(rownames(m), rownames(meta))
unifrac.dist.meta <- cbind(meta, m)
metaotu.ad <- adonis(unifrac.dist.meta[,-c(1:23)] ~source*treatment*year, method = "bray", data = unifrac.dist.meta, perm=1000, set.seed=42)
#metaotu.ad <- adonis(metaotu[,-c(1:19)] ~source*treatment*year, method = "bray", data = metaotu, perm=1000, set.seed=42)
metaotu.ad 


all.equal(rownames(meta), rownames(phylo_final))
# Set treatments
treat1 <- as.factor(meta$source)
levels(treat1) <- ordered(c("bulk","grass", "forb"))
treat2 <- as.factor(meta$treatment)
treat3 <-as.factor(meta$year)
#levels(treat3) <- c("M", "MF")

pcoa.groups <- paste(meta$source, meta$treatment, meta$year, sep = "_")
#pcoa.groups <- paste(meta$source, meta$treatment, sep = "_")


pcoa.points <- data.frame(otu.2.rel.dist.pcoa$points, group = pcoa.groups)

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # treatment
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # source
pcoa.col2 <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 3)) #year

#pcoa.shape <- ordered(c("A. virginicus", "Bulk Soil", "E. caroliniana"))
#pcoa.shape <- ordered(pcoa.shape,levels=c("bulk","av", "ec"))
#pcoa.shape <- revalue(pcoa.shape, c("bulk"="Bulk Soil", "av"="Grass", "ec"="Forb"))
#pcoa.shape <- mapvalues(pcoa.shape, from = c("bulk","av","ec"), to = c("Bulk Soil", "Grass", "Forb"))

levels(pcoa.col) <- c("Unfertilized","Fertilized")

#Removed RS samples becuase only have soil porperty values for RP samples
#Run envfit to see which soil varibales are predicators 
colnames(meta)

###Are these just the environmental parameters have values for bulk and rhizospehre soils?
#fit<-envfit(otu.2.rel.dist.pcoa$points, meta[,-c(1:12)], perm=1000, na.rm=T, set.seed=42)
#fit

#A <-as.list(fit$vectors)
#vec<-as.data.frame(fit$vectors$arrows*sqrt(fit$vectors$r)*0.35)
#p<- as.data.frame(A$pvals)
#vec <- cbind(vec, p) 
#vec <-  subset(vec, A$pvals<=0.05)

#vec$parm<-rownames(vec)
#colnames(vec)
#vec$parm<-c("Moisture","pH","C%","N%")

#plot PCoA
df1a <- as.data.frame(pcoa.cent.dataframe)
plot1 <- ggplot(df1a, aes(x=V1, y=V2,  color=pcoa.col, shape = pcoa.shape, 
                 group = interaction(pcoa.col, pcoa.shape), color=pcoa.col2)) + theme_bw() +
#group = interaction(pcoa.col, pcoa.shape), colour=pcoa.col2))
  geom_point(aes(shape = pcoa.shape, colour = pcoa.col), size=7) + 
  geom_point(aes(colour=pcoa.col2), size =5.0) +
  #geom_point(aes(colour=pcoa.col2), size =4.0)+
  #removes gridlines from plot
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  #Plot soil property vectors from envfit
  #geom_segment(data=vec, aes(x=0,xend=Dim1,y=0,yend=Dim2), size=1, 
  #   arrow = arrow(length = unit(0.2, "cm")),colour="black",inherit.aes=F)+ 
  #Label soil vectors. Using annotate because I can't figure out how to move segment text
  #Note when re-run should double check envfit stats to make sure that soil parameters are the same
  #geom_text(data=vec,aes(x=Dim1,y=Dim2,label=vec$parm),size=5, inherit.aes=F)+
  #annotate("text",x=-0.02,y=0.2, label="C%") +
  #annotate("text",x=0.02,y=0.2, label="N%") +
  #annotate("text",x=0,y=-.2, label="moisture") +
  #annotate("text",x=0.1,y=-.225, label="pH") +
  #Set error bacrs for geom_point
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  #Set colors for treatments
  #scale_colour_manual(values = c("#999999", "#000000")) +
  scale_colour_manual(values = c("gray", "darkgreen","purple","red","yellow","blue","gray","pink","magenta")) +
  #Would have liked to set shapes but couldn't get this to work
  #scale_shape_manual(values = c(22, 21, 24)) +
  #Sets map coordinates
  coord_cartesian(xlim = c(-.1, 0.1), ylim = c(-0.1, 0.1)) +
  #Sets axis text and put border around plot
  theme(axis.title = element_text(size=14), 
        axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
        panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title textsize
  theme(plot.title=element_text(size=14)) +
  #Set legend text size
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  #Sets size of tick marks on axis
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  #Sets labels for plot title, axis titles, and legend headings
  xlab("PCoA 1 (39.9%)") + ylab("PCoA 2 (23.4%)") + 
  labs(shape = "Soil Source") +
  labs(colour="treatment") +
  ggtitle("Phylogenetic Diversity") 
  
plot1 

#Save a copy of the plot
ggsave("../figures/WRC15-18_Rhizo/16SrRNA_phylo_WRC2018Rhizo.jpg", plot=last_plot(), device=NULL, path=NULL, scale=1, width=8.7, height=6.3, dpi=900, limitsize=TRUE)

```